{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d4907c98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'context': ['. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptMHV-JHM is able to infect neural cells where CEACAM1 expression level is very low, at \\nleast in part because its spike can mediate receptor-independent entry (131, 132). Taken \\ntogether, the membrane fusion mechanism of MHV spike depends on both proteolysis and \\nreceptor binding, and it may or may not depend on the low pH of endosomes; in addition, \\nreceptor-independent membrane fusion by MHV-JHM spike contributes to the neutral \\ntropism of MHV-JHM.\\nResearch on the cell entry mechanism of SARS-CoV has led to novel findings. First, SARS-\\nCoV spike is not cleaved by proprotein convertases during virus packaging and hence \\nremains intact on mature virions (133, 134). Instead, SARS-CoV enters host cells through \\nendocytosis, and its spike is processed by lysosomal proteases (e.g., cathepsin L and \\ncathepsin B) (135–137). This is supported by the observation that inhibitors against either \\nendosomal acidification or lysosomal cysteine proteases block SARS-CoV entry. However, \\nlow pH itself is not a trigger for SARS-CoV entry. This is supported by the observation that \\nwhen expressed on the cell surface and cleaved by exogenous proteases, SARS-CoV spike \\ncan mediate cell-cell fusion with ACE2-expressing cells at neutral pH (136). Thus, the role \\nof low pH in SARS-CoV entry is to activate lysosomal proteases, which further activate \\nSARS-CoV spike for membrane fusion. This is different from influenza virus HA, which is \\nactivated through binding protons in the low-pH environment of endosomes. Second, in \\naddition to lysosomal proteases, both extracellular proteases (e.g., elastases in the respiratory \\ntract) and cell surface proteases (e.g., TMPRSS2 on the surface of lung cells) also activate \\nSARS-CoV spike for membrane fusion (138–142). Due to their cell and tissue specificities, \\nthese proteases likely contribute to the respiratory tract and lung tropism of SARS-CoV . \\nThird, in addition to the cleavage site at the S1/S2 boundary, a second site, S2′, has been \\nidentified at the N terminus of the internal fusion peptide within S2 (143, 144). Whereas the \\ncleavage at the S1/S2 boundary removes the structural constraint of S1 on S2, the cleavage \\nat the S2′ site releases the internal fusion peptide for insertion into target membranes \\n(Figures 1c and 2b). Fourth, it is not clear whether the binding of receptor ACE2 is a trigger \\nfor SARS-CoV spike to fuse membranes. Two electron microscopy studies observed no or \\nmoderate conformational changes of SARS-CoV spike associated with ACE2 binding (18, \\n145). However, other studies suggested that ACE2 binding triggers a conformational change \\nin SARS-CoV spike, which exposes previously cryptic protease sites for cleavage (135, \\n146). The role of ACE2 binding in triggering membrane fusion waits to be further \\ninvestigated. Nevertheless, SARS-CoV entry does not depend on low pH, but it requires at \\nleast two protease cleavages in the spike by lysosomal proteases, extracellular proteases, or \\ncell surface proteases.\\nThe overall cell entry mechanism of MERS-CoV is similar to that of SARS-CoV . Like \\nSARS-CoV spike, MERS-CoV spike must be cleaved at both the S1/S2 boundary and the \\nS2′ site for membrane fusion to occur (147). MERS-CoV also enters host cells through \\nendocytosis and is activated by lysosomal cysteine proteases for membrane fusion (27, 148). \\nMoreover, extracellular proteases and cell surface proteases help activate MERS-CoV entry \\n(148–150). Unlike SARS-CoV , MERS-CoV spike is cleaved by host proprotein convertases \\nduring virus packaging (27, 151). Interestingly, despite recognizing the same receptor DPP4, \\nMERS-CoV and HKU4 spikes differ in their activities to mediate virus entry: HKU4 spike \\nmediates virus entry into bat cells but not human cells, whereas MERS-CoV spike mediates \\nLi Page 10\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscriptvirus entry into both bat and human cells (152). Two residue differences have been identified \\nbetween MERS-CoV and HKU4 spikes that account for this functional difference (152); \\nthey allow MERS-CoV spike, but not HKU4 spike, to be activated by human proprotein \\nconvertases and lysosomal cysteine proteases. Thus, the corresponding two mutations played \\na critical role in the transmission of MERS-CoV from its likely natural reservoir, bats, to \\nhumans, either directly or through intermediate host camels. On the other hand, HKU4 spike \\ncan be activated by bat lysosomal proteases but not human lysosomal proteases, suggesting \\nthat human and bat lysosomal proteases process viral spikes differently. These studies on \\nMERS-CoV entry reveal that different activities of spike-processing proteases from different \\nhosts can pose a barrier for cross-species transmissions of viruses.\\nIn sum, proteolysis has been established as an essential trigger for coronavirus spikes to fuse \\nmembranes, as cleavages at the S1/S2 boundary and S2′ site can remove the structural \\nconstraint of S1 on S2 and release the internal fusion peptide, respectively. Among the host \\nproteases, lysosomal proteases provide the most reliable source for spike processing because \\nthey are ubiquitous and abundant in many cell types. The availability of some other \\nproteases (e.g., proprotein convertases, extracellular proteases, and cell surface proteases) \\ndepends on the types of cells and tissues, regulating tissue tropisms of coronaviruses. \\nMoreover, protease activities from different host species may vary, regulating host ranges of \\ncoronaviruses. Some other triggers for coronavirus entry (e.g., receptor binding and low pH) \\nmay depend on specific coronaviruses or different strains of the same coronavirus. The \\noverall goal of these triggers is to overcome the energy barrier for the conformational \\ntransition of coronavirus spikes.\\nEVOLUTION OF CORONAVIRUS SPIKE PROTEINS\\nCoronavirus spikes, like other class I viral membrane fusion proteins, are amazing \\nmolecules. They single-handedly lead coronaviruses to enter host cells by first binding to a \\nreceptor on the host cell surface and then fusing the viral and host membranes. They exist in \\ntwo distinct conformations: The prefusion trimeric spike contains three receptor-binding S1 \\nheads and a trimeric membranefusion S2 stalk, whereas the postfusion trimeric S2 is a six-\\nhelix bundle with exposed fusion peptides. The transition of the spikes from prefusion to \\npostfusion conformation is regulated by a variety of triggers. Both receptor recognition and \\nmembrane fusion are critical determinants of the host range and tissue tropism of \\ncoronaviruses. How have these complex structures and functions of coronavirus spikes \\nevolved?\\nStructure determinations of coronavirus S1 domains provide insight into the evolution of \\ncoronavirus S1. The finding that betacoronavirus S1-NTDs have a galectin fold indicates a \\nhost origin of coronavirus S1-NTDs. The origin of coronavirus S1-CTDs is less clear. \\nAlphacoronavirus S1-CTDs and host galectins also share some similarity in the structural \\ntopologies of their (β -sandwich folds (Figure 8a), although this similarity is less significant \\nthan that between S1-NTDs and host galectins (Figure 5e,f). β-Sandwich folds are common \\nand stable structures, and two β-sandwich folds may result from convergent evolution with \\nprotein stability as the evolutionary driving force. However, two β-sandwich folds with \\nrelated structural topologies may indicate a common evolutionary ancestor when a \\nLi Page 11\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscriptsignificant number of their constituent β -strands are connected in the same order. Thus, \\nthere is a possibility that S1-CTD and host galectins are evolutionarily related. One possible \\nscenario is that after S1-NTD was generated through gene capture, S1-CTD was generated \\nthrough gene duplication of S1-NTD (Figure 8b). S1-CTDs appear to evolve at a quickened \\npace, as evidenced by the different tertiary structures between alpha-and betacoronavirus S1-\\nCTDs. This may be associated with their location on the very top of the prefusion trimeric \\nspike (Figure 2), which is the most protruding and exposed area on virions. Hence, S1-CTDs \\nare under heavy selective pressure to escape host immune surveillance. The resulting fast-\\npaced evolution of S1-CTDs may have permanently erased their evolutionary traces, except \\nfor the limited information from their structural topology. Whether S1-CTDs originated \\nfrom host galectins or not, the two-domain structure of S1 gives coronaviruses two potential \\nreceptor-binding domains: The more structurally and functionally conserved S1-NTD uses \\nsugar as the fallback receptor, whereas the more aggressively evolving S1-CTD exploits \\nnovel protein receptors (Figure 8).\\nThe structural and functional similarities between coronavirus S2 and other class I viral \\nmembrane fusion proteins are profound. These proteins all exist in prefusion and postfusion \\nconformations. Their prefusion structures can be triggered in a number of similar ways, \\nundergo similar conformational rearrangement, and transition to highly similar postfusion \\nsix-helix bundle structures with exposed fusion peptides. Although it cannot be completely \\nruled out that the same membrane fusion mechanism evolved independently in these viruses, \\nthe complexity and intricacy of this mechanism indicate that class I viral membrane fusion \\nproteins likely share a common evolutionary ancestor.\\nWhich function evolved first for coronaviruses: receptor recognition by S1, membrane \\nfusion by S2, or both simultaneously? Because coronaviruses must enter cells for \\nreplication, membrane fusion is the central function of coronavirus spikes. Receptor \\nrecognition, though, can specifically attach coronaviruses to host cell surfaces and position \\nthe spikes within striking distance of target host membranes. The spike of neurotropic strain \\nMHV-JHM can mediate receptor-independent virus entry into cells that do not express its \\nreceptor, suggesting that receptor binding can be circumvented under some extreme \\nsituations. Therefore, the primordial form of coronavirus spikes might contain S2 only \\n(Figure 8b). Such a primordial spike might function inefficiently because the ancestral virus \\nwould have to diffuse nonspecifically to the close proximity of target cells so that membrane \\nfusion could occur. Later, the spike would evolve a galectin-like S1-NTD through gene \\ncapture, which would enhance its efficiency in mediating virus entry. Next, the spike would \\nevolve an S1-CTD through gene duplication of its S1-NTD or some other mechanism, \\nfurther strengthening its receptor recognition function. Understanding the structure and \\nfunction of coronavirus spikes and their evolution can enhance our understanding of the \\norigin of viruses and the evolutionary relationship between viruses and host cells.\\nAcknowledgments\\nThis work was supported by the National Institutes of Health (R01AI089728, R01AI110700).\\nLi Page 12\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptLITERATURE CITED\\n1. Ksiazek TG, Erdman D, Goldsmith CS, Zaki SR, Peret T, et al. Anovel coronavirus associated with \\nsevere acute respiratory syndrome. N Engl J Med. 2003; 348:1953–66. [PubMed: 12690092] \\n2. Peiris JSM, Lai ST, Poon LLM, Guan Y , Yam LYC, et al. Coronavirus as a possible cause of severe \\nacute respiratory syndrome. Lancet. 2003; 361:1319–25. [PubMed: 12711465] \\n3. Marra MA, Jones SJM, Astell CR, Holt RA, Brooks-Wilson A, et al. The genome sequence of the \\nSARS-associated coronavirus. Science. 2003; 300:1399–404. [PubMed: 12730501] \\n4. Rota PA, Oberste MS, Monroe SS, Nix W A, Campagnoli R, et al. Characterization of a novel \\ncoronavirus associated with severe acute respiratory syndrome. Science. 2003; 300:1394–99. \\n[PubMed: 12730500] \\n5. Zaki AM, van Boheemen S, Bestebroer TM, Osterhaus A, Fouchier RAM. Isolation of a novel \\ncoronavirus from a man with pneumonia in Saudi Arabia. N Engl J Med. 2012; 367:1814–20. \\n[PubMed: 23075143] \\n6. de Groot RJ, Baker SC, Baric RS, Brown CS, Drosten C, et al. Middle East respiratory syndrome \\ncoronavirus (MERS-CoV): announcement of the Coronavirus Study Group. J Virol. 2013; 87:7790–\\n92. [PubMed: 23678167] \\n7. Mole B. Deadly pig virus slips through US borders. Nature. 2013; 499:388. [PubMed: 23887408] \\n8. Stevenson GW, Hoang H, Schwartz KJ, Burrough ER, Sun D, et al',\n",
       "  '.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptThe structures of S1-NTDs from alpha-, gamma-, and deltacoronaviruses are unavailable, \\nbut on the basis of the following observations, they probably all have a galectin fold. First, \\nthe related structural topology between alpha-and betacoronavirus S1-CTDs suggests that \\nthe S1 subunits across different genera have a common evolutionary origin (42). Second, the \\nS1-NTDs from the alphacoronaviruses TGEV and PEDV and the gammacoronavirus IBV all \\nrecognize sugar receptors, although TGEV S1-NTD recognizes N-glycolylneuraminic acid \\n(Neu5Gc) and N-acetylneuraminic acid (Neu5Ac), PEDV S1-NTD recognizes Neu5Ac, and \\nIBV S1-NTD recognizes Neu5Gc (23, 31, 34, 44). Therefore, the S1-NTDs from different \\ngenera are likely all evolutionarily and structurally related.\\nBased on the above studies, the following evolutionary scenario has been proposed for \\ncoronavirus S1-NTDs (19). Through gene capture, ancestral coronaviruses might have \\nacquired a host galectin, which would become the S1-NTD of their spikes. Consequently, \\ncoronaviruses would recognize sugar receptors for cell entry. To aid viral release from \\ninfected cells, some coronaviruses would also evolve a hemagglutinin-esterase protein (HE) \\nas a receptor-destroying enzyme. Later, coronavirus S1-NTDs would evolve a ceiling-like \\nstructure that could protect their sugar-binding site from host immune surveillance; this \\nceiling-like structure is absent in galectins because as host proteins, galectins are not \\nrecognized by the host immune system. The outer surface of the ceiling-like structure in \\nMHV would further evolve structural features that could function as RBMs and recognize \\nCEACAM1. Because protein receptors in general provide higher affinity and specificity for \\nviral attachment than sugar receptors do, the sugar-binding function of MHV S1-NTD \\nwould become dispensable and be lost. The S1-NTDs of some other contemporary \\ncoronaviruses, such as the alphacoronaviruses TGEV and PEDV , the betacoronaviruses \\nBCoV and OC43, and the gammacoronavirus IBV , have retained the lectin function, but \\ntheir sugar specificities have diverged. The galectin-like domain has also been found in a \\nnumber of other viral spikes (103) (Figure 5d,g), including influenza virus HA1, rotavirus \\nVP4, and the adenovirus galectin domain. These viral galectin-like domains display diverse \\nsugar-binding modes, but their sugar-binding sites are all located in cavities, possibly to \\nevade host immune surveillance. Overall, it appears to be a successful strategy for viruses to \\nacquire a host lectin and evolve it into viral RBDs with novel specificity for a protein \\nreceptor or altered specificity for a different sugar receptor.\\nSTRUCTURAL MECHANISM FOR MEMBRANE FUSION BY CORONAVIRUS \\nSPIKE PROTEINS\\nThe coronavirus spike is believed to be a member of the class I viral membrane fusion \\nproteins that also include those from influenza virus, human immunodeficiency virus (HIV), \\nand Ebola virus (Figure 6a). Among these proteins, the hemagglutinin glycoprotein (HA) of \\nthe influenza virus is arguably the best studied (104, 105). HA is expressed as a single-chain \\nprecursor. During molecular maturation, it trimerizes and is cleaved by host proteases into \\nreceptor-binding subunit HA1 and membrane-fusion subunit HA2, which still associate \\ntogether through noncovalent interactions (106, 107). This prefusion state of HA on the \\nnewly packaged virions is a proteolytically primed and metastable trimer. During cell entry, \\nHA1 binds to a sugar receptor on the host cell surface for viral attachment, and then HA1 \\nLi Page 7\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscriptdissociates and HA2 undergoes a dramatic conformational change to transition to the \\npostfusion state. During this transition, three pairs of heptad repeat regions HR-N and HR-C \\nin trimeric HA2 form a six-helix bundle structure. Three previously buried hydrophobic \\nfusion peptides in trimeric HA2 become exposed and insert into the target host membrane. \\nThe fusion peptides and transmembrane anchors are eventually positioned on the same end \\nof the six-helix bundle, bringing the viral and host membranes together to fuse. Because the \\nsix-helix bundle structure is energetically stable, a large amount of energy is released during \\nthe conformational transition of HA, driving membrane fusion forward. However, an initial \\nenergy barrier for the conformational transition of HA must be overcome through the \\naforementioned proteolytic priming and one or more subsequent triggers. These triggers can \\nbe receptor binding (e.g., HIV) (108), low pH (e.g., influenza virus) (109), or a combination \\nof the two (e.g., avian leucosis virus) (110). Consequently, membrane fusion occurs either \\non the host cell surface (e.g., HIV) or in the endosomes (e.g., influenza virus).\\nThe prefusion structures of betacoronavirus MHV and HKU1 spikes were recently \\ndetermined using cryo-electron microscopy (15, 16) (Figure 2a). The overall architecture of \\nthe prefusion coronavirus spikes is similar to, albeit significantly larger and more complex \\nthan, that of influenza virus HA. In each spike, three S1 heads sit on top of a trimeric S2 \\nstalk, preventing S2 from undergoing conformational transitions. Between the two major S1 \\ndomains, S1-CTD is located at the very top of the spike, whereas S1-NTD directly contacts \\nand structurally constrains S2. In the S2 stalk, HR-N forms several helices and arranges \\nitself along the central threefold symmetry axis of trimeric S2, whereas HR-C is poorly \\nordered. Unlike the fusion peptide in influenza virus HA, which is located at the N terminus \\nof HA2, the fusion peptide in coronavirus spikes is located downstream from the N terminus \\nof S2 and is hence an internal fusion peptide. The coronavirus fusion peptide forms a short \\nhelix and a loop, with most of the hydrophobic residues buried inside the prefusion structure. \\nTwo proteolysis sites are essential for the conformational transition of S2, one at the S1/S2 \\nboundary and the other at the N terminus of the internal fusion peptide (111, 112). Like \\nprefusion influenza virus HA, the prefusion coronavirus spike is in a metastable state and \\nprimed to undergo conformational transitions for membrane fusion.\\nAtomic models of full-length postfusion coronavirus spikes are not available, but a negative-\\nstain electron microscopy study on SARS-CoV spike provided a direct view of its \\nconformational transition that is likely associated with membrane fusion during virus entry \\n(18) (Figure 6b). In vitro triggers (e.g., trypsin cleavage and urea incubation) induce the \\nprefusion SARS-CoV spike to transition to its postfusion state, in which S1 dissociates and \\nS2 forms a dumbbell-shaped structure with a rod-like structure in the middle and a globular \\nstructure at each end. These trimeric S2 molecules further associate at one end to form \\nrosettes in solution. Comparison with influenza virus HA suggests that the rod-like structure \\nin the middle likely represents the six-helix bundle formed by HR-N and HR-C, whereas the \\nglobular structures at the two ends likely correspond to the regions N-terminal to HR-N and \\nbetween HR-N and HR-C, respectively (Figures 1c and 6c). Three hydrophobic fusion \\npeptides, located N-terminal to HR-N, were previously buried in the prefusion spike but \\nbecome exposed in the postfusion S2 and associate to form rosettes in solution (they would \\ninsert into host membranes, were the membranes present). Crystal structures of the six-helix \\nbundle have been determined for a number of coronaviruses such as MHV , SARS-CoV , \\nLi Page 8\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptMERS-CoV , and HCoV-NL63 (113–119) (Figures 1d and 6c). Compared with influenza \\nvirus HA, the six-helix bundle formed by coronavirus HR-N and HR-C is unusually long \\n(18, 117), indicating the abundant amount of energy that can be released during the \\nconformational transition of S2 and available for use in membrane fusion. Overall, these \\nstudies reveal that coronavirus spikes fuse membranes using the same structural mechanism \\nas other class I membrane fusion proteins, but not for lack of some unique features such as \\ntheir large size, internal fusion peptide, double cleavage sites, and long six-helix bundle.\\nTRIGGERS FOR MEMBRANE FUSION BY CORONAVIRUS SPIKE PROTEINS\\nThe triggers for coronavirus spikes to undergo conformational transitions demonstrate a \\nmore complex pattern than many other class I membrane fusion proteins, probably a \\nreflection of their unique structural features discussed above. For example, although \\ninfluenza virus HA is primed by proteolysis during virus packaging, many coronavirus \\nspikes are not. Instead, coronavirus spikes are often subjected to proteolysis later in the cell \\nentry process, sometimes after receptor binding. Thus, proteolysis of coronavirus spikes can \\nlead directly to membrane fusion and thereby serves as an essential trigger for membrane \\nfusion (111, 112). The host proteases that cleave coronavirus spikes mainly come from four \\ndifferent stages of the virus infection cycle: (a) proprotein convertases (e.g., furin) during \\nvirus packaging in virus-producing cells, (b) extracellular proteases (e.g., elastase) after \\nvirus release into extracellular space, (c) cell surface proteases [e.g., type II transmembrane \\nserine protease (TMPRSS2)] after virus attachment to virus-targeting cells, and (d) \\nlysosomal proteases (e.g., cathepsin L and cathepsin B) after virus endocytosis in virus-\\ntargeting cells (120) (Figure 7). In addition to proteolysis, traditional triggers such as \\nreceptor binding and low pH may also play a role in membrane fusion. Here, I discuss \\ndetailed triggers for membrane fusion by the spikes from three representative coronaviruses.\\nAs the prototypic coronavirus, MHV has been extensively examined for its cell entry \\nmechanism. The findings are complicated and in some cases contradictory. First, MHV \\nspike is cleaved by proprotein convertases during virus packaging in the virus-producing \\ncells (121). This proteolysis is critical for MHV entry into virus-targeting cells. Second, the \\nbinding of CEACAM1 triggers the conformational transition of MHV spike and hence \\nmembrane fusion. This is supported by the observations that incubation of MHV spike with \\nrecombinant soluble CEACAM1 led to enhanced hydrophobicity of MHV S2 and the \\nappearance of a protease-resistant S2 fragment (122, 123). These observations indicated the \\nexposure of fusion peptides and formation of the six-helix bundle, respectively, in postfusion \\nS2. Third, there have been contradictory reports about whether MHV enters target cells at \\nthe cell membrane or through endocytosis and whether MHV spike undergoes \\nconformational transitions at low, neutral, or even elevated pH (123–125). This may depend \\non the MHV strains examined or experimental approaches used in the studies. The roles of \\npH and endocytosis in MHV entry still need to be further clarified. Interestingly, the \\nneurotropic strain MHV-JHM can mediate virus entry into host cells that do not express \\nCEACAM1 (126–128). This receptor-independent entry by MHV-JHM is unique among \\nviruses. Biochemical characterization of MHV-JHM spike suggested that it is more labile \\nthan the spikes from other MHV strains, meaning that it undergoes conformational \\ntransitions spontaneously in the absence of the receptor (94, 129, 130). It is believed that \\nLi Page 9\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptMHV-JHM is able to infect neural cells where CEACAM1 expression level is very low, at \\nleast in part because its spike can mediate receptor-independent entry (131, 132). Taken \\ntogether, the membrane fusion mechanism of MHV spike depends on both proteolysis and \\nreceptor binding, and it may or may not depend on the low pH of endosomes; in addition, \\nreceptor-independent membrane fusion by MHV-JHM spike contributes to the neutral \\ntropism of MHV-JHM.\\nResearch on the cell entry mechanism of SARS-CoV has led to novel findings. First, SARS-\\nCoV spike is not cleaved by proprotein convertases during virus packaging and hence \\nremains intact on mature virions (133, 134). Instead, SARS-CoV enters host cells through \\nendocytosis, and its spike is processed by lysosomal proteases (e.g., cathepsin L and \\ncathepsin B) (135–137). This is supported by the observation that inhibitors against either \\nendosomal acidification or lysosomal cysteine proteases block SARS-CoV entry. However, \\nlow pH itself is not a trigger for SARS-CoV entry',\n",
       "  '. Wong SK, Li WH, Moore MJ, Choe H, Farzan M. A 193-amino acid fragment of the SARS \\ncoronavirus S protein efficiently binds angiotensin-converting enzyme 2. J Biol Chem. 2004; \\n279:3197–201. [PubMed: 14670965] \\n47. Lin HX, Fen Y , Wong G, Wang LP, Li B, et al. Identification of residues in the receptor-binding \\ndomain (RBD) of the spike protein of human coronavirus NL63 that are critical for the RBD-\\nACE2 receptor interaction. J Gen Virol. 2008; 89:1015–24. [PubMed: 18343844] \\n48. Hofmann H, Simmons G, Rennekamp AJ, Chaipan C, Gramberg T, et al. Highly conserved regions \\nwithin the spike proteins of human coronaviruses 229E and NL63 determine recognition of their \\nrespective cellular receptors. J Virol. 2006; 80:8639–52. [PubMed: 16912312] \\n49. Godet M, Grosclaude J, Delmas B, Laude H. Major receptor-binding and neutralization \\ndeterminants are located within the same domain of the transmissible gastroenteritis virus \\n(coronavirus) spike protein. J Virol. 1994; 68:8008–16. [PubMed: 7525985] \\n50. Du L, Zhao G, Kou Z, Ma C, Sun S, et al. Identification of a receptor-binding domain in the S \\nprotein of the novel human coronavirus Middle East respiratory syndrome coronavirus as an \\nessential target for vaccine development. J Virol. 2013; 87:9939–42. [PubMed: 23824801] \\n51. Mou H, Raj VS, van Kuppeveld FJ, Rottier PJ, Haagmans BL, Bosch BJ. The receptor binding \\ndomain of the new MERS coronavirus maps to a 231-residue region in the spike protein that \\nefficiently elicits neutralizing antibodies. J Virol. 2013; doi: 10.1128/JVI.01277-13\\n52. Li F, Li WH, Farzan M, Harrison SC. Structure of SARS coronavirus spike receptor-binding \\ndomain complexed with receptor. Science. 2005; 309:1864–68. [PubMed: 16166518] \\n53. Li, F., Li, WH., Farzan, M., Harrison, SC. Interactions between SARS coronavirus and its receptor. \\nIn: Perlman, S., Holmes, KV ., editors. Nidoviruses: Toward Control of SARS and Other Nidovirus \\nDiseases. New York: Springer; 2006. p. 229-34.\\n54. Towler P, Staker B, Prasad SG, Menon S, Tang J, et al. ACE2 X-ray structures reveal a large hinge-\\nbending motion important for inhibitor binding and catalysis. J Biol Chem. 2004; 279:17996–\\n8007. [PubMed: 14754895] \\n55. Li WH, Zhang CS, Sui JH, Kuhn JH, Moore MJ, et al. Receptor and viral determinants of SARS-\\ncoronavirus adaptation to human ACE2. EMBO J. 2005; 24:1634–43. [PubMed: 15791205] \\n56. Guan Y , Zheng BJ, He YQ, Liu XL, Zhuang ZX, et al. Isolation and characterization of viruses \\nrelated to the SARS coronavirus from animals in southern China. Science. 2003; 302:276–78. \\n[PubMed: 12958366] \\n57. Li F. Structural analysis of major species barriers between humans and palm civets for severe acute \\nrespiratory syndrome coronavirus infections. J Virol. 2008; 82:6984–91. [PubMed: 18448527] \\n58. Wu KL, Peng GQ, Wilken M, Geraghty RJ, Li F. Mechanisms of host receptor adaptation by \\nsevere acute respiratory syndrome coronavirus. J Biol Chem. 2012; 287:8904–11. [PubMed: \\n22291007] \\n59. Wu K, Chen L, Peng G, Zhou W, Pennell CA, et al. A virus-binding hot spot on human \\nangiotensin-converting enzyme 2 is critical for binding of two different coronaviruses. J Virol. \\n2011; 85:5331–37. [PubMed: 21411533] \\n60. Song HD, Tu CC, Zhang GW, Wang SY , Zheng K, et al. Cross-host evolution of severe acute \\nrespiratory syndrome coronavirus in palm civet and human. PNAS. 2005; 102:2430–35. [PubMed: \\n15695582] \\n61. Qu XX, Hao P, Song XJ, Jiang SM, Liu YX, et al. Identification of two critical amino acid residues \\nof the severe acute respiratory syndrome coronavirus spike protein for its variation in zoonotic \\ntropism transition via a double substitution strategy. J Biol Chem. 2005; 280:29588–95. [PubMed: \\n15980414] \\nLi Page 15\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript62. Li WH, Greenough TC, Moore MJ, Vasilieva N, Somasundaran M, et al. Efficient replication of \\nsevere acute respiratory syndrome coronavirus in mouse cells is limited by murine angiotensin-\\nconverting enzyme 2. J Virol. 2004; 78:11429–33. [PubMed: 15452268] \\n63. Frieman M, Yount B, Agnihothram S, Page C, Donaldson E, et al. Molecular determinants of \\nsevere acute respiratory syndrome coronavirus pathogenesis and virulence in young and aged \\nmouse models of human disease. J Virol. 2012; 86:884–97. [PubMed: 22072787] \\n64. Ge XY , Li JL, Yang XL, Chmura AA, Zhu G, et al. Isolation and characterization of a bat SARS-\\nlike coronavirus that uses the ACE2 receptor. Nature. 2013; 503:535–38. [PubMed: 24172901] \\n65. Menachery VD, Yount BL Jr, Debbink K, Agnihothram S, Gralinski LE, et al. A SARS-like cluster \\nof circulating bat coronaviruses shows potential for human emergence. Nat Med. 2015; 21:1508–\\n13. [PubMed: 26552008] \\n66. Li WD, Shi ZL, Yu M, Ren WZ, Smith C, et al. Bats are natural reservoirs of SARS-like \\ncoronaviruses. Science. 2005; 310:676–79. [PubMed: 16195424] \\n67. Lau SKP, Woo PCY , Li KSM, Huang Y , Tsoi HW, et al. Severe acute respiratory syndrome \\ncoronavirus-like virus in Chinese horseshoe bats. PNAS. 2005; 102:14040–45. [PubMed: \\n16169905] \\n68. Hou YX, Peng C, Yu M, Li Y , Han ZG, et al. Angiotensin-converting enzyme 2 (ACE2) proteins of \\ndifferent bat species confer variable susceptibility to SARS-CoV entry. Arch Virol. 2010; \\n155:1563–69. [PubMed: 20567988] \\n69. Lu G, Hu Y , Wang Q, Qi J, Gao F, et al. Molecular basis of binding between novel human \\ncoronavirus MERS-CoV and its receptor CD26. Nature. 2013; 500:227–31. [PubMed: 23831647] \\n70. Wang N, Shi X, Jiang L, Zhang S, Wang D, et al. Structure of MERS-CoV spike receptor-binding \\ndomain complexed with human receptor DPP4. Cell Res. 2013; 23:986–93. [PubMed: 23835475] \\n71. Chen Y , Rajashankar KR, Yang Y , Agnihothram SS, Liu C, et al. Crystal structure of the \\nreceptorbinding domain from newly emerged Middle East respiratory syndrome coronavirus. J \\nVirol. 2013; 87:10777–83. [PubMed: 23903833] \\n72. Rasmussen HB, Branner S, Wiberg FC, Wagtmann N. Crystal structure of human dipeptidyl \\npeptidase IV/CD26 in complex with a substrate analog. Nat Struct Biol. 2003; 10:19–25. \\n[PubMed: 12483204] \\n73. Peck KM, Cockrell AS, Yount BL, Scobey T, Baric RS, Heise MT. Glycosylation of mouse DPP4 \\nplays a role in inhibiting Middle East respiratory syndrome coronavirus infection. J Virol. 2015; \\n89:4696–99. [PubMed: 25653445] \\n74. Cockrell AS, Peck KM, Yount BL, Agnihothram SS, Scobey T, et al. Mouse dipeptidyl peptidase 4 \\nis not a functional receptor for Middle East respiratory syndrome coronavirus infection. J Virol. \\n2014; 88:5195–99. [PubMed: 24574399] \\n75. Fukuma A, Tani H, Taniguchi S, Shimojima M, Saijo M, Fukushi S. Inability of rat DPP4 to allow \\nMERS-CoV infection revealed by using a VSV pseudotype bearing truncated MERS-CoV spike \\nprotein. Arch Virol. 2015; 160:2293–300. [PubMed: 26138557] \\n76. Barlan A, Zhao J, Sarkar MK, Li K, McCray PB Jr, et al. Receptor variation and susceptibility to \\nMiddle East respiratory syndrome coronavirus infection. J Virol. 2014; 88:4953–61. [PubMed: \\n24554656] \\n77. Haagmans BL, Al Dhahiry SH, Reusken CB, Raj VS, Galiano M, et al. Middle East respiratory \\nsyndrome coronavirus in dromedary camels: an outbreak investigation. Lancet Infect Dis. 2014; \\n14:140–45. [PubMed: 24355866] \\n78. Alagaili AN, Briese T, Mishra N, Kapoor V , Sameroff SC, et al. Middle East respiratory syndrome \\ncoronavirus infection in dromedary camels in Saudi Arabia. mBio. 2014; 5:e00884–14. [PubMed: \\n24570370] \\n79. Annan A, Baldwin HJ, Corman VM, Klose SM, Owusu M, et al. Human β-coronavirus 2c EMC/\\n2012-related viruses in bats, Ghana and Europe. Emerg Infect Dis. 2013; 19:456–59. [PubMed: \\n23622767] \\n80. Holmes KV , Dominguez SR. The new age of virus discovery: genomic analysis of a novel human \\nβ-coronavirus isolated from a fatal case of pneumonia. mBio. 2013; 4:e00548–12. [PubMed: \\n23300251] \\nLi Page 16\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript81. Lau SK, Li KS, Tsang AK, Lam CS, Ahmed S, et al. Genetic characterization of Betacoronavirus \\nlineage C viruses in bats reveals marked sequence divergence in the spike protein of Pipistrellus \\nbat coronavirus HKU5 in Japanese pipistrelle: implications for the origin of the novel Middle East \\nrespiratory syndrome coronavirus. J Virol. 2013; 87:8638–50. [PubMed: 23720729] \\n82. Wang Q, Qi J, Yuan Y , Xuan Y , Han P, et al. Bat origins of MERS-CoV supported by bat \\ncoronavirus HKU4 usage of human receptor CD26. Cell Host Microbe. 2014; 16:328–37. \\n[PubMed: 25211075] \\n83. Wu K, Li W, Peng G, Li F. Crystal structure of NL63 respiratory coronavirus receptor-binding \\ndomain complexed with its human receptor. PNAS. 2009; 106:19970–74. [PubMed: 19901337] \\n84. Reguera J, Santiago C, Mudgal G, Ordono D, Enjuanes L, Casasnovas JM. Structural bases of \\ncoronavirus attachment to host aminopeptidase N and its inhibition by neutralizing antibodies. \\nPLOS Pathog. 2012; 8:e1002859. [PubMed: 22876187] \\n85. Chen L, Lin YL, Peng G, Li F. Structural basis for multifunctional roles of mammalian aminopep-\\ntidase N. PNAS. 2012; 109:17966–71. [PubMed: 23071329] \\n86. Wong AH, Zhou D, Rini JM. The X-ray crystal structure of human aminopeptidase N reveals a \\nnovel dimer and the basis for peptide processing. J Biol Chem. 2012; 287:36804–13. [PubMed: \\n22932899] \\n87. Tusell SM, Schittone SA, Holmes KV . Mutational analysis of aminopeptidase N, a receptor for \\nseveral group 1 coronaviruses, identifies key determinants of viral host range. J Virol. 2007; \\n81:1261–73. [PubMed: 17093189] \\n88. Peng GQ, Sun DW, Rajashankar KR, Qian ZH, Holmes KV , Li F',\n",
       "  \"Thorax1984;39:475-476\\nRuptureofananeurysmofthesinusofValsalvaintothesuperiorvenacava\\nGCKAYE,SJEDMONSON,JLCAPLIN,DSTUNSTALL-PEDOE\\nFromtheDepartmentofCardiology,StBartholomew'sHospital,London\\nWepresentanunusualcaseinwhichruptureofasinusof\\nValsalvaduringasepticaemicillnessproducedafistula\\nbetweentheaortaandthesuperiorvenacava.\\nCasereport\\nA64yearoldman(caucasian),previouslyingoodhealth,\\npresentedinJuly1982witha24hourhistoryofrigors,\\nabdominalpain,andjointpains.Onexaminationhewas\\nfebrile(39.5°C),withabloodpressureof140/60mmHg.\\nAsoftapicalsystolicmurmurwasnoted.Thehaemoglobin\\nconcentrationwas13-1g/dlandthewhitecellcount8-9x\\n109/1(84%neutrophils).Amidstreamspecimenofurine\\ngrewEscherichiacoli,asdidoneofsixbloodculturebot-\\ntles.Adiagnosisofsepticaemiasecondarytoaurinary\\ntractinfectionwasmade.Hewastreatedfor10dayswith\\noralamoxycillin,500mgthricedaily,andmadearapid\\nrecovery.\\nInAugust1982hehadarecurrenceofrigors.Againhe\\nwasfebrile(395°C)andbilateralpleuralfrictionrubswere\\nnotedaswellasarightpleuraleffusion.Thehaemoglobin\\nconcentrationwas12-9g/dlandthewhitecellcount161x\\n109/l(92%neutrophils).Amidstreamspecimenofurine\\nwassterilebutbloodculturesgrewEcoli.Investigationof\\ntherenaltractshowedasmallfluidfilledcystintheright\\nkidneyandsomeprostaticenlargement.Itwasconsidered\\nthattherecurrentrenaltractinfectioncouldbeattributed\\ntothecystortotheprostatichypertrophy.Therenalcyst\\nwasconsideredtoosmalltoallowaspiration.Aftera\\ncourseofgentamicin,80mgthricedaily,atransurethral\\nprostatectomywasperformedandbenignprostatictissue\\nobtained.Thiswasfollowedbyfurtherrigorsandfever,\\nwhichrespondedtointramuscularcephradine,1gthrice\\ndaily.\\nThepatientwasreadmittedinDecember1982com-\\nplainingofincreasingdyspnoeaandpalpitations;hehad\\nclinicalsignsoffluidretention.Onexaminationhewas\\nafebrileandatrialfibrillationwasnotedforthefirsttime.\\nThepulsewasboundingandthebloodpressurewas140/\\n70mmHg.Thejugularvenouspressurewasraisedby5\\ncm.Aloud,longsystolicandanearlydiastolicmurmur\\nwereheardmaximallyatthethirdleftintercostalspace.\\nBilateralbasalcrepitationsandasmallrightpleural\\neffusionwerepresent.Theliverwasenlargedandnon-\\npulsatile.\\nThehaemoglobinconcentrationwas9-8g/dl.Achest\\nradiographshowedcardiomegaly;theelectrocardiogram\\nAddressforreprintrequests:DrGKaye,DepartmentofCardiol-\\nogy,StBartholomew'sHospital,LondonEClA7BE.\\nAccepted17January1984\\nFig1Angiogramintherightanteriorobliqueposition.Ao-aorta;f-fistula;SVC-superiorvenacava;Ra-right\\natrium.\\nwasunremarkableexceptforatrialfibrillation.An\\nechocardiogramshoweddiastolicflutteroftheanterior\\nmitralvalveleafletcompatiblewithaorticregurgitation\\nandleftventriculardilatation.Cardiaccatheterisation\\nshowedastepupintheoxygensaturationbetweenthe\\nhighsuperiorvenacavaandthelowsuperiorvenacava\\nfrom68%to89%,withapulmonaryarterysaturationof\\n78%.Anaortogramintherightanteriorobliqueposition\\nshowedaconnectionbetweentheaorticrootandthe\\nsuperiorvenacava(fig1).Trivialaorticregurgitationwas\\nseen.\\nAtoperation(fig2)therewasamoderatelylarge,clear\\npericardialeffusion.Withfullcardiopulmonarybypass,\\naftercrossclamping,theaortawasopened.Inrelationto\\nthenon-coronarysinustherewasadefect3cmindia-\\nmeter,theroofofwhichcommunicatedwiththemedial\\nwallofthesuperiorvenacavaviaasmoothwalledsinus.\\nTheaorticvalvewasnormal.Theholeinthesuperiorvena\\ncavawasclosedwithinterruptedsutures.Thedefectinthe\\naortawastoolargetobecloseddirectly,necessitating\\nreplacementabovethecoronaryostiawithaDacrontube\\ngraft.TheaorticwallgrewStreptococcusfaeciumandanti-\\n475\\nKaye,Edmonson,Caplin,Tunstall-Pedoe\\nROOTOFAORTAOPENED\\nFistulainSVC\\nSinus-\\nLeftcoronaryartery\\nRightcoronaryartery\\nFig2Diagramofdefectsfoundatoperation.\\nSVC-superiorvenacava.\\nbioticswerecontinuedintravenouslyforsixweeks.The\\npatientisnowfullyrecovered.\\nDiscussion\\nConventionallyaneurysmsofthesinusofValsalvahave\\nbeenthoughtofascongenitaldefectsintheaorticmedia.'\\nTheyaccountfor3-5%ofcasesofcongenitalheartdisease\\ntreatedsurgically2andareusuallyassociatedwithother\\ncongenitallesions,notablyventricularseptaldefects,34\\naorticcoarctation,andbicuspidaorticvalve.2Acquired\\naneurysmsofthesinusofValsalvaaremuchlesscommon,\\nthecausesincludingsyphilis,tuberculosis,andbacterial\\nendocarditis.'I\\nAneurysmsofthesinusofValsalvamorecommonly\\nariseintherightcoronarysinus.'3IThosearisinginthe\\nnon-coronaryandleftcoronarycuspsarelesscommonand\\nallmayruptureintotherightsideoftheheart.Without\\nsurgerysurvivalafterruptureisintheregionofoneyear,3\\nalthoughpatientssurviving15yearshavebeenreported.\\nCasesofruptureintotheleftatrium,leftventricle,inter-\\nventricularseptum,andpulmonaryinfundibulumhave\\nbeendescribed.6\\nRuptureintothesuperiorvenacavahasnotpreviously\\nbeendocumented.Thiscaseisofparticularinterestasthe\\naneurysmaroseinthenon-coronarycusp,whichtendsto\\nruptureintotherightheartchambers.'3Astheedgesof\\nthesinusweresowelldemarcated,acongenitaldefect\\nseemedlikely.Sfaeciumwasisolatedfromtheaorticwall,\\nhowever,suggestingthattheorganism,presumablyintro-\\nducedduringprostatectomy,mayhavedamagedthesinus\\nfurther,eitherproducingafistulaor,lesslikely,openingup\\nasmalldefectthatwasalreadypresent.Casesofbacterial\\nendocarditisproducingaorticocardiacfistulaswithout\\naneurysmformationhavebeennoted;2possiblytheinitial\\ndamagetotheaortawascausedbytheEcolisepticaemia,\\nSfaeciumcausingthesubsequentaneurysmtorupture.\\nReferences\\n'SakakibaraS,KonnoS.Congenitalaneurysmofthesinusof\\nValsalva:anatomyandclassification.AmHeartJ\\n1962;63:405-22.\\n2CondeCA,MellerJ,DonosoE,DackS.Bacterialendocarditis\\nwithrupturedsinusofValsalvaandaorticocardiacfistula.Am\\nJCardiol1975;35:912-6.\\nGerbodeF,OsborneJJ,JohnstonJB,KeithWJ.Ruptured\\naneurysmsoftheaorticsinus.AmJSurg1961;102:268-77.\\n4JonesAM,LangleyFA.Aorticsinusaneurysm.BrHeartJ\\n1949;11:325-41.\\n5OramS,EastT.Ruptureofaneurysmofaorticsinus(ofVal-\\nsalva)intorightsideoftheheart.BrHeartJ1955;17:541-51.\\n6DeSa-NetoA,PadnikMB,DesserKB,SteinhoffNG.RightsinusofValsalvaaneurysmcausingarightatrialfistulafollow-\\ningnon-penetratingchesttrauma.Circulation1979;60:205-9.\\n476\",\n",
       "  '. 1). The steady-state plasma PK of\\nlorlatinib in combination with avelumab was similar to\\nthat of previously reported exposure with lorlatinib\\nmonotherapy (Supplementary Table 5).10 Serum con-\\ncentrations of avelumab were similar across patients in\\neither treatment group. Exposures for crizotinib and its\\nmetabolite in combination with avelumab were similar\\nto those reported for crizotinib monotherapy\\n(Supplementary Tables 6 and 7).40,41 The variability of\\ncrizotinib CL/F was high because of the small sample\\nsize.\\nDiscussion\\nJAVELIN Lung 101 evaluated the safety, ef ﬁcacy,\\nand PK of avelumab in combination with lorlatinib or\\ncrizotinib for patients with ALK-negative or ALK-posi-\\ntive advanced NSCLC, respectively. Most patients had\\nreceived previous treatment for advanced or meta-\\nstatic disease. Avelumab plus lorlatinib had a\\nmanageable safety pro ﬁle, whereas avelumab plus\\ncrizotinib resulted in DLT inﬁve of 12 patients (42%).\\nThe MTD and RP2D for avelumab plus lorlatinib was\\ndetermined to be avelumab 10 mg/kg IV every 2\\nweeks plus lorlatinib 100 mg once daily. The MTD for\\navelumab plus crizotinib was exceeded at the doses\\nadministered (10 mg/kg IV every 2 wk and 250 mg\\ntwice daily, respectively), and this combination did\\nnot proceed to phase 2. Enrollment was stopped\\nearly based on the incidence of hepatic toxicity in the\\navelumab plus crizotinib group and the changing\\ntreatment landscape for patients with metastatic\\nALK-positive NSCLC. Co-administration of avelumab\\nwith lorlatinib or crizotinib had no clinically meaningful\\nTable 4.Summary of Responses per Investigator Assessment\\nResponse\\nAvelumab þ\\nLorlatinib\\n(n ¼ 31)\\nAvelumab þ\\nCrizotinib\\n(n ¼ 12)\\nConﬁrmed best overall\\nresponse, n (%)\\nCR 1 (3) 0\\nPR 15 (48) 3 (25)\\nSD 6 (19) 4 (33)\\nPD 7 (23) 5 (42)\\nNE 2 (7) 0\\nORR (95% CI), % 52 (33–70) 25 (6 –57)\\nMedian TTR\\n(range), mo\\n1.8 (1.3–3.7) 1.4 (1.4 –6.9)\\nMedian DOR\\n(95% CI), mo\\n14.7 (3.7–NE) 3.7 (3.7 –4.6)\\nDCR (95% CI), % 71 (52–86) 58 (28 –85)\\nCI, conﬁdence interval; CR, complete response; DCR, disease control rate;\\nDOR, duration of response; NE, not evaluable; ORR, objective response rate;\\nPD, progressive disease; PR, partial response; SD, stable disease; TTR, time\\nto response.\\nFigure 1.Best percentage change from baseline in sum of diameters for target lesions per investigator assessment in the (A)\\navelumab plus lorlatinib and (B) avelumab plus crizotinib groups. CR, complete response; PD, progressive disease; PD-L1,\\nprogrammed death-ligand 1; PR, partial response; SD, stable disease.\\nJuly 2024 Avelumab + Lorlatinib or Crizotinib in NSCLC 7effect on lorlatinib or crizotinib PK based on compari-\\nsons with historical monotherapy data.3,9\\nAntitumor activity was observed with avelumab plus\\nlorlatinib in patients with ALK-positive tumors and\\navelumab plus crizotinib in patients withALK-negative\\ntumors. The ORR for avelumab plus lorlatinib was\\nsimilar to the ORR reported in a prior study of lorlatinib\\nmonotherapy in patients with advanced, ALK-positive\\nNSCLC who had previously received two or more TKIs\\n(52% versus 46%, respectively).\\n10 Median PFS with\\navelumab plus lorlatinib was comparable with results\\nfrom a phase 2 study of lorlatinib monotherapy in sub-\\ngroups of patients with ALK-positive NSCLC who had\\nprior ALK TKI treatment (6.4 versus 6.9 –7.3 mo,\\nrespectively).\\n11 The ORR for avelumab plus crizotinib in\\nour study (25%) was similar to the ORR reported with\\navelumab monotherapy in a cohort of patients with un-\\ntreated advanced NSCLC without ALK or EGFR alter-\\nations who were unselected based on PD-L1 status\\n(20%) and higher than the ORR reported with avelumab\\nmonotherapy in a cohort of platinum-treated patients\\n(12%).\\n14,42 No meaningful difference was observed in\\nresponse to treatment based on PD-L1 status. This study\\nwas limited by the small sample size, andﬁrm conclu-\\nsions cannot be drawn.\\nIn all previous clinical studies of ICI plus ALK TKI\\ncombination therapy in patients with ALK-positive\\nNSCLC, increased toxicity has been reported compared\\nwith single-agent treatment. 27–30 The reason for the\\nhigher toxicity in the avelumab plus crizotinib group\\nversus the avelumab plus lorlatinib is unknown but\\ncould be due to the lower kinase selectivity of crizotinib\\nversus lorlatinib, which may result in additional effects\\nwhen crizotinib is administered in combination with an\\nICI. The high toxicity with avelumab plus crizotinib in\\nthis study is consistent with previous reports of ICIs\\ncombined with crizotinib. In the CheckMate 370 trial,\\n38% of patients treated with nivolumab plus crizotinib\\nhad grade 3 or higher hepatotoxicity, and the study was\\nterminated early.\\n27 Similarly, with pembrolizumab plus\\ncrizotinib, the incidence of grade 3 or higher ALT or AST\\nwas high, and the study was terminated early. 30\\nAlthough combination therapy with nivolumab plus\\nceritinib or atezolizumab plus alectinib was found to\\nhave preliminary antitumor activity, increased toxicity\\ncompared with single agents was also observed.\\n28,29\\nConclusion\\nAvelumab plus lorlatinib combination treatment was\\nfeasible in patients withALK-positive advanced NSCLC,\\nbut antitumor activity did not seem to be substantially\\nincreased compared with previous studies of lorlatinib\\nmonotherapy. Avelumab plus crizotinib combination\\ntreatment was associated with toxicity issues at the\\ndoses evaluated. Results from this study and changes in\\nthe treatment landscape in advanced NSCLC do not\\nsupport further studies of these combinations.\\nData Sharing Statement\\nAny requests for data by qualiﬁed scientiﬁc and med-\\nical researchers for legitimate research purposes will be\\nsubject to the healthcare business of Merck KGaA, Darm-\\nstadt, Germany ’s (CrossRef Funder ID: 10.13039/\\n100009945) Data Sharing Policy. All requests should be\\nsubmitted in writing to the healthcare business of Merck\\nKGaA, Darmstadt, Germany’s data sharing portal (https://\\nwww.emdgroup.com/e n/research/our-approach-to-re\\nsearch-and-development/healthcare/clinical-trials/\\ncommitment-responsible-data-sharing.html ). When\\nthe healthcare business of Merck KGaA, Darmstadt, Ger-\\nmany, has a co-research, co-development, or co-marketing\\nor co-promotion agreement, or when the product has been\\noutlicensed, the responsibility for disclosure might be\\ndependent on the agreement between parties. Under\\nthese circumstances, the healthcare business of Merck\\nKGaA, Darmstadt, Germany, will endeavor to gain agree-\\nment to share data in response to requests.\\nFigure 2.Kaplan-Meier analysis of OS and PFS in the (A) avelumab plus lorlatinib and (B) avelumab plus crizotinib groups. OS,\\noverall survival; NR, not reached; PFS, progression-free survival.\\n8 Solomon et al JTO Clinical and Research Reports Vol. 5 No. 7CRediT Authorship Contribution\\nStatement\\nBenjamin J. Solomon: Investigation, Writing— re-\\nview and editing.\\nIbiayi Dagogo-Jack: Investigation, Writing— review\\nand editing.\\nSe-Hoon Lee: Investigation, Writing — review and\\nediting.\\nMichael J. Boyer:Investigation, Writing— review and\\nediting.\\nSuresh S. Ramalingam: Investigation, Writing— re-\\nview and editing.\\nEnric Carcereny:Investigation, Writing— review and\\nediting.\\nEnriqueta Felip:Investigation, Writing— review and\\nediting.\\nJi-Youn Han: Investigation, Writing — review and\\nediting.\\nToyoaki Hida: Investigation, Writing— review and\\nediting.\\nBrett G.M. Hughes: Investigation, Writing— review\\nand editing.\\nSang-We Kim: Investigation, Writing— review and\\nediting.\\nMakoto Nishio: Investigation, Writing— review and\\nediting.\\nTakashi Seto: Investigation, Writing— review and\\nediting.\\nTatsuro Okamoto: Investigation, Writing — review\\nand editing.\\nXiaoxi Zhang: Data curation, Formal analysis, Visu-\\nalization, Writing— review and editing,.\\nJean-Francois Martini: Data curation, Formal anal-\\nysis, Visualization, Writing— review and editing,.\\nErjian Wang: Data curation, Formal analysis, Visu-\\nalization, Writing— review and editing.\\nSteven De Beukelaer: Conceptualization, Data\\ncuration, Supervision, Writing— review and editing.\\nTodd M. Bauers:Investigation, Writing— review and\\nediting.\\nDisclosure\\nDr. Solomon reports providing a consulting or advi-\\nsory role for Amgen, AstraZeneca, BeiGene, Bristol Myers\\nSquibb, Cancer Council of Victoria, D3 Bio, Janssen, Lilly,\\nthe healthcare business of Merck KGaA, Darmstadt,\\nGermany, Pﬁzer, Roche/Genentech, Takeda, and Thoracic\\nOncology Group of Australasia; has provided speaker\\nservices for Amgen, AstraZeneca, P ﬁzer, and Roche/\\nGenentech; and has received institutional research\\nfunding from BeiGene, Bristol Myers Squibb, Lilly,\\nNovartis, Nuvalent, Pﬁzer, Roche/Genentech, and Sanoﬁ.\\nProf. Dagogo-Jack reports providing a consulting or\\nadvisory role for AstraZeneca, Bayer, Boehringer Ingel-\\nheim, BostonGene, Catalyst Pharmaceuticals, Genentech,\\nJanssen, Novocure, Pﬁzer, Sanoﬁ, Syros Pharmaceuticals,\\nand Xcovery; has received travel, accommodations, or\\nexpenses from Array BioPharma, Creative Educational\\nConcepts, DAVA Oncology, Medscape, OncLive/MJH Life\\nSciences, P ﬁzer, The ASCO Post, and Total Health\\nConferencing; and has received institutional research\\nfunding from Array BioPharma, Calithera Biosciences,\\nGenentech, Pﬁzer, Novartis, and Vivace Therapeutics.\\nProf. Lee reports providing a consulting or advisory role\\nfor AstraZeneca/MedImmune, Bristol Myers Squibb, and\\nNovartis; has received honoraria from AstraZeneca/\\nMedImmune, the healthcare business of Merck KGaA,\\nDarmstadt, Germany, and Roche; and has received travel,\\naccommodations, or expenses from Novartis. Prof. Boyer\\nreports providing a consulting or advisory role for\\nAstraZeneca/MedImmune, Bristol Myers Squibb, and\\nMerck & Co., Kenilworth, NJ; has received institutional\\nresearch funding from Amgen, Ascentage Pharma,\\nAstraZeneca, BeiGene, Boehringer Ingelheim, Bristol\\nMyers Squibb, Lilly, Merck & Co., Kenilworth, NJ,\\nOncoMed, Peregrine Pharmaceuticals, Pﬁzer, and Roche/\\nGenentech; and has received travel, accommodations, or\\nexpenses from Boehringer Ingelheim, Bristol Myers\\nSquibb, Merck & Co., Kenilworth, NJ, and Roche/Gen-\\nentech. Prof. Ramalingam reports providing a consulting\\nor advisory role for AbbVie, Amgen, AstraZeneca, Boeh-\\nringer Ingelheim, Bristol Myers Squibb, Celgene, Lilly/\\nImClone, Roche/Genentech, Takeda, and the healthcare\\nbusiness of Merck KGaA, Darmstadt, Germany. Dr. Car-\\ncereny has no relationships to disclose. Prof. Felip re-\\nports providing a consulting or advisory role for AbbVie,\\nAmgen, AstraZeneca, Bayer, Blueprint Medicines, Boeh-\\nringer Ingelheim, Bristol Myers Squibb, Lilly, Glax-\\noSmithKline, Janssen, Merck & Co., Kenilworth, NJ,\\nNovartis, P ﬁzer, Puma Biotechnology, Roche, Sano ﬁ,\\nGenzyme, Takeda, and the healthcare business of Merck\\nKGaA, Darmstadt, Germany; has provided speaker ser-\\nvices for AstraZeneca, Boehringer Ingelheim, Bristol\\nMyers Squibb, Lilly, Medscape, Merck & Co., Kenilworth,\\nNJ, Novartis, PeerVoice, Pﬁzer, Prime Oncology, Roche,\\nSpringer, Takeda, touchIME, and CME Out ﬁtters; has\\nreceived research funding from grant for Oncology\\nInnovation, Fundación Merck Salud, Madrid, Spain, an\\nafﬁliate of Merck KGaA, Darmstadt, Germany; and is an\\nindependent member of the board of Grifol. Dr. Han re-\\nports providing a consulting or advisory role for Merck &\\nCo., Kenilworth, NJ; has received honoraria from\\nNovartis; and has received institutional research funding\\nfrom Roche. Dr',\n",
       "  '.\\nBased on these genomes, we developed a real-time \\nPCR assay and tested the original clinical samples from \\nthe BGI (WH01, \\nWH02, WH03, and WH04) again to \\ndetermine their threshold cycle (Ct) values (table). The \\nremaining samples were tested by a different real-time \\nPCR assay developed by the Chinese CDC, with Ct values \\nranging from 22·85 to 32·41 (table). These results \\nconfirmed the presence of 2019-nCoV in the patients.\\nBronchoalveolar lavage fluid samples or cultured \\nviruses of nine patients were used for next-generation \\nsequencing. After removing host (human) reads, de novo \\nassembly was done and the contigs obtained used as \\nqueries to search the non-redundant protein database. \\nSome contigs identified in all the samples were \\nclosely related to the bat SARS-like betacoronavirus \\nbat-SL-CoVZC45 betacoronavirus.\\n23 Bat-SL-CoVZC45 \\nwas then used as the reference genome and reads from \\neach pool were mapped to it, generating consensus \\nsequences corresponding to all the pools. These \\nconsensus sequences were then used as new reference \\ngenomes. Eight complete genomes and two partial \\ngenomes (from samples WH19002 and WH02; table) \\nwere obtained. The de novo assembly of the clean reads \\nfrom all the pools did not identify any other long contigs \\nthat corresponded to other viruses at high abundance.\\nThe eight complete genomes were nearly identical \\nacross the whole genome, with sequence identity above \\n99·98%, indicative of a very recent emergence into the \\nhuman population (figure 1A). The largest nucleotide \\ndifference was four mutations. Notably, the sequence \\nidentity between the two virus genomes from the same \\npatient (WH19001, from bronchoalveolar lavage fluid, \\nand WH19005, from cell culture) was more than 99·99%, \\nSee Online for appendix\\nFor Genbank see https://www.\\nncbi.nlm.nih.gov/genbank\\nFor the China National \\nMicrobiological Data Center \\nwebsite see http://nmdc.cn/\\nFor the data from BGI on the \\nChina National GeneBank see \\nhttps://db.cngb.org/datamart/\\ndisease/DATAdis19/Articles\\nwww.thelancet.com   Vol 395   February 22, 2020 569\\nFigure 1: Sequence comparison and genomic organisation of 2019-nCoV\\n(A) Sequence alignment of eight full-length genomes of 2019-nCoV, 29 829 base pairs in length, with a few nucleotides truncated at both ends \\nof the genome. \\n(B) Coding regions of 2019-nCoV, bat-SL-CoVZC45, bat-SL-CoVZXC21, SARS-CoV, and MERS-CoV. Only open reading frames of more than 100 nucleotides are \\nshown. 2019-nCoV=2019 novel coronavirus. SARS-CoV=severe acute respiratory syndrome coronavirus. MERS-CoV=Middle East respiratory syndrome coronavirus.\\nE M 7 10bN\\nE M 6 8b 9a\\nMG772933\\nBat_SL-CoVZC45\\n29802 base pairs\\nMG772934\\nBat_SLCoVZXC21\\n29 732 base pairs\\nNC_004718.3\\nSARS-CoV\\n29 751 base pairs 3a 7a 9b\\n8a3b\\n2019-nCoV\\nWH04\\n29844 base pairs\\nNon-structural polyprotein 1ab (7073 amino acids)\\n265 21 486\\nSpike protein (1255 amino acids)\\n21 493 25  260+C, 13 398\\nTTTTTAAACCGGG\\nConsensus\\nWH04\\nWH03\\nWH01\\nYH8011\\nSample\\nWH19001\\nWH19004\\nWH19005\\nWH19008\\nA\\nB\\nNC_019843.3\\nMERS-CoV\\n30 119 base pairs 3\\n54b E\\nM\\nN\\n8b4a\\nNon-structural polyprotein 1ab (7078 amino acids)\\n279 21 514\\nSpike protein (1270 amino acids)\\n21 7052 5 517+C, 13433\\nTTTTTAAACCGGG\\n Non-synonymous mutations\\n Synonymous mutations\\n5’\\n5’\\n5’\\n5’\\n5’\\n5’\\n5’\\n5’\\n5’\\n5’ 3’\\n5’ 3’\\nTTTTTAAACCGGG\\n+C, 13  443\\nCC TG GA C T C\\nA\\nT\\nAA\\nG\\nT\\nC\\nT\\nA\\n3’\\n3’\\n3’\\n3’\\n3’\\n3’\\n3’\\n3’\\n3’\\n1 5000 10 000 15 000 20 000 25 0002 9 829\\n7b\\n3a 8\\nNon-structural polyprotein 1ab (7070 amino acids)\\n265 21 477\\nSpike protein (1245 amino acids)\\n21 4842 5 221+C, 13 389\\nTTTTTAAACCGGG\\n5’ 3’\\n3a 8\\n3b\\nNon-structural polyprotein 1ab  (7092 amino acids)\\n265 21 543\\nSpike protein (1246 amino acids)\\n21 5502 5 290+C, 13455\\nTTTTTAAACCGGG\\n5’ 3’\\n9 13 14\\n3b 91 31 4\\n3\\nE M 7\\n8\\n10b\\nNon-structural polyprotein 1ab (7096 amino acids)\\n251 21 541\\nSpike protein (1273 amino acids)\\n21 5492 5 730+C, 13 443\\nTTTTTAAACCGGG\\n5’ 3’\\nN\\nE M 7 10bN\\n9 13 14\\n6943 8757 11 739 20 64620 655 24 301 27 46928 12028 229\\nBase pairsArticles\\n570 www.thelancet.com   Vol 395   February 22, 2020\\nwith 100% identity at the amino acid level. In addition, \\nthe partial genomes from samples WH02 and WH19002 \\nalso had nearly 100% identity to the complete genomes \\nacross the aligned gene regions.\\nA Blastn search of the complete genomes of 2019-nCoV \\nrevealed that the most closely related viruses available \\non GenBank were bat-SL-CoVZC45 (sequence identity \\n87·99%; query coverage 99%) and another SARS-like \\nbetacoronavirus of bat origin, bat-SL-CoVZXC21 (acces\\n-\\nsion number MG772934;23 87·23%; query coverage 98%). \\nIn five gene regions (E, M, 7, N, and 14), the sequence \\nidentities were greater than 90%, with the highest \\nbeing 98·7% in the E gene (fi gure 2A). The S gene of \\n2019-nCoV exhibited the lowest sequence identity with \\nbat-SL-CoVZC45 and bat-SL-CoVZXC21, at only around \\n75%. In addition, the sequence identity in 1b (about 86%) \\nwas lower than that in 1a (about 90%; figure 2A). Most of \\nthe encoded proteins exhibited high sequence identity \\nbetween 2019-nCoV and the related bat-derived corona\\n-\\nviruses (figure 2a). The notable exception was the spike \\nprotein, with only around 80% sequence identity\\n, and \\nprotein 13, with 73·2% sequence identity. Notably, the \\n2019-nCoV strains were less genetically similar to \\nSARS-CoV (about 79%) and MERS-CoV (about 50%). The \\nsimilarity between 2019-nCoV and related viruses was \\nvisualised using SimPlot software, with the 2019-nCoV \\nconsensus sequence employed as the query (figure 2B).\\nComparison of the predicted coding regions of \\n2019-nCoV showed that they possessed a similar genomic \\norganisation to bat-SL-CoVZC45, bat-SL-CoVZXC21, and \\nSARS-CoV (figure 1B). At least 12 coding regions were \\npredicted, including 1ab, S, 3, E, M, 7, 8, 9, 10b, N, 13, and \\n14 (figure 1B). The lengths of most of the proteins encoded \\nby 2019-nCoV, bat-SL-CoVZC45, and bat-SL-CoVZXC21 \\nwere similar, with only a few minor insertions or dele\\n-\\ntions. A notable difference was a longer spike protein \\nencoded by 2019-nC\\noV compared with the bat SARS-like \\ncoronaviruses, SARS-CoV, and MERS-CoV (figure 1B).\\nPhylogenetic analysis of 2019-nCoV and its closely \\nrelated reference genomes, as well as representative beta-\\ncoronaviruses, revealed that the five subgenera formed \\nfive well supported branches (\\nfigure 3). The subgenus \\nFigure 2: Sequence identity between the consensus of 2019-nCoV and representative betacoronavirus genomes\\n(A) Sequence identities for 2019-nCoV compared with SARS-CoV GZ02 (accession number AY390556) and the bat SARS-like coronaviruses bat-SL-CoVZC45 \\n(MG772933) and bat-SL-CoVZXC21 (MG772934). (B) Similarity between 2019-nCoV and related viruses. 2019-nCoV=2019 novel coronavirus. SARS-CoV=severe \\nacute respiratory syndrome coronavirus.\\n1abS 3\\nE\\nM\\n7\\n8\\n10b\\nN\\n13 14\\nStrain Complete\\ngenome\\n(%)\\nGene region (%)\\nNucleotide\\nsequences\\nAmino acid \\nsequences\\nBat-SL-CoVZC45\\nBat-SL-CoVZXC21\\nSARS-CoVGZ02\\n87·6\\n87·5\\n79·0\\n88·9\\n88·7\\n79·5\\n90·7\\n90·3\\n75·4\\n86·0\\n86·1\\n86·3\\n75·2\\n74·7\\n72·7\\n87·8\\n88·9\\n75·6\\n98·7\\n98·7\\n93·5\\n93·4\\n93·4\\n85·1\\n95·2\\n95·2\\n74·5\\n88·8\\n89·1\\n82·1\\n88·5\\n88·5\\n··\\n91·1\\n91·2\\n88·1\\n89·1\\n89·5\\n··\\n96·7\\n96·7\\n··\\n95·6\\n95·2\\n86·2\\n95·6\\n95·1\\n80·5\\n95·8\\n95·5\\n95·6\\n80·2\\n79·6\\n76·2\\n90·9\\n92·0\\n73·1\\n100·0\\n100·0\\n94·7\\n98·6\\n98·6\\n90·1\\n93·4\\n93·4\\n68·9\\n87·6\\n88·4\\n85·2\\n94·2\\n94·2\\n··\\n94·3\\n94·3\\n90·3\\n73·2\\n73·2\\n ··\\n92·9\\n92·9\\n··\\n··\\n··\\n··\\nBat-SL-CoVZC45\\nBat-SL-CoVZXC21\\nSARS-CoVGZ02\\n1a 1b\\n1ab 1a 1b S3 EM 78 10b N1 31 4\\nA\\nB\\n1000 3000 5000 7000 9000 11 000 13 000 15 000 17 000 19 000 21 000 23 000 27 00025 000 29 000\\n0\\n0·05\\n0·10\\n0·15\\n0·20\\n0·25\\n0·30\\n0·35\\n0·40\\n0·45\\n0·50\\n0·55\\n0·60\\n0·65\\n0·70\\n0·75\\n0·80\\n0·85\\n0·90\\n0·95\\n1·00\\nSimilarity (%)\\nPosition\\nSARS-CoV GZ02\\nBat-SL-CoVZC45\\nBat-SL-CoVZXC21\\nGene\\n2019-nCoV\\nConsensusArticles\\nwww.thelancet.com   Vol 395   February 22, 2020 571\\nSarbecovirus could be classified into three well supported \\nclades: two SARS-CoV-related strains from Rhinolophus sp \\nfrom Bulgaria (accession number GU190215) and Kenya \\n(KY352407) formed clade 1; the ten 2019-nCoV from \\nWuhan and the two bat-derived SARS-like strains from \\nZhoushan in eastern China (bat-SL-CoVZC45 and \\nbat-SL-CoVZXC21) formed clade 2, which was notable for \\nthe long branch separating the human and bat viruses; \\nand SARS-CoV strains from humans and many genetically \\nsimilar SARS-like corona\\n viruses from bats collected \\nfrom southwestern China formed clade 3, with bat-derived\\n \\ncoronaviruses also falling in the basal positions (figure 3). \\nIn addition, 2019-nCoV was distinct from SARS-CoV  \\nin a phylogeny of the complete RNA-dependent RNA \\npolymerase (RdRp) gene (appendix p 2). This evidence \\nindicates that 2019-nCoV is a novel beta\\n coronavirus \\nfrom the subgenus S\\narbecovirus.\\nAs the sequence similarity plot revealed changes in \\ngenetic distances among viruses across the 2019-nCoV  \\ngenome, we did additional phylogenetic analyses of \\nthe major encoding regions of representative members \\nof the subgenus Sarbecovirus. Consistent with the \\ngenome phylogeny, 2019-nCoV, bat-SL-CoVZC45, and \\nbat-SL-CoVZXC21 clustered together in trees of \\nthe 1a and spike genes (appendix p 3). By contrast, \\n2019-nCoV did not cluster with bat-SL-CoVZC45 and \\nbat-SL-CoVZXC21 in the 1b tree, but instead formed a \\ndistinct clade with SARS-CoV, bat-SL-CoVZC45, and \\nbat-SL-CoVZXC21 (appendix p 3), indicative of potential \\nrecombination events in 1b, although these probably \\noccurred in the bat coronaviruses rather than 2019-nCoV. \\nPhylogenetic analysis of the 2019-nCoV genome ex\\n-\\ncluding 1b revealed similar evolutionary relationships as \\nthe full-length viral genome (appendix p 3).\\nThe envelope spike (S) protein mediates receptor \\nbinding and membrane fusion\\n24 and is crucial for \\ndetermining host tropism and transmission capacity',\n",
       "  '.\\nLi Page 22\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 2. \\nCryo-electron microscopy structures of prefusion trimeric coronavirus spikes. (a) Trimeric \\nmouse hepatitis coronavirus (MHV) spike (PDB ID: 3JCL) (16). Three monomers are shown \\n(magenta, cyan, and green). (b) One monomer from the trimeric MHV spike. The important \\nfunctional elements of the spike [the S1 N-terminal domain (S1-NTD), the S1 C-terminal \\ndomain (S1-CTD), the fusion peptide (FP), and the heptad repeat (HR-N)] are colored in the \\nsame way as in Figure 1c. The dotted curve indicates a disordered loop. Scissors indicate \\ntwo critical proteolysis sites.\\nLi Page 23\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 3. \\nCrystal structures of betacoronavirus S1 C-terminal domains (S1-CTDs). (a) Structure of \\nsevere acute respiratory syndrome coronavirus (SARS-CoV) S1-CTD complexed with \\nhuman ACE2 (PDB ID: 2AJF) (52). Shown are the core structure of S1-CTD (cyan), the \\nreceptor-binding motif (red), and ACE2 (green). (b) Interface between human SARS-CoV \\nS1-CTD and human ACE2, showing two virus-binding hot spots on human ACE2. Dashed \\nlines indicate salt bridges. (c) Interface between palm civet SARS-CoV S1-CTD and human \\nACE2. Critical residue changes from human to civet SARS-CoV strains are labeled. (d) \\nInterface between human SARS-CoV S1-CTD and rat or mouse ACE2. Critical residue \\nchanges from human to rat or mouse ACE2 are labeled. (e) Structure of Middle East \\nrespiratory syndrome coronavirus (MERS-CoV) S1-CTD complexed with human DPP4 \\n(PDB ID: 4KR0) (69).\\nLi Page 24\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 4. \\nCrystal structures of alphacoronavirus S1 C-terminal domains (S1-CTDs). (a) Structure of \\nhuman coronavirus NL63 (HCoV-NL63) S1-CTD complexed with human ACE2 (PDB ID: \\n4KBH) (83). (b) Structure of porcine respiratory coronavirus (PRCV) S1-CTD complexed \\nwith porcine APN (PDB ID: 4F5C) (84). (c) Structural topology of alphacoronavirus S1-\\nCTDs. β-Strands are shown as arrows. (d) Structural topology of betacoronavirus S1-CTDs. \\nα-Helices are shown as cylinders. All of the secondary structural elements in panels c and d \\nare connected in the same order, even though strands β4, β1, and β2 in panel c become \\nhelices α4, α1, and a loop, respectively, in panel d. (e) Interface between HCoV-NL63 S1-\\nLi Page 25\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptCTD and human ACE2. Virus-binding motifs (VBMs) on ACE2 are shown in blue. \\nReceptor-binding motifs (RBMs) on S1-CTD are shown in red. (f) Interface between SARS-\\nCoV S1-CTD and human ACE2.\\nLi Page 26\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 5. \\nCrystal structures of betacoronavirus S1 N-terminal domains (S1-NTDs). (a) Structure of \\nmouse hepatitis coronavirus (MHV) S1-NTD complexed with murine CEACAM1 (PDB ID: \\n3R4D) (88). The core structure of MHV S1-NTD is shown in magenta and green, the \\nreceptor-binding motifs in red, and the rest in cyan. The N-terminal immunoglobulin domain \\nof CEACAM1 is shown in yellow and virus-binding motifs in blue. (b) Structure of bovine \\ncoronavirus (BCoV) S1-NTD (PDB ID: 4H14) (43). The asterisk indicates the binding site \\nfor sugar receptor Neu5,9Ac2. (c) Structure of human galectin-3 complexed with galactose \\n(PDB ID: 1A3K). Sugar receptor is shown in blue. (d) Structure of influenza virus HA1 \\n(PDB ID: 1JSO). Sugar receptor is shown in blue. (e–g) Structural topologies of (e) \\nbetacoronavirus S1-NTDs, (f) human galectins, and (g) influenza virus HA1.\\nLi Page 27\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 6. \\nStructural mechanism for membrane fusion by coronavirus spikes. (a) Structural mechanism \\nfor membrane fusion by class I viral membrane fusion proteins. Schematics of these proteins \\nin both prefusion and postfusion conformations are shown. (b) Negative-stain electron \\nmicroscopy images of SARS-CoV spike in both prefusion and postfusion conformations are \\nshown (18). (c) Schematics of SARS-CoV postfusion S2 in solution (left) and in vivo \\n(right). Abbreviations: FP, fusion peptide; HR-N, heptad repeat region N; HR-C, heptad \\nrepeat region C; IC, intracellular tail; SARS-CoV , severe acute respiratory syndrome \\ncoronavirus; TM, transmembrane anchor.\\nLi Page 28\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 7. \\nTriggers for coronavirus spikes to fuse membranes. Scissors indicate potential spike-\\nprocessing host proteases. Shown are virus particles (green spheres), virus surface spikes \\n(blue protrusions), viral genome (magenta coils), cells (large gray circles), endosome/\\nlysosome (small shaded gray circle), and receptor (light green base on cell surface). Spike-\\nprocessing host proteases are labeled for representative coronaviruses: Middle East \\nrespiratory syndrome coronavirus (MERS-CoV), mouse hepatitis coronavirus (MHV), and \\nsevere acute respiratory syndrome coronavirus (SARS-CoV).\\nLi Page 29\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 8. \\nEvolution of coronavirus spikes. (a) Structural comparison between human galectins and \\nalphacoronavirus HCoV-NL63 S1-CTD. Both the crystal structures and structural topologies \\nof the two proteins are shown. Common subcore structures in the two proteins are \\nhighlighted with gray shading. (b) Hypothesized evolution of coronavirus spike proteins. \\nAbbreviations: HCoV-NL63, human coronavirus NL63; IC, intracellular tail; S1-CTD, S1 \\nC-terminal domain; S1-NTD, S1 N-terminal domain; TM, transmembrane anchor.\\nLi Page 30\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript',\n",
       "  '.5 /C2 10\\n9/L, platelet\\ncount /C21 100 /C2 109/L, hemoglobin /C21 109 g/dL, esti-\\nmated creatinine clearance /C21 30 mL/min according to\\nthe Cockcroft-Gault formula, total bilirubin level /C20\\n1.5/C2 the upper limit of norm al [ULN], aspartate\\naminotransferase [AST] or alanine aminotransferase\\n[ALT] /C20 2.5 /C2 ULN, and serum amylase or lipase <\\n1.5 /C2 ULN); no prior immunotherapy; and available\\ntumor tissue (archived tissue or fresh biopsy).\\nIn the phase 1b portion, patients in the avelumab plus\\nlorlatinib group hadALK-positive tumors and any num-\\nber of prior regimens (including zero); prior treatment\\nwith lorlatinib was permitted. In the phase 2 portion, the\\navelumab plus lorlatinib group was limited to patients\\nwith no prior systemic treatment for advanced or met-\\nastatic disease (adjuvant or neoadjuvant treatment was\\npermitted if completed/C21 6 mo before study entry) and\\nno prior TKI treatment at any time. PD-L1 status was not\\nconsidered in eligibility criteria. Patients in the avelumab\\nplus crizotinib group were previously treated and had\\nALK-negative tumors with noROS1 translocations or c-\\nMET ampliﬁcation or exon 14 skipping mutations; pa-\\ntients with EGFR mutations were permitted if they had\\ndisease progression after prior EGFR inhibitor treatment.\\nExclusion criteria included major surgery within 4\\nweeks or radiation therapy within 2 weeks before study\\nentry (prior palliative radiotherapy was permitted if\\ncompleted /C21 48 h before patient registration); systemic\\ncytotoxic anticancer therapy within 2 weeks or TKI\\ntherapy within 5 half-lives before study entry; brain\\nmetastases with some exceptions (patients with brain\\nmetastases were eligible for the avelumab plus crizotinib\\ngroup if they had completed treatment and recovered\\nfrom acute effects of radiotherapy or surgery before\\nstudy entry, had discontinued corticosteroid treatment\\nfor /C21 4 wk, and were neurologically stable; patients with\\nbrain metastases were eligible for the avelumab plus\\nlorlatinib group if they had asymptomatic metastases\\nrequiring <10 mg/d prednisone or equivalent); persis-\\ntent toxicity greater than grade 1 related to previous\\ntherapy (excluding grade 2 alopecia); diagnosis of any\\nother malignancy within 3 years before study entry\\n(except for adequately treated basal or squamous cell\\nskin cancer, carcinoma in situ of the breast or cervix, or\\nlow-grade prostate cancer [Gleason 6 or below] without\\nany plans for treatment intervention); and prior use of\\nimmunosuppressive medication within 7 days before\\nrandomization (except for intranasal, inhaled, or topical\\nsteroids or local steroid injections; systemic corticoste-\\nroids at physiological doses; or steroids as premed-\\nication for hypersensitivity).\\nThis trial was conducted in accordance with the\\nethical principles of the Declaration of Helsinki and the\\nGood Clinical Practice guidelines deﬁned by the Inter-\\nnational Council for Harmonisation. All participating\\npatients provided written informed consent. The proto-\\ncol was approved by the institutional review board or\\nindependent ethics committee at each participating\\ncenter.\\nProcedures\\nAvelumab 10 mg/kg was administered every 2 weeks\\nby intravenous (IV) infusion in both treatment groups.\\nLorlatinib was administered orally as 25 mg tablets, with\\na starting dose of 100 mg once daily. Crizotinib was\\nadministered orally as 250 mg capsules, with a starting\\ndose of 250 mg twice daily. Patients were treated until\\ndisease progression, unacceptable toxicity, or other\\nprotocol-speciﬁed criteria for withdrawal occurred.\\nDiscontinuation of one drug due to treatment-related\\ntoxicity with continuation of treatment as monotherapy\\nwas permitted outside of the dose-limiting toxicity (DLT)\\nperiod. Follow-up periods were 30, 60, and 90 days after\\nthe last study treatment.\\nEnd Points and Assessments\\nThe primary objective in phase 1b was to determine\\nthe maximum tolerated dose (MTD) and recommended\\nphase 2 dose (RP2D) for avelumab in combination with\\neither lorlatinib or crizotinib. The primary objective in\\nphase 2 was to assess the objective response rate (ORR)\\nin all patients and the complete response (CR) rate in the\\navelumab plus lorlatinib group per RECIST 1.1. The\\nprimary end point in phase 1b was occurrence of DLT\\nwithin the ﬁrst 2 treatment cycles (14-d cycle). The\\nprimary end point in phase 2 was conﬁrmed objective\\nresponse per RECIST 1.1 by investigator assessment\\n(including CR in the avelumab plus lorlatinib group).\\nSecondary end points included safety, OS, PK, tumor\\nJuly 2024 Avelumab + Lorlatinib or Crizotinib in NSCLC 3tissue biomarkers, duration of response (DOR; assessed\\nfrom CR or partial response [PR] until progressive dis-\\nease [PD], death, or last tumor assessment), disease\\ncontrol rate (DCR), time to response (TTR), and PFS, all\\nassessed by investigator per RECIST 1.1.\\nThe MTD was the highest dose of avelumab plus\\nlorlatinib or crizotinib associated with DLT in fewer than\\n33% of patients within theﬁrst 2 cycles of treatment.\\nDLT was de ﬁned as the occurrence of any of the\\nfollowing adverse events (AEs) attributable to one or\\nmore drug during the primary DLT observation period:\\n(1) hematologic AEs comprising grade 4 neutropenia\\nlasting more than 7 days; febrile neutropenia; grade 3 or\\nhigher neutropenic infection; grade 3 or higher throm-\\nbocytopenia with bleeding; grade 4 thrombocytopenia;\\nand grade 4 anemia; (2) nonhematologic AEs comprising\\nany toxicity of grade 3 or higher except for the following:\\ntransient (/C20 6 h) grade 3 ﬂu-like symptoms or fever\\ncontrolled with medical management; transient (/C20 24 h)\\ngrade 3 fatigue, local reactions, or headache; grade 3\\nnausea or vomiting that resolved to grade 1 or lower\\nwithin 7 days with medical management; grade 3 diar-\\nrhea or skin toxicity that resolved to grade 1 or lower\\nwithin 7 days with medical management; grade 3 or\\nhigher amylase or lipase abnormality not associated with\\npancreatitis; tumor ﬂare phenomenon (local pain, irri-\\ntation, or rash localized at tumor site); or an abnormal\\nlaboratory value that is unlikely related to study treat-\\nment; and (3) inability to complete 75% of lorlatinib or\\ncrizotinib treatment or two infusions of avelumab due to\\ntreatment-related toxicities.\\nAEs and laboratory test abnormalities were graded\\naccording to the National Cancer Institute Common\\nTerminology Criteria for Adverse Events version 4.03\\nand coded using the Medical Dictionary for Regulatory\\nActivities. Infusion-re lated reaction (IRR) was\\nanalyzed as an AE of special interest using a com-\\nposite deﬁnition that included the following preferred\\nterms: infusion-related reaction, chills, pyrexia, and\\ndyspnea. Antitumor activity was assessed radiologi-\\ncally at screening, at week 8, and then every 8 weeks\\nuntil PD. Tumor responses were conﬁrmed at least 4\\nweeks after initial documentation of a CR or PR. PD-\\nL1 status was determined centrally using the Ven-\\ntana PD-L1 (SP263) immunohistochemistry assay\\n(Roche Diagnostics, In dianapolis, IN); PD-L1 –positive\\nstatus was deﬁned as a combined positive score (tu-\\nmor and immune cells) of at least 1%.\\nPK parameters of avelumab, lorlatinib, and crizoti-\\nnib were evaluated using blood samples. For PK ana-\\nlyses of avelumab, 3.5-mL samples were collected\\nbefore infusion and 1 hour after infusion on day 1 and\\nat any time on day 8 of cycles 1 and 2. Additional\\nsamples were collected on day 1 of cycles 3 to 5,\\nf o l l o w e db yd a y1e v e r y6c y c l e st h e r e a f t e r ,a n da tt h e\\nend of treatment. For PK analyses of lorlatinib and\\ncrizotinib, samples (4 mL and 3 mL, respectively) were\\nc o l l e c t e dp r e d o s ea n da t1 ,2 ,4 ,6 ,a n d8h o u r sp o s t d o s e\\non day 1 of cycle 2. Sparse blood samples were\\ncollected predose on day 1 of cycles 4 and 7. For all\\npatients enrolled in phase 2, predose blood samples\\nwere collected on day 1 of cycles 2, 4, and 7.\\nSafety and efﬁcacy were assessed in all patients who\\nreceived at least one dose of the study drug (safety\\nanalysis set and full analysis set). DLT was evaluated in\\nall patients in phase 1b who received at least one dose\\nof the study drug and either had DLT during theﬁrst 2\\ncycles of treatment or completed the observation period\\nfor theﬁrst 2 cycles; patients without DLT who received\\nl e s st h a n7 5 %o ft h ed o s e so fl o r l a t i n i bo rc r i z o t i n i bi n\\nthe ﬁrst 2 cycles or less than two avelumab infusions for\\nreasons other than study drug-related toxicity were\\nreplaced. Response according to PD-L1 status was\\nassessed in the biomarker analysis set, which included\\nall patients in the safety analysis set who had at least\\none baseline biomarker assessment and had received at\\nleast one dose of any study drug. Kaplan-Meier esti-\\nmates were used for time-to-event analyses, and the\\nconﬁdence interval (CI) for the median was calculated\\naccording to Brookmeyer and Crowley methodology.\\nPK parameters were assessed in the PK analysis set,\\nwhich included all patients in the safety analysis set who\\nhad sufﬁcient concentration data to estimate at least one\\nPK parameter of interest. Trough plasma concentration\\n(Ctrough) and maximum plasma concentration (Cmax) for\\navelumab, lorlatinib, and crizotinib were summarized\\ndescriptively (n, mean, standard deviation, percentage of\\ncoefﬁcient of variation [CV], median, minimum,\\nmaximum, geometric mean, CV, and 95% CI) by treat-\\nment group, cycle, and day.\\nOther PK parameters included time to maximum\\nplasma concentration, time of last measurable concen-\\ntration, area under the plasma concentration-time curve\\nduring the dosing interval time course, area under the\\nconcentration-time curve from time of dosing to the last\\ncollection time point, and apparent plasma clearance (CL/\\nF). For the metabolite of crizotinib, metabolite-to-parent\\nratio for area under the plasma concentration versus\\ntime curve during the dosing interval time course and the\\nmetabolite-to-parent ratio for C\\nmax were also determined.\\nDose-normalized parameters for lorlatinib and crizotinib\\n(including its metabolite) were also determined.\\nResults\\nPatient Characteristics and Disposition\\nBetween December 18, 2015, and March 6, 2018, 66\\npatients were screened and 43 were enrolled at 16\\n4 Solomon et al JTO Clinical and Research Reports Vol. 5 No. 7centers across the United States, Australia, France, Spain,\\nSouth Korea, and Japan. Patients received either avelu-\\nmab plus lorlatinib (n¼ 31; 28 in phase 1b and three in\\nphase 2) or avelumab plus crizotinib (n ¼ 12; all in\\nphase 1b). In both treatment groups, all patients\\nreceived starting doses (no other dose levels were\\nadministered). Most patients were Asian, had Eastern\\nCooperative Oncology Group performance status of 1 or\\n2, had PD-L1–positive tumors, had tumors with non-\\nsquamous histology, and had received at least two prior\\nanticancer drug regimens (Table 1). Most patients (27 of\\n31) in the avelumab plus lorlatinib group had received\\nprior therapy with an ALK TKI, and most patients (eight\\nof 12) in the avelumab plus crizotinib group had a his-\\ntory of smoking.\\nOn July 13, 2022 (data cutoff date), the study was\\nstopped, and ﬁve patients with ongoing avelumab plus\\nlorlatinib treatment were moved to a continued access\\nstudy (NCT05059522); no patient had ongoing avelu-\\nmab plus crizotinib treatment. The most common reason\\nfor permanent treatment discontinuation in both treat-\\nment groups was PD (Supplementary Table 1). Median\\nfollow-up for OS was 42 months in the avelumab plus\\nlorlatinib group and 35 months in the avelumab plus\\ncrizotinib group. Median duration of exposure in the\\navelumab plus lorlatinib group was 42 weeks (range, 2–\\n308 wk) for avelumab and 46 weeks (range, 2–308 wk)\\nfor lorlatinib; in the avelumab plus crizotinib group,\\nmedian duration of exposure was 15 weeks (range, 4–\\n126 wk) for avelumab and 6 weeks (range, 2–32 wk) for\\ncrizotinib',\n",
       "  '.Oncoimmunology. 2021;10:1951019.\\n25. Vladimer GI, Snijder B, Krall N, et al. Global survey of\\nthe immunomodulatory potential of common drugs.Nat\\nChem Biol. 2017;13:681–690\\n.\\n26. Zhou P , Shaffer DR, Alvarez Arias DA, et al. In vivo dis-\\ncovery of immunotherapy targets in the tumour micro-\\nenvironment. Nature. 2014;506:52–57.\\n27. Spigel DR, Reynolds C, Waterhouse D, et al. Phase 1/2\\nstudy of the safety and tolerability of nivolumab plus\\ncrizotinib for theﬁrst-line treatment of anaplastic lym-\\nphoma kinase translocation - positive advanced non-\\nsmall cell lung cancer (CheckMate 370). J Thorac\\nOncol. 2018;13:682–688.\\n28. Felip E, de Braud FG, Maur M, et al. Ceritinib plus\\nnivolumab in patients with advanced ALK-rearranged\\nnon-small cell lung cancer: results of an open-label,\\nmulticenter , phase 1b study. J Thorac Oncol .\\n2020;15:392–403.\\n29. Kim DW, Gadgeel S, Gettinger SN, et al. Brief report:\\nsafety and antitumor activity of alectinib plus atezoli-\\nzumab from a phase 1b study in advanced ALK-positive\\nNSCLC. JTO Clin Res Rep. 2022;3:100367.\\n30. Patel SP , Pakkala S, Pennell NA, et al. Phase Ib study of\\ncrizotinib plus pembrolizumab in patients with previ-\\nously untreated advanced non-small cell lung cancer\\nwith ALK translocation. Oncologist. 2020;25:e562 –\\ne1012.\\n31. Brahmer J, Reckamp KL, Baas P , et al. Nivolumab versus\\ndocetaxel in advanced squamous-cell non-small-cell lung\\ncancer. N Engl J Med. 2015;373:123–135.\\n32. Borghaei H, Paz-Ares L, Horn L, et al. Nivolumab versus\\ndocetaxel in advanced nonsquamous non-small-cell lung\\ncancer. N Engl J Med. 2015;373:1627–1639.\\n33. Solomon BJ, Mok T , Kim DW, et al. First-line crizotinib\\nversus chemotherapy in ALK-positive lung cancer.N Engl\\nJ Med. 2014;371:2167–2177.\\n34. Shaw AT , Kim TM, Crinò L, et al. Ceritinib versus chemo-\\ntherapy in patients with ALK-rearranged non-small-cell\\nlung cancer previously given chemotherapy and crizotinib\\n(ASCEND-5): a randomised, controlled, open-label, phase\\n3 trial.Lancet Oncol. 2017;18:874–886.\\n35. Rittmeyer A, Barlesi F , Waterkamp D, et al. Atezolizu-\\nmab versus docetaxel in patients with previously treated\\nnon-small-cell lung cancer (OAK): a phase 3, open-label,\\nmulticentre randomised controlled trial. Lancet.\\n2017;389:255–265.\\n36. Herbst RS, Giaccone G, de Marinis F , et al. Atezolizumab\\nfor ﬁ\\nrst-line treatment of PD-L1-selected patients with\\nNSCLC. N Engl J Med. 2020;383:1328–1339.\\n37. Peters S, Camidge DR, Shaw AT , et al. Alectinib versus\\ncrizotinib in untreated ALK-positive non-small-cell lung\\ncancer. N Engl J Med. 2017;377:829–838.\\n38. Reck M, Rodriguez-Abreu D, Robinson AG, et al. Pem-\\nbrolizumab versus chemotherapy for PD-L1-positive non-\\nsmall-cell lung cancer .N Engl J Med. 2016;375:1823–1833.\\n39. Dempke WCM, Fenchel K. Has programmed cell death\\nligand-1 MET an accomplice in non-small cell lung\\ncancer?-a narrative review. Transl Lung Cancer Res .\\n2021;10:2667–2682.\\n40. Johnson TR, Tan W, Goulet L, et al. Metabolism, excre-\\ntion and pharmacokinetics of [14C]crizotinib following\\noral administration to healthy subjects. Xenobiotica.\\n2015;45:45–59.\\n41. Xu H, O’Gorman M, Boutros T , et al. Evaluation of cri-\\nzotinib absolute bioavailability, the bioequivalence of\\nthree oral formulations, and the effect of food on cri-\\nzotinib pharmacokinetics in healthy subjects. J Clin\\nPharmacol. 2015;55:104–113.\\n42. Gulley JL, Rajan A, Spigel DR, et al. Avelumab for pa-\\ntients with previously treated metastatic or recurrent\\nnon-small-cell lung cancer (JAVELIN Solid T umor): dose-\\nexpansion cohort of a multicentre, open-label, phase\\n1b trial. Lancet Oncol. 2017;18:599–610.\\nJuly 2024 Avelumab + Lorlatinib or Crizotinib in NSCLC 11',\n",
       "  '. [PubMed: 19901337] \\n84. Reguera J, Santiago C, Mudgal G, Ordono D, Enjuanes L, Casasnovas JM. Structural bases of \\ncoronavirus attachment to host aminopeptidase N and its inhibition by neutralizing antibodies. \\nPLOS Pathog. 2012; 8:e1002859. [PubMed: 22876187] \\n85. Chen L, Lin YL, Peng G, Li F. Structural basis for multifunctional roles of mammalian aminopep-\\ntidase N. PNAS. 2012; 109:17966–71. [PubMed: 23071329] \\n86. Wong AH, Zhou D, Rini JM. The X-ray crystal structure of human aminopeptidase N reveals a \\nnovel dimer and the basis for peptide processing. J Biol Chem. 2012; 287:36804–13. [PubMed: \\n22932899] \\n87. Tusell SM, Schittone SA, Holmes KV . Mutational analysis of aminopeptidase N, a receptor for \\nseveral group 1 coronaviruses, identifies key determinants of viral host range. J Virol. 2007; \\n81:1261–73. [PubMed: 17093189] \\n88. Peng GQ, Sun DW, Rajashankar KR, Qian ZH, Holmes KV , Li F. Crystal structure of mouse \\ncoronavirus receptor-binding domain complexed with its murine receptor. PNAS. 2011; \\n108:10696–701. [PubMed: 21670291] \\n89. Beauchemin N, Draber P, Dveksler G, Gold P, Gray-Owen S, et al. Redefined nomenclature for \\nmembers of the carcinoembryonic antigen family. Exp Cell Res. 1999; 252:243–49. [PubMed: \\n11501563] \\n90. Tan KM, Zelus BD, Meijers R, Liu JH, Bergelson JM, et al. Crystal structure of murine \\nsCEACAM1a[1,4]: a coronavirus receptor in the CEA family. EMBOJ. 2002; 21:2076–86.\\n91. Wessner DR, Shick PC, Lu JH, Cardellichio CB, Gagneten SE, et al. Mutational analysis of the \\nvirus and monoclonal antibody binding sites in MHVR, the cellular receptor of the murine \\ncoronavirus mouse hepatitis virus strain A59. J Virol. 1998; 72:1941–48. [PubMed: 9499047] \\n92. Thackray LB, Turner BC, Holmes KV . Substitutions of conserved amino acids in the \\nreceptorbinding domain of the spike glycoprotein affect utilization of murine CEACAM1a by the \\nmurine coronavirus MHV-A59. Virology. 2005; 334:98–110. [PubMed: 15749126] \\n93. Leparc-Goffart I, Hingley ST, Chua MM, Jiang XH, Lavi E, Weiss SR. Altered pathogenesis of a \\nmutant of the murine coronavirus MHV-A59 is associated with a Q159L amino acid substitution in \\nthe spike protein. Virology. 1997; 239:1–10. [PubMed: 9426441] \\n94. Tsai JC, Zelus BD, Holmes KV , Weiss SR. The N-terminal domain of the murine coronavirus spike \\nglycoprotein determines the CEACAM1 receptor specificity of the virus strain. J Virol. 2003; \\n77:841–50. [PubMed: 12502800] \\n95. Ohtsuka N, Taguchi F. Mouse susceptibility to mouse hepatitis virus infection is linked to viral \\nreceptor genotype. J Virol. 1997; 71:8860–63. [PubMed: 9343248] \\n96. Ohtsuka N, Yamada YK, Taguchi F. Difference in virus-binding activity of two distinct receptor \\nproteins for mouse hepatitis virus. J Gen Virol. 1996; 77:1683–92. [PubMed: 8760415] \\n97. Hirai A, Ohtsuka N, Ikeda T, Taniguchi R, Blau D, et al. Role of mouse hepatitis virus (MHV) \\nreceptor murine CEACAM1 in the resistance of mice to MHV infection: studies of mice with \\nchimeric mCEACAMla and mCEACAMlb. J Virol. 2010; 84:6654–66. [PubMed: 20410265] \\n98. Seetharaman J, Kanigsberg A, Slaaby R, Leffler H, Barondes SH, Rini JM. X-ray crystal structure \\nof the human galectin-3 carbohydrate recognition domain at 2.1-Åresolution. J Biol Chem. 1998; \\n273:13047–52. [PubMed: 9582341] \\n99. Kunkel F, Herrler G. Structural and functional-analysis of the surface protein of human coronavirus \\nOC43. Virology. 1993; 195:195–202. [PubMed: 8317096] \\nLi Page 17\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript100. Vijgen L, Keyaerts E, Lemey P, Maes P, van Reeth K, et al. Evolutionary history of the closely \\nrelated group 2 coronaviruses: porcine hemagglutinating encephalomyelitis virus, bovine \\ncoronavirus, and human coronavirus OC43. J Virol. 2006; 80:7270–74. [PubMed: 16809333] \\n101. Vijgen L, Keyaerts E, Moes E, Thoelen I, Wollants E, et al. Complete genomic sequence of \\nhuman coronavirus OC43: molecular clock analysis suggests a relatively recent zoonotic \\ncoronavirus transmission event. J Virol. 2005; 79:1595–604. [PubMed: 15650185] \\n102. Klein A, Krishna M, Varki NM, Varki A. 9-O-acetylated sialic acids have widespread but \\nselective expression: analysis using a chimeric dual-function probe derived from influenza C \\nhemagglutinin-esterase. PNAS. 1994; 91:7782–86. [PubMed: 8052660] \\n103. Chen L, Li F. Structural analysis of the evolutionary origins of influenza virus hemagglutinin and \\nother viral lectins. J Virol. 2013; 87:4118–20. [PubMed: 23365425] \\n104. Eckert DM, Kim PS. Mechanisms of viral membrane fusion and its inhibition. Annu Rev \\nBiochem. 2001; 70:777–810. [PubMed: 11395423] \\n105. Skehel JJ, Wiley DC. Receptor binding and membrane fusion in virus entry: the influenza \\nhemagglutinin. Annu Rev Biochem. 2000; 69:531–69. [PubMed: 10966468] \\n106. Klenk HD, Garten W. Host cell proteases controlling virus pathogenicity. Trends Microbiol. 1994; \\n2:39–43. [PubMed: 8162439] \\n107. Wilson IA, Skehel JJ, Wiley DC. Structure of the haemagglutinin membrane glycoprotein of \\ninfluenza virus at 3 Å resolution. Nature. 1981; 289:366–73. [PubMed: 7464906] \\n108. Stein BS, Gowda SD, Lifson JD, Penhallow RC, Bensch KG, Engleman EG. pH-independent \\nHIV entry into CD4-positive T cells via virus envelope fusion to the plasma membrane. Cell. \\n1987; 49:659–68. [PubMed: 3107838] \\n109. White J, Matlin K, Helenius A. Cell fusion by Semliki Forest, influenza, and vesicular stomatitis \\nviruses. J Cell Biol. 1981; 89:674–79. [PubMed: 6265470] \\n110. Mothes W, Boerger AL, Narayan S, Cunningham JM, Young JA. Retroviral entry mediated by \\nreceptor priming and low pH triggering of an envelope glycoprotein. Cell. 2000; 103:679–89. \\n[PubMed: 11106737] \\n111. Belouzard S, Millet JK, Licitra BN, Whittaker GR. Mechanisms of coronavirus cell entry \\nmediated by the viral spike protein. Viruses. 2012; 4:1011–33. [PubMed: 22816037] \\n112. Heald-Sargent T, Gallagher T. Ready, set, fuse! The coronavirus spike protein and acquisition of \\nfusion competence. Viruses. 2012; 4:557–80. [PubMed: 22590686] \\n113. Xu YH, Lou ZY , Liu YW, Pang H, Tien P, et al. Crystal structure of severe acute respiratory \\nsyndrome coronavirus spike protein fusion core. J Biol Chem. 2004; 279:49414–19. [PubMed: \\n15345712] \\n114. Lu L, Liu Q, Zhu Y , Chan KH, Qin L, et al. Structure-based discovery of Middle East respiratory \\nsyndrome coronavirus fusion inhibitor. Nat Commun. 2014; 5:3067. [PubMed: 24473083] \\n115. Zheng Q, Deng Y , Liu J, van der Hoek L, Berkhout B, Lu M. Core structure of S2 from the \\nhuman coronavirus NL63 spike glycoprotein. Biochemistry. 2006; 45:15205–15. [PubMed: \\n17176042] \\n116. Xu Y , Liu Y , Lou Z, Qin L, Li X, et al. Structural basis for coronavirus-mediated membrane \\nfusion. Crystal structure of mouse hepatitis virus spike protein fusion core. J Biol Chem. 2004; \\n279:30514–22. [PubMed: 15123674] \\n117. Duquerroy S, Vigouroux AN, Rottier PJM, Rey FA, Bosch BJ. Central ions and lateral \\nasparagine/glutamine zippers stabilize the post-fusion hairpin conformation of the SARS \\ncoronavirus spike glycoprotein. Virology. 2005; 335:276–85. [PubMed: 15840526] \\n118. Gao J, Lu G, Qi J, Li Y , Wu Y , et al. Structure of the fusion core and inhibition of fusion by a \\nheptad repeat peptide derived from the S protein of Middle East respiratory syndrome \\ncoronavirus. J Virol. 2013; 87:13134–40. [PubMed: 24067982] \\n119. Supekar VM, Bruckmann C, Ingallinella P, Bianchi E, Pessi A, Carfi A. Structure of a \\nproteolytically resistant core from the severe acute respiratory syndrome coronavirus S2 fusion \\nprotein. PNAS. 2004; 101:17958–63. [PubMed: 15604146] \\n120. Millet JK, Whittaker GR. Host cell proteases: critical determinants of coronavirus tropism and \\npathogenesis. Virus Res. 2015; 202:120–34. [PubMed: 25445340] \\nLi Page 18\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript121. Spaan W, Cavanagh D, Horzinek MC. Coronaviruses: structure and genome expression. J Gen \\nVirol. 1988; 69:2939–52. [PubMed: 3058868] \\n122. Matsuyama S, Taguchi F. Receptor-induced conformational changes of murine coronavirus spike \\nprotein. J Virol. 2002; 76:11819–26. [PubMed: 12414924] \\n123. Zelus BD, Schickli JH, Blau DM, Weiss SR, Holmes KV . Conformational changes in the spike \\nglycoprotein of murine coronavirus are induced at 37° C either by soluble murine CEACAM1 \\nreceptors or by pH 8. J Virol. 2003; 77:830–40. [PubMed: 12502799] \\n124. Qiu Z, Hingley ST, Simmons G, Yu C, Das Sarma J, et al. Endosomal proteolysis by cathepsins is \\nnecessary for murine coronavirus mouse hepatitis virus type 2 spike-mediated entry. J Virol. \\n2006; 80:5768–76. [PubMed: 16731916] \\n125. Eifart P, Ludwig K, Bottcher C, de Haan CA, Rottier PJ, et al. Role of endocytosis and low pH in \\nmurine hepatitis virus strain A59 cell entry. J Virol. 2007; 81:10758–68. [PubMed: 17626088] \\n126. Nakagaki K, Nakagaki K, Taguchi F. Receptor-independent spread of a highly neurotropic murine \\ncoronavirus JHMV strain from initially infected microglial cells in mixed neural cultures. J Virol. \\n2005; 79:6102–10. [PubMed: 15857995] \\n127. Gallagher TM, Buchmeier MJ, Perlman S. Cell receptor-independent infection by a neurotropic \\nmurine coronavirus. Virology. 1992; 191:517–22. [PubMed: 1413526] \\n128. Taguchi F, Matsuyama S, Saeki K. Difference in Bgp-independent fusion activity among mouse \\nhepatitis viruses. Arch Virol. 1999; 144:2041–49'],\n",
       " 'sources': ['27578435',\n",
       "  '27578435',\n",
       "  '27578435',\n",
       "  '06463922',\n",
       "  '39034968',\n",
       "  '32007145',\n",
       "  '27578435',\n",
       "  '39034968',\n",
       "  '39034968',\n",
       "  '27578435'],\n",
       " 'scores': [0.9376382,\n",
       "  0.9336061,\n",
       "  0.926324,\n",
       "  0.91503865,\n",
       "  0.90320814,\n",
       "  0.89567137,\n",
       "  0.8926408,\n",
       "  0.88726485,\n",
       "  0.8760837,\n",
       "  0.83742017],\n",
       " 'store_name': 's'}"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from langchain_community.vectorstores import FAISS\n",
    "from langchain_community.embeddings import HuggingFaceEmbeddings \n",
    "query=\"molecule\"\n",
    "top_k=10\n",
    "embedding_name=\"BAAI/bge-small-en-v1.5\"\n",
    "embedding_model = HuggingFaceEmbeddings(model_name=embedding_name,\n",
    "                                    model_kwargs={\"device\": \"mps\"},\n",
    "                                    encode_kwargs={\"normalize_embeddings\": True},)\n",
    "\n",
    "\n",
    "vector_store = FAISS.load_local(\n",
    "    \"tmp/vector_stores/ex1\", \n",
    "    embedding_model, \n",
    "    allow_dangerous_deserialization=True  \n",
    "    )\n",
    "\n",
    "vector_data = {\"num_vectors\" : vector_store.index.ntotal,\n",
    "\"vector_dim\" : vector_store.index.d,\n",
    "\"distance_strategy\" : vector_store.distance_strategy}\n",
    "\n",
    "\n",
    "results = vector_store.similarity_search_with_score(query, top_k)\n",
    "\n",
    "context = [r[0].page_content for r in results][::-1]\n",
    "sources = [r[0].metadata[\"source\"] for r in results][::-1]\n",
    "scores = [r[1] for r in results][::-1]\n",
    "\n",
    "dict_ = {\n",
    "    \"context\": context,\n",
    "    \"sources\": sources,\n",
    "    \"scores\": scores,\n",
    "    \"store_name\": \"s\"\n",
    "}\n",
    "dict_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "f6f6e901",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on IndexFlatL2 in module faiss.swigfaiss object:\n",
      "\n",
      "class IndexFlatL2(IndexFlat)\n",
      " |  IndexFlatL2(*args)\n",
      " |  \n",
      " |  Method resolution order:\n",
      " |      IndexFlatL2\n",
      " |      IndexFlat\n",
      " |      IndexFlatCodes\n",
      " |      Index\n",
      " |      builtins.object\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __getstate__ = index_getstate(self)\n",
      " |  \n",
      " |  __init__(self, *args)\n",
      " |      :type d: int\n",
      " |      :param d: dimensionality of the input vectors\n",
      " |  \n",
      " |  __repr__ = _swig_repr(self)\n",
      " |  \n",
      " |  __setattr__ = replacement_setattr(self, name, value)\n",
      " |  \n",
      " |  __setstate__ = index_setstate(self, st)\n",
      " |  \n",
      " |  clear_l2norms(self)\n",
      " |  \n",
      " |  get_FlatCodesDistanceComputer(self)\n",
      " |       a FlatCodesDistanceComputer offers a distance_to_code method\n",
      " |      \n",
      " |      The default implementation explicitly decodes the vector with sa_decode.\n",
      " |  \n",
      " |  sync_l2norms(self)\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Static methods defined here:\n",
      " |  \n",
      " |  __swig_destroy__ = delete_IndexFlatL2(...)\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors defined here:\n",
      " |  \n",
      " |  cached_l2norms\n",
      " |  \n",
      " |  thisown\n",
      " |      The membership flag\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from IndexFlat:\n",
      " |  \n",
      " |  add = replacement_add(self, x, numeric_type=0)\n",
      " |      Adds vectors to the index.\n",
      " |      The index must be trained before vectors can be added to it.\n",
      " |      The vectors are implicitly numbered in sequence. When `n` vectors are\n",
      " |      added to the index, they are given ids `ntotal`, `ntotal + 1`, ..., `ntotal + n - 1`.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      x : array_like\n",
      " |          Query vectors, shape (n, d) where d is appropriate for the index.\n",
      " |          `dtype` must be float32.\n",
      " |  \n",
      " |  add_c = add(self, n, x)\n",
      " |      default add uses sa_encode\n",
      " |  \n",
      " |  add_sa_codes = replacement_add_sa_codes(self, codes, ids=None)\n",
      " |  \n",
      " |  add_sa_codes_c = add_sa_codes(self, n, x, xids)\n",
      " |  \n",
      " |  compute_distance_subset(self, n, x, k, distances, labels)\n",
      " |       compute distance with a subset of vectors\n",
      " |      \n",
      " |      :type x: float\n",
      " |      :param x:       query vectors, size n * d\n",
      " |      :type labels: int\n",
      " |      :param labels:  indices of the vectors that should be compared\n",
      " |                         for each query vector, size n * k\n",
      " |      :type distances: float\n",
      " |      :param distances:\n",
      " |                         corresponding output distances, size n * k\n",
      " |  \n",
      " |  get_xb(self, *args)\n",
      " |  \n",
      " |  permute_entries = replacement_permute_entries(self, perm)\n",
      " |  \n",
      " |  permute_entries_c = permute_entries(self, perm)\n",
      " |  \n",
      " |  range_search = replacement_range_search(self, x, thresh, *, params=None)\n",
      " |      Search vectors that are within a distance of the query vectors.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      x : array_like\n",
      " |          Query vectors, shape (n, d) where d is appropriate for the index.\n",
      " |          `dtype` must be float32.\n",
      " |      thresh : float\n",
      " |          Threshold to select neighbors. All elements within this radius are returned,\n",
      " |          except for maximum inner product indexes, where the elements above the\n",
      " |          threshold are returned\n",
      " |      params : SearchParameters\n",
      " |          Search parameters of the current search (overrides the class-level params)\n",
      " |      \n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      lims: array_like\n",
      " |          Starting index of the results for each query vector, size n+1.\n",
      " |      D : array_like\n",
      " |          Distances of the nearest neighbors, shape `lims[n]`. The distances for\n",
      " |          query i are in `D[lims[i]:lims[i+1]]`.\n",
      " |      I : array_like\n",
      " |          Labels of nearest neighbors, shape `lims[n]`. The labels for query i\n",
      " |          are in `I[lims[i]:lims[i+1]]`.\n",
      " |  \n",
      " |  range_search_c = range_search(self, n, x, radius, result, params=None)\n",
      " |  \n",
      " |  reconstruct = replacement_reconstruct(self, key, x=None)\n",
      " |      Approximate reconstruction of one vector from the index.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      key : int\n",
      " |          Id of the vector to reconstruct\n",
      " |      x : array_like, optional\n",
      " |          pre-allocated array to store the results\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      x : array_like reconstructed vector, size `self.d`, `dtype`=float32\n",
      " |  \n",
      " |  reconstruct_c = reconstruct(self, key, recons)\n",
      " |  \n",
      " |  reconstruct_n = replacement_reconstruct_n(self, n0=0, ni=-1, x=None)\n",
      " |      Approximate reconstruction of vectors `n0` ... `n0 + ni - 1` from the index.\n",
      " |      Missing vectors trigger an exception.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      n0 : int\n",
      " |          Id of the first vector to reconstruct (default 0)\n",
      " |      ni : int\n",
      " |          Number of vectors to reconstruct (-1 = default = ntotal)\n",
      " |      x : array_like, optional\n",
      " |          pre-allocated array to store the results\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      x : array_like\n",
      " |          Reconstructed vectors, size (`ni`, `self.d`), `dtype`=float32\n",
      " |  \n",
      " |  reconstruct_n_c = reconstruct_n(self, i0, ni, recons)\n",
      " |  \n",
      " |  remove_ids = replacement_remove_ids(self, x)\n",
      " |      Remove some ids from the index.\n",
      " |      This is a O(ntotal) operation by default, so could be expensive.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      x : array_like or faiss.IDSelector\n",
      " |          Either an IDSelector that returns True for vectors to remove, or a\n",
      " |          list of ids to reomove (1D array of int64). When `x` is a list,\n",
      " |          it is wrapped into an IDSelector.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      n_remove: int\n",
      " |          number of vectors that were removed\n",
      " |  \n",
      " |  remove_ids_c = remove_ids(self, sel)\n",
      " |       remove some ids. NB that because of the structure of the\n",
      " |      index, the semantics of this operation are\n",
      " |      different from the usual ones: the new ids are shifted\n",
      " |  \n",
      " |  sa_decode = replacement_sa_decode(self, codes, x=None)\n",
      " |  \n",
      " |  sa_decode_c = sa_decode(self, n, bytes, x)\n",
      " |  \n",
      " |  sa_encode = replacement_sa_encode(self, x, codes=None)\n",
      " |  \n",
      " |  sa_encode_c = sa_encode(self, n, x, bytes)\n",
      " |  \n",
      " |  search = replacement_search(self, x, k, *, params=None, D=None, I=None, numeric_type=0)\n",
      " |      Find the k nearest neighbors of the set of vectors x in the index.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      x : array_like\n",
      " |          Query vectors, shape (n, d) where d is appropriate for the index.\n",
      " |          `dtype` must be float32.\n",
      " |      k : int\n",
      " |          Number of nearest neighbors.\n",
      " |      params : SearchParameters\n",
      " |          Search parameters of the current search (overrides the class-level params)\n",
      " |      D : array_like, optional\n",
      " |          Distance array to store the result.\n",
      " |      I : array_like, optional\n",
      " |          Labels array to store the results.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      D : array_like\n",
      " |          Distances of the nearest neighbors, shape (n, k). When not enough results are found\n",
      " |          the label is set to +Inf or -Inf.\n",
      " |      I : array_like\n",
      " |          Labels of the nearest neighbors, shape (n, k).\n",
      " |          When not enough results are found, the label is set to -1\n",
      " |  \n",
      " |  search_c = search(self, n, x, k, distances, labels, params=None)\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from IndexFlatCodes:\n",
      " |  \n",
      " |  check_compatible_for_merge(self, otherIndex)\n",
      " |       check that the two indexes are compatible (ie, they are\n",
      " |      trained in the same way and have the same\n",
      " |      parameters). Otherwise throw.\n",
      " |  \n",
      " |  get_CodePacker(self)\n",
      " |  \n",
      " |  get_distance_computer(self)\n",
      " |       Get a DistanceComputer (defined in AuxIndexStructures) object\n",
      " |      for this kind of index.\n",
      " |      \n",
      " |      DistanceComputer is implemented for indexes that support random\n",
      " |      access of their vectors.\n",
      " |  \n",
      " |  merge_from(self, otherIndex, add_id=0)\n",
      " |       moves the entries from another dataset to self.\n",
      " |      On output, other is empty.\n",
      " |      add_id is added to all moved ids\n",
      " |      (for sequential ids, this would be this->ntotal)\n",
      " |  \n",
      " |  reset(self)\n",
      " |      removes all elements from the database.\n",
      " |  \n",
      " |  sa_code_size(self)\n",
      " |      size of the produced codes in bytes\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors inherited from IndexFlatCodes:\n",
      " |  \n",
      " |  code_size\n",
      " |  \n",
      " |  codes\n",
      " |      encoded dataset, size ntotal * code_size\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from Index:\n",
      " |  \n",
      " |  add_ex(self, n, x, numeric_type)\n",
      " |  \n",
      " |  add_with_ids = replacement_add_with_ids(self, x, ids, numeric_type=0)\n",
      " |      Adds vectors with arbitrary ids to the index (not all indexes support this).\n",
      " |      The index must be trained before vectors can be added to it.\n",
      " |      Vector `i` is stored in `x[i]` and has id `ids[i]`.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      x : array_like\n",
      " |          Query vectors, shape (n, d) where d is appropriate for the index.\n",
      " |          `dtype` must be float32.\n",
      " |      ids : array_like\n",
      " |          Array if ids of size n. The ids must be of type `int64`. Note that `-1` is reserved\n",
      " |          in result lists to mean \"not found\" so it's better to not use it as an id.\n",
      " |  \n",
      " |  add_with_ids_c = add_with_ids(self, n, x, xids)\n",
      " |       Same as add, but stores xids instead of sequential ids.\n",
      " |      \n",
      " |      The default implementation fails with an assertion, as it is\n",
      " |      not supported by all indexes.\n",
      " |      \n",
      " |      :type n: int\n",
      " |      :param n:         number of vectors\n",
      " |      :type x: float\n",
      " |      :param x:         input vectors, size n * d\n",
      " |      :type xids: int\n",
      " |      :param xids:      if non-null, ids to store for the vectors (size n)\n",
      " |  \n",
      " |  add_with_ids_ex(self, n, x, numeric_type, xids)\n",
      " |  \n",
      " |  assign = replacement_assign(self, x, k, labels=None)\n",
      " |      Find the k nearest neighbors of the set of vectors x in the index.\n",
      " |      This is the same as the `search` method, but discards the distances.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      x : array_like\n",
      " |          Query vectors, shape (n, d) where d is appropriate for the index.\n",
      " |          `dtype` must be float32.\n",
      " |      k : int\n",
      " |          Number of nearest neighbors.\n",
      " |      labels : array_like, optional\n",
      " |          Labels array to store the results.\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      labels: array_like\n",
      " |          Labels of the nearest neighbors, shape (n, k).\n",
      " |          When not enough results are found, the label is set to -1\n",
      " |  \n",
      " |  assign_c = assign(self, n, x, labels, k=1)\n",
      " |       return the indexes of the k vectors closest to the query x.\n",
      " |      \n",
      " |      This function is identical as search but only return labels of\n",
      " |      neighbors.\n",
      " |      :type n: int\n",
      " |      :param n:           number of vectors\n",
      " |      :type x: float\n",
      " |      :param x:           input vectors to search, size n * d\n",
      " |      :type labels: int\n",
      " |      :param labels:      output labels of the NNs, size n*k\n",
      " |      :type k: int, optional\n",
      " |      :param k:           number of nearest neighbours\n",
      " |  \n",
      " |  compute_residual(self, x, residual, key)\n",
      " |       Computes a residual vector after indexing encoding.\n",
      " |      \n",
      " |      The residual vector is the difference between a vector and the\n",
      " |      reconstruction that can be decoded from its representation in\n",
      " |      the index. The residual can be used for multiple-stage indexing\n",
      " |      methods, like IndexIVF's methods.\n",
      " |      \n",
      " |      :type x: float\n",
      " |      :param x:           input vector, size d\n",
      " |      :type residual: float\n",
      " |      :param residual:    output residual vector, size d\n",
      " |      :type key: int\n",
      " |      :param key:         encoded index, as returned by search and assign\n",
      " |  \n",
      " |  compute_residual_n(self, n, xs, residuals, keys)\n",
      " |       Computes a residual vector after indexing encoding (batch form).\n",
      " |      Equivalent to calling compute_residual for each vector.\n",
      " |      \n",
      " |      The residual vector is the difference between a vector and the\n",
      " |      reconstruction that can be decoded from its representation in\n",
      " |      the index. The residual can be used for multiple-stage indexing\n",
      " |      methods, like IndexIVF's methods.\n",
      " |      \n",
      " |      :type n: int\n",
      " |      :param n:           number of vectors\n",
      " |      :type xs: float\n",
      " |      :param xs:          input vectors, size (n x d)\n",
      " |      :type residuals: float\n",
      " |      :param residuals:   output residual vectors, size (n x d)\n",
      " |      :type keys: int\n",
      " |      :param keys:        encoded index, as returned by search and assign\n",
      " |  \n",
      " |  reconstruct_batch = replacement_reconstruct_batch(self, key, x=None)\n",
      " |      Approximate reconstruction of several vectors from the index.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      key : array of ints\n",
      " |          Ids of the vectors to reconstruct\n",
      " |      x : array_like, optional\n",
      " |          pre-allocated array to store the results\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      x : array_like\n",
      " |          reconstrcuted vectors, size `len(key), self.d`\n",
      " |  \n",
      " |  reconstruct_batch_c = reconstruct_batch(self, n, keys, recons)\n",
      " |       Reconstruct several stored vectors (or an approximation if lossy\n",
      " |      coding)\n",
      " |      \n",
      " |      this function may not be defined for some indexes\n",
      " |      :type n: int\n",
      " |      :param n:           number of vectors to reconstruct\n",
      " |      :type keys: int\n",
      " |      :param keys:        ids of the vectors to reconstruct (size n)\n",
      " |      :type recons: float\n",
      " |      :param recons:      reconstructed vector (size n * d)\n",
      " |  \n",
      " |  search_and_reconstruct = replacement_search_and_reconstruct(self, x, k, *, params=None, D=None, I=None, R=None)\n",
      " |      Find the k nearest neighbors of the set of vectors x in the index,\n",
      " |      and return an approximation of these vectors.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      x : array_like\n",
      " |          Query vectors, shape (n, d) where d is appropriate for the index.\n",
      " |          `dtype` must be float32.\n",
      " |      k : int\n",
      " |          Number of nearest neighbors.\n",
      " |      params : SearchParameters\n",
      " |          Search parameters of the current search (overrides the class-level params)\n",
      " |      D : array_like, optional\n",
      " |          Distance array to store the result.\n",
      " |      I : array_like, optional\n",
      " |          Labels array to store the result.\n",
      " |      R : array_like, optional\n",
      " |          reconstruction array to store\n",
      " |      \n",
      " |      Returns\n",
      " |      -------\n",
      " |      D : array_like\n",
      " |          Distances of the nearest neighbors, shape (n, k). When not enough results are found\n",
      " |          the label is set to +Inf or -Inf.\n",
      " |      I : array_like\n",
      " |          Labels of the nearest neighbors, shape (n, k). When not enough results are found,\n",
      " |          the label is set to -1\n",
      " |      R : array_like\n",
      " |          Approximate (reconstructed) nearest neighbor vectors, shape (n, k, d).\n",
      " |  \n",
      " |  search_and_reconstruct_c = search_and_reconstruct(self, n, x, k, distances, labels, recons, params=None)\n",
      " |       Similar to search, but also reconstructs the stored vectors (or an\n",
      " |      approximation in the case of lossy coding) for the search results.\n",
      " |      \n",
      " |      If there are not enough results for a query, the resulting arrays\n",
      " |      is padded with -1s.\n",
      " |      \n",
      " |      :type n: int\n",
      " |      :param n:           number of vectors\n",
      " |      :type x: float\n",
      " |      :param x:           input vectors to search, size n * d\n",
      " |      :type k: int\n",
      " |      :param k:           number of extracted vectors\n",
      " |      :type distances: float\n",
      " |      :param distances:   output pairwise distances, size n*k\n",
      " |      :type labels: int\n",
      " |      :param labels:      output labels of the NNs, size n*k\n",
      " |      :type recons: float\n",
      " |      :param recons:      reconstructed vectors size (n, k, d)\n",
      " |  \n",
      " |  search_ex(self, n, x, numeric_type, k, distances, labels, params=None)\n",
      " |  \n",
      " |  train = replacement_train(self, x, numeric_type=0)\n",
      " |      Trains the index on a representative set of vectors.\n",
      " |      The index must be trained before vectors can be added to it.\n",
      " |      \n",
      " |      Parameters\n",
      " |      ----------\n",
      " |      x : array_like\n",
      " |          Query vectors, shape (n, d) where d is appropriate for the index.\n",
      " |          `dtype` must be float32.\n",
      " |  \n",
      " |  train_c = train(self, n, x)\n",
      " |       Perform training on a representative set of vectors\n",
      " |      \n",
      " |      :type n: int\n",
      " |      :param n:      nb of training vectors\n",
      " |      :type x: float\n",
      " |      :param x:      training vectors, size n * d\n",
      " |  \n",
      " |  train_ex(self, n, x, numeric_type)\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors inherited from Index:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      " |  \n",
      " |  d\n",
      " |      vector dimension\n",
      " |  \n",
      " |  is_trained\n",
      " |      set if the Index does not require training, or if training is\n",
      " |      done already\n",
      " |  \n",
      " |  metric_arg\n",
      " |      argument of the metric type\n",
      " |  \n",
      " |  metric_type\n",
      " |      type of metric this index uses for search\n",
      " |  \n",
      " |  ntotal\n",
      " |      total nb of indexed vectors\n",
      " |  \n",
      " |  verbose\n",
      " |      verbosity level\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(vector_store.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "706212f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'num_vectors': 22, 'vector_dim': 384, 'distance_strategy': 0.0}"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "{\"num_vectors\" : vector_store.index.ntotal,\n",
    "\"vector_dim\" : vector_store.index.d,\n",
    "\"distance_strategy\" : vector_store.distance_strategy}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "307b8020",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on Document in module langchain_core.documents.base object:\n",
      "\n",
      "class Document(BaseMedia)\n",
      " |  Document(page_content: 'str', *, id: typing.Annotated[str | None, _PydanticGeneralMetadata(coerce_numbers_to_str=True)] = None, metadata: dict = <factory>, type: Literal['Document'] = 'Document') -> None\n",
      " |  \n",
      " |  Class for storing a piece of text and associated metadata.\n",
      " |  \n",
      " |  !!! note\n",
      " |      `Document` is for **retrieval workflows**, not chat I/O. For sending text\n",
      " |      to an LLM in a conversation, use message types from `langchain.messages`.\n",
      " |  \n",
      " |  Example:\n",
      " |      ```python\n",
      " |      from langchain_core.documents import Document\n",
      " |  \n",
      " |      document = Document(\n",
      " |          page_content=\"Hello, world!\", metadata={\"source\": \"https://example.com\"}\n",
      " |      )\n",
      " |      ```\n",
      " |  \n",
      " |  Method resolution order:\n",
      " |      Document\n",
      " |      BaseMedia\n",
      " |      langchain_core.load.serializable.Serializable\n",
      " |      pydantic.main.BaseModel\n",
      " |      abc.ABC\n",
      " |      builtins.object\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __init__(self, page_content: 'str', **kwargs: 'Any') -> 'None'\n",
      " |      Pass page_content in as positional or named arg.\n",
      " |  \n",
      " |  __str__(self) -> 'str'\n",
      " |      Override `__str__` to restrict it to page_content and metadata.\n",
      " |      \n",
      " |      Returns:\n",
      " |          A string representation of the `Document`.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Class methods defined here:\n",
      " |  \n",
      " |  get_lc_namespace() -> 'list[str]' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      Get the namespace of the LangChain object.\n",
      " |      \n",
      " |      Returns:\n",
      " |          [\"langchain\", \"schema\", \"document\"]\n",
      " |  \n",
      " |  is_lc_serializable() -> 'bool' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      Return `True` as this class is serializable.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data and other attributes defined here:\n",
      " |  \n",
      " |  __abstractmethods__ = frozenset()\n",
      " |  \n",
      " |  __annotations__ = {'page_content': 'str', 'type': \"Literal['Document']...\n",
      " |  \n",
      " |  __class_vars__ = set()\n",
      " |  \n",
      " |  __private_attributes__ = {}\n",
      " |  \n",
      " |  __pydantic_complete__ = True\n",
      " |  \n",
      " |  __pydantic_computed_fields__ = {}\n",
      " |  \n",
      " |  __pydantic_core_schema__ = {'cls': <class 'langchain_core.documents.ba...\n",
      " |  \n",
      " |  __pydantic_custom_init__ = True\n",
      " |  \n",
      " |  __pydantic_decorators__ = DecoratorInfos(validators={}, field_validato...\n",
      " |  \n",
      " |  __pydantic_fields__ = {'id': FieldInfo(annotation=Union[str, NoneType]...\n",
      " |  \n",
      " |  __pydantic_generic_metadata__ = {'args': (), 'origin': None, 'paramete...\n",
      " |  \n",
      " |  __pydantic_parent_namespace__ = None\n",
      " |  \n",
      " |  __pydantic_post_init__ = None\n",
      " |  \n",
      " |  __pydantic_serializer__ = SchemaSerializer(serializer=Model(\n",
      " |      Model...\n",
      " |  \n",
      " |  __pydantic_setattr_handlers__ = {}\n",
      " |  \n",
      " |  __pydantic_validator__ = SchemaValidator(title=\"Document\", validator=M...\n",
      " |  \n",
      " |  __signature__ = <Signature (page_content: 'str', *, id: typing.A... ty...\n",
      " |  \n",
      " |  model_config = {'extra': 'ignore'}\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from langchain_core.load.serializable.Serializable:\n",
      " |  \n",
      " |  __repr_args__(self) -> Any\n",
      " |  \n",
      " |  to_json(self) -> langchain_core.load.serializable.SerializedConstructor | langchain_core.load.serializable.SerializedNotImplemented\n",
      " |      Serialize the object to JSON.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValueError: If the class has deprecated attributes.\n",
      " |      \n",
      " |      Returns:\n",
      " |          A JSON serializable object or a `SerializedNotImplemented` object.\n",
      " |  \n",
      " |  to_json_not_implemented(self) -> langchain_core.load.serializable.SerializedNotImplemented\n",
      " |      Serialize a \"not implemented\" object.\n",
      " |      \n",
      " |      Returns:\n",
      " |          `SerializedNotImplemented`.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Class methods inherited from langchain_core.load.serializable.Serializable:\n",
      " |  \n",
      " |  lc_id() -> list[str] from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      Return a unique identifier for this class for serialization purposes.\n",
      " |      \n",
      " |      The unique identifier is a list of strings that describes the path\n",
      " |      to the object.\n",
      " |      \n",
      " |      For example, for the class `langchain.llms.openai.OpenAI`, the id is\n",
      " |      `[\"langchain\", \"llms\", \"openai\", \"OpenAI\"]`.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Readonly properties inherited from langchain_core.load.serializable.Serializable:\n",
      " |  \n",
      " |  lc_attributes\n",
      " |      List of attribute names that should be included in the serialized kwargs.\n",
      " |      \n",
      " |      These attributes must be accepted by the constructor.\n",
      " |      \n",
      " |      Default is an empty dictionary.\n",
      " |  \n",
      " |  lc_secrets\n",
      " |      A map of constructor argument names to secret ids.\n",
      " |      \n",
      " |      For example, `{\"openai_api_key\": \"OPENAI_API_KEY\"}`\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors inherited from langchain_core.load.serializable.Serializable:\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from pydantic.main.BaseModel:\n",
      " |  \n",
      " |  __copy__(self) -> 'Self'\n",
      " |      Returns a shallow copy of the model.\n",
      " |  \n",
      " |  __deepcopy__(self, memo: 'dict[int, Any] | None' = None) -> 'Self'\n",
      " |      Returns a deep copy of the model.\n",
      " |  \n",
      " |  __delattr__(self, item: 'str') -> 'Any'\n",
      " |      Implement delattr(self, name).\n",
      " |  \n",
      " |  __eq__(self, other: 'Any') -> 'bool'\n",
      " |      Return self==value.\n",
      " |  \n",
      " |  __getattr__(self, item: 'str') -> 'Any'\n",
      " |  \n",
      " |  __getstate__(self) -> 'dict[Any, Any]'\n",
      " |  \n",
      " |  __iter__(self) -> 'TupleGenerator'\n",
      " |      So `dict(model)` works.\n",
      " |  \n",
      " |  __pretty__(self, fmt: 'Callable[[Any], Any]', **kwargs: 'Any') -> 'Generator[Any]'\n",
      " |      Used by devtools (https://python-devtools.helpmanual.io/) to pretty print objects.\n",
      " |  \n",
      " |  __replace__(self, **changes: 'Any') -> 'Self'\n",
      " |      # Because we make use of `@dataclass_transform()`, `__replace__` is already synthesized by\n",
      " |      # type checkers, so we define the implementation in this `if not TYPE_CHECKING:` block:\n",
      " |  \n",
      " |  __repr__(self) -> 'str'\n",
      " |      Return repr(self).\n",
      " |  \n",
      " |  __repr_name__(self) -> 'str'\n",
      " |      Name of the instance's class, used in __repr__.\n",
      " |  \n",
      " |  __repr_recursion__(self, object: 'Any') -> 'str'\n",
      " |      Returns the string representation of a recursive object.\n",
      " |  \n",
      " |  __repr_str__(self, join_str: 'str') -> 'str'\n",
      " |  \n",
      " |  __rich_repr__(self) -> 'RichReprResult'\n",
      " |      Used by Rich (https://rich.readthedocs.io/en/stable/pretty.html) to pretty print objects.\n",
      " |  \n",
      " |  __setattr__(self, name: 'str', value: 'Any') -> 'None'\n",
      " |      Implement setattr(self, name, value).\n",
      " |  \n",
      " |  __setstate__(self, state: 'dict[Any, Any]') -> 'None'\n",
      " |  \n",
      " |  copy(self, *, include: 'AbstractSetIntStr | MappingIntStrAny | None' = None, exclude: 'AbstractSetIntStr | MappingIntStrAny | None' = None, update: 'Dict[str, Any] | None' = None, deep: 'bool' = False) -> 'Self'\n",
      " |      Returns a copy of the model.\n",
      " |      \n",
      " |      !!! warning \"Deprecated\"\n",
      " |          This method is now deprecated; use `model_copy` instead.\n",
      " |      \n",
      " |      If you need `include` or `exclude`, use:\n",
      " |      \n",
      " |      ```python {test=\"skip\" lint=\"skip\"}\n",
      " |      data = self.model_dump(include=include, exclude=exclude, round_trip=True)\n",
      " |      data = {**data, **(update or {})}\n",
      " |      copied = self.model_validate(data)\n",
      " |      ```\n",
      " |      \n",
      " |      Args:\n",
      " |          include: Optional set or mapping specifying which fields to include in the copied model.\n",
      " |          exclude: Optional set or mapping specifying which fields to exclude in the copied model.\n",
      " |          update: Optional dictionary of field-value pairs to override field values in the copied model.\n",
      " |          deep: If True, the values of fields that are Pydantic models will be deep-copied.\n",
      " |      \n",
      " |      Returns:\n",
      " |          A copy of the model with included, excluded and updated fields as specified.\n",
      " |  \n",
      " |  dict(self, *, include: 'IncEx | None' = None, exclude: 'IncEx | None' = None, by_alias: 'bool' = False, exclude_unset: 'bool' = False, exclude_defaults: 'bool' = False, exclude_none: 'bool' = False) -> 'Dict[str, Any]'\n",
      " |  \n",
      " |  json(self, *, include: 'IncEx | None' = None, exclude: 'IncEx | None' = None, by_alias: 'bool' = False, exclude_unset: 'bool' = False, exclude_defaults: 'bool' = False, exclude_none: 'bool' = False, encoder: 'Callable[[Any], Any] | None' = PydanticUndefined, models_as_dict: 'bool' = PydanticUndefined, **dumps_kwargs: 'Any') -> 'str'\n",
      " |  \n",
      " |  model_copy(self, *, update: 'Mapping[str, Any] | None' = None, deep: 'bool' = False) -> 'Self'\n",
      " |      !!! abstract \"Usage Documentation\"\n",
      " |          [`model_copy`](../concepts/models.md#model-copy)\n",
      " |      \n",
      " |      Returns a copy of the model.\n",
      " |      \n",
      " |      !!! note\n",
      " |          The underlying instance's [`__dict__`][object.__dict__] attribute is copied. This\n",
      " |          might have unexpected side effects if you store anything in it, on top of the model\n",
      " |          fields (e.g. the value of [cached properties][functools.cached_property]).\n",
      " |      \n",
      " |      Args:\n",
      " |          update: Values to change/add in the new model. Note: the data is not validated\n",
      " |              before creating the new model. You should trust this data.\n",
      " |          deep: Set to `True` to make a deep copy of the model.\n",
      " |      \n",
      " |      Returns:\n",
      " |          New model instance.\n",
      " |  \n",
      " |  model_dump(self, *, mode: \"Literal['json', 'python'] | str\" = 'python', include: 'IncEx | None' = None, exclude: 'IncEx | None' = None, context: 'Any | None' = None, by_alias: 'bool | None' = None, exclude_unset: 'bool' = False, exclude_defaults: 'bool' = False, exclude_none: 'bool' = False, exclude_computed_fields: 'bool' = False, round_trip: 'bool' = False, warnings: \"bool | Literal['none', 'warn', 'error']\" = True, fallback: 'Callable[[Any], Any] | None' = None, serialize_as_any: 'bool' = False) -> 'dict[str, Any]'\n",
      " |      !!! abstract \"Usage Documentation\"\n",
      " |          [`model_dump`](../concepts/serialization.md#python-mode)\n",
      " |      \n",
      " |      Generate a dictionary representation of the model, optionally specifying which fields to include or exclude.\n",
      " |      \n",
      " |      Args:\n",
      " |          mode: The mode in which `to_python` should run.\n",
      " |              If mode is 'json', the output will only contain JSON serializable types.\n",
      " |              If mode is 'python', the output may contain non-JSON-serializable Python objects.\n",
      " |          include: A set of fields to include in the output.\n",
      " |          exclude: A set of fields to exclude from the output.\n",
      " |          context: Additional context to pass to the serializer.\n",
      " |          by_alias: Whether to use the field's alias in the dictionary key if defined.\n",
      " |          exclude_unset: Whether to exclude fields that have not been explicitly set.\n",
      " |          exclude_defaults: Whether to exclude fields that are set to their default value.\n",
      " |          exclude_none: Whether to exclude fields that have a value of `None`.\n",
      " |          exclude_computed_fields: Whether to exclude computed fields.\n",
      " |              While this can be useful for round-tripping, it is usually recommended to use the dedicated\n",
      " |              `round_trip` parameter instead.\n",
      " |          round_trip: If True, dumped values should be valid as input for non-idempotent types such as Json[T].\n",
      " |          warnings: How to handle serialization errors. False/\"none\" ignores them, True/\"warn\" logs errors,\n",
      " |              \"error\" raises a [`PydanticSerializationError`][pydantic_core.PydanticSerializationError].\n",
      " |          fallback: A function to call when an unknown value is encountered. If not provided,\n",
      " |              a [`PydanticSerializationError`][pydantic_core.PydanticSerializationError] error is raised.\n",
      " |          serialize_as_any: Whether to serialize fields with duck-typing serialization behavior.\n",
      " |      \n",
      " |      Returns:\n",
      " |          A dictionary representation of the model.\n",
      " |  \n",
      " |  model_dump_json(self, *, indent: 'int | None' = None, ensure_ascii: 'bool' = False, include: 'IncEx | None' = None, exclude: 'IncEx | None' = None, context: 'Any | None' = None, by_alias: 'bool | None' = None, exclude_unset: 'bool' = False, exclude_defaults: 'bool' = False, exclude_none: 'bool' = False, exclude_computed_fields: 'bool' = False, round_trip: 'bool' = False, warnings: \"bool | Literal['none', 'warn', 'error']\" = True, fallback: 'Callable[[Any], Any] | None' = None, serialize_as_any: 'bool' = False) -> 'str'\n",
      " |      !!! abstract \"Usage Documentation\"\n",
      " |          [`model_dump_json`](../concepts/serialization.md#json-mode)\n",
      " |      \n",
      " |      Generates a JSON representation of the model using Pydantic's `to_json` method.\n",
      " |      \n",
      " |      Args:\n",
      " |          indent: Indentation to use in the JSON output. If None is passed, the output will be compact.\n",
      " |          ensure_ascii: If `True`, the output is guaranteed to have all incoming non-ASCII characters escaped.\n",
      " |              If `False` (the default), these characters will be output as-is.\n",
      " |          include: Field(s) to include in the JSON output.\n",
      " |          exclude: Field(s) to exclude from the JSON output.\n",
      " |          context: Additional context to pass to the serializer.\n",
      " |          by_alias: Whether to serialize using field aliases.\n",
      " |          exclude_unset: Whether to exclude fields that have not been explicitly set.\n",
      " |          exclude_defaults: Whether to exclude fields that are set to their default value.\n",
      " |          exclude_none: Whether to exclude fields that have a value of `None`.\n",
      " |          exclude_computed_fields: Whether to exclude computed fields.\n",
      " |              While this can be useful for round-tripping, it is usually recommended to use the dedicated\n",
      " |              `round_trip` parameter instead.\n",
      " |          round_trip: If True, dumped values should be valid as input for non-idempotent types such as Json[T].\n",
      " |          warnings: How to handle serialization errors. False/\"none\" ignores them, True/\"warn\" logs errors,\n",
      " |              \"error\" raises a [`PydanticSerializationError`][pydantic_core.PydanticSerializationError].\n",
      " |          fallback: A function to call when an unknown value is encountered. If not provided,\n",
      " |              a [`PydanticSerializationError`][pydantic_core.PydanticSerializationError] error is raised.\n",
      " |          serialize_as_any: Whether to serialize fields with duck-typing serialization behavior.\n",
      " |      \n",
      " |      Returns:\n",
      " |          A JSON string representation of the model.\n",
      " |  \n",
      " |  model_post_init(self, context: 'Any', /) -> 'None'\n",
      " |      Override this method to perform additional initialization after `__init__` and `model_construct`.\n",
      " |      This is useful if you want to do some validation that requires the entire model to be initialized.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Class methods inherited from pydantic.main.BaseModel:\n",
      " |  \n",
      " |  __class_getitem__(typevar_values: 'type[Any] | tuple[type[Any], ...]') -> 'type[BaseModel] | _forward_ref.PydanticRecursiveRef' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  __get_pydantic_core_schema__(source: 'type[BaseModel]', handler: 'GetCoreSchemaHandler', /) -> 'CoreSchema' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  __get_pydantic_json_schema__(core_schema: 'CoreSchema', handler: 'GetJsonSchemaHandler', /) -> 'JsonSchemaValue' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      Hook into generating the model's JSON schema.\n",
      " |      \n",
      " |      Args:\n",
      " |          core_schema: A `pydantic-core` CoreSchema.\n",
      " |              You can ignore this argument and call the handler with a new CoreSchema,\n",
      " |              wrap this CoreSchema (`{'type': 'nullable', 'schema': current_schema}`),\n",
      " |              or just call the handler with the original schema.\n",
      " |          handler: Call into Pydantic's internal JSON schema generation.\n",
      " |              This will raise a `pydantic.errors.PydanticInvalidForJsonSchema` if JSON schema\n",
      " |              generation fails.\n",
      " |              Since this gets called by `BaseModel.model_json_schema` you can override the\n",
      " |              `schema_generator` argument to that function to change JSON schema generation globally\n",
      " |              for a type.\n",
      " |      \n",
      " |      Returns:\n",
      " |          A JSON schema, as a Python object.\n",
      " |  \n",
      " |  __pydantic_init_subclass__(**kwargs: 'Any') -> 'None' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      This is intended to behave just like `__init_subclass__`, but is called by `ModelMetaclass`\n",
      " |      only after basic class initialization is complete. In particular, attributes like `model_fields` will\n",
      " |      be present when this is called, but forward annotations are not guaranteed to be resolved yet,\n",
      " |      meaning that creating an instance of the class may fail.\n",
      " |      \n",
      " |      This is necessary because `__init_subclass__` will always be called by `type.__new__`,\n",
      " |      and it would require a prohibitively large refactor to the `ModelMetaclass` to ensure that\n",
      " |      `type.__new__` was called in such a manner that the class would already be sufficiently initialized.\n",
      " |      \n",
      " |      This will receive the same `kwargs` that would be passed to the standard `__init_subclass__`, namely,\n",
      " |      any kwargs passed to the class definition that aren't used internally by Pydantic.\n",
      " |      \n",
      " |      Args:\n",
      " |          **kwargs: Any keyword arguments passed to the class definition that aren't used internally\n",
      " |              by Pydantic.\n",
      " |      \n",
      " |      Note:\n",
      " |          You may want to override [`__pydantic_on_complete__()`][pydantic.main.BaseModel.__pydantic_on_complete__]\n",
      " |          instead, which is called once the class and its fields are fully initialized and ready for validation.\n",
      " |  \n",
      " |  __pydantic_on_complete__() -> 'None' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      This is called once the class and its fields are fully initialized and ready to be used.\n",
      " |      \n",
      " |      This typically happens when the class is created (just before\n",
      " |      [`__pydantic_init_subclass__()`][pydantic.main.BaseModel.__pydantic_init_subclass__] is called on the superclass),\n",
      " |      except when forward annotations are used that could not immediately be resolved.\n",
      " |      In that case, it will be called later, when the model is rebuilt automatically or explicitly using\n",
      " |      [`model_rebuild()`][pydantic.main.BaseModel.model_rebuild].\n",
      " |  \n",
      " |  construct(_fields_set: 'set[str] | None' = None, **values: 'Any') -> 'Self' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  from_orm(obj: 'Any') -> 'Self' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  model_construct(_fields_set: 'set[str] | None' = None, **values: 'Any') -> 'Self' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      Creates a new instance of the `Model` class with validated data.\n",
      " |      \n",
      " |      Creates a new model setting `__dict__` and `__pydantic_fields_set__` from trusted or pre-validated data.\n",
      " |      Default values are respected, but no other validation is performed.\n",
      " |      \n",
      " |      !!! note\n",
      " |          `model_construct()` generally respects the `model_config.extra` setting on the provided model.\n",
      " |          That is, if `model_config.extra == 'allow'`, then all extra passed values are added to the model instance's `__dict__`\n",
      " |          and `__pydantic_extra__` fields. If `model_config.extra == 'ignore'` (the default), then all extra passed values are ignored.\n",
      " |          Because no validation is performed with a call to `model_construct()`, having `model_config.extra == 'forbid'` does not result in\n",
      " |          an error if extra values are passed, but they will be ignored.\n",
      " |      \n",
      " |      Args:\n",
      " |          _fields_set: A set of field names that were originally explicitly set during instantiation. If provided,\n",
      " |              this is directly used for the [`model_fields_set`][pydantic.BaseModel.model_fields_set] attribute.\n",
      " |              Otherwise, the field names from the `values` argument will be used.\n",
      " |          values: Trusted or pre-validated data dictionary.\n",
      " |      \n",
      " |      Returns:\n",
      " |          A new instance of the `Model` class with validated data.\n",
      " |  \n",
      " |  model_json_schema(by_alias: 'bool' = True, ref_template: 'str' = '#/$defs/{model}', schema_generator: 'type[GenerateJsonSchema]' = <class 'pydantic.json_schema.GenerateJsonSchema'>, mode: 'JsonSchemaMode' = 'validation', *, union_format: \"Literal['any_of', 'primitive_type_array']\" = 'any_of') -> 'dict[str, Any]' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      Generates a JSON schema for a model class.\n",
      " |      \n",
      " |      Args:\n",
      " |          by_alias: Whether to use attribute aliases or not.\n",
      " |          ref_template: The reference template.\n",
      " |          union_format: The format to use when combining schemas from unions together. Can be one of:\n",
      " |      \n",
      " |              - `'any_of'`: Use the [`anyOf`](https://json-schema.org/understanding-json-schema/reference/combining#anyOf)\n",
      " |              keyword to combine schemas (the default).\n",
      " |              - `'primitive_type_array'`: Use the [`type`](https://json-schema.org/understanding-json-schema/reference/type)\n",
      " |              keyword as an array of strings, containing each type of the combination. If any of the schemas is not a primitive\n",
      " |              type (`string`, `boolean`, `null`, `integer` or `number`) or contains constraints/metadata, falls back to\n",
      " |              `any_of`.\n",
      " |          schema_generator: To override the logic used to generate the JSON schema, as a subclass of\n",
      " |              `GenerateJsonSchema` with your desired modifications\n",
      " |          mode: The mode in which to generate the schema.\n",
      " |      \n",
      " |      Returns:\n",
      " |          The JSON schema for the given model class.\n",
      " |  \n",
      " |  model_parametrized_name(params: 'tuple[type[Any], ...]') -> 'str' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      Compute the class name for parametrizations of generic classes.\n",
      " |      \n",
      " |      This method can be overridden to achieve a custom naming scheme for generic BaseModels.\n",
      " |      \n",
      " |      Args:\n",
      " |          params: Tuple of types of the class. Given a generic class\n",
      " |              `Model` with 2 type variables and a concrete model `Model[str, int]`,\n",
      " |              the value `(str, int)` would be passed to `params`.\n",
      " |      \n",
      " |      Returns:\n",
      " |          String representing the new class where `params` are passed to `cls` as type variables.\n",
      " |      \n",
      " |      Raises:\n",
      " |          TypeError: Raised when trying to generate concrete names for non-generic models.\n",
      " |  \n",
      " |  model_rebuild(*, force: 'bool' = False, raise_errors: 'bool' = True, _parent_namespace_depth: 'int' = 2, _types_namespace: 'MappingNamespace | None' = None) -> 'bool | None' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      Try to rebuild the pydantic-core schema for the model.\n",
      " |      \n",
      " |      This may be necessary when one of the annotations is a ForwardRef which could not be resolved during\n",
      " |      the initial attempt to build the schema, and automatic rebuilding fails.\n",
      " |      \n",
      " |      Args:\n",
      " |          force: Whether to force the rebuilding of the model schema, defaults to `False`.\n",
      " |          raise_errors: Whether to raise errors, defaults to `True`.\n",
      " |          _parent_namespace_depth: The depth level of the parent namespace, defaults to 2.\n",
      " |          _types_namespace: The types namespace, defaults to `None`.\n",
      " |      \n",
      " |      Returns:\n",
      " |          Returns `None` if the schema is already \"complete\" and rebuilding was not required.\n",
      " |          If rebuilding _was_ required, returns `True` if rebuilding was successful, otherwise `False`.\n",
      " |  \n",
      " |  model_validate(obj: 'Any', *, strict: 'bool | None' = None, extra: 'ExtraValues | None' = None, from_attributes: 'bool | None' = None, context: 'Any | None' = None, by_alias: 'bool | None' = None, by_name: 'bool | None' = None) -> 'Self' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      Validate a pydantic model instance.\n",
      " |      \n",
      " |      Args:\n",
      " |          obj: The object to validate.\n",
      " |          strict: Whether to enforce types strictly.\n",
      " |          extra: Whether to ignore, allow, or forbid extra data during model validation.\n",
      " |              See the [`extra` configuration value][pydantic.ConfigDict.extra] for details.\n",
      " |          from_attributes: Whether to extract data from object attributes.\n",
      " |          context: Additional context to pass to the validator.\n",
      " |          by_alias: Whether to use the field's alias when validating against the provided input data.\n",
      " |          by_name: Whether to use the field's name when validating against the provided input data.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValidationError: If the object could not be validated.\n",
      " |      \n",
      " |      Returns:\n",
      " |          The validated model instance.\n",
      " |  \n",
      " |  model_validate_json(json_data: 'str | bytes | bytearray', *, strict: 'bool | None' = None, extra: 'ExtraValues | None' = None, context: 'Any | None' = None, by_alias: 'bool | None' = None, by_name: 'bool | None' = None) -> 'Self' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      !!! abstract \"Usage Documentation\"\n",
      " |          [JSON Parsing](../concepts/json.md#json-parsing)\n",
      " |      \n",
      " |      Validate the given JSON data against the Pydantic model.\n",
      " |      \n",
      " |      Args:\n",
      " |          json_data: The JSON data to validate.\n",
      " |          strict: Whether to enforce types strictly.\n",
      " |          extra: Whether to ignore, allow, or forbid extra data during model validation.\n",
      " |              See the [`extra` configuration value][pydantic.ConfigDict.extra] for details.\n",
      " |          context: Extra variables to pass to the validator.\n",
      " |          by_alias: Whether to use the field's alias when validating against the provided input data.\n",
      " |          by_name: Whether to use the field's name when validating against the provided input data.\n",
      " |      \n",
      " |      Returns:\n",
      " |          The validated Pydantic model.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValidationError: If `json_data` is not a JSON string or the object could not be validated.\n",
      " |  \n",
      " |  model_validate_strings(obj: 'Any', *, strict: 'bool | None' = None, extra: 'ExtraValues | None' = None, context: 'Any | None' = None, by_alias: 'bool | None' = None, by_name: 'bool | None' = None) -> 'Self' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |      Validate the given object with string data against the Pydantic model.\n",
      " |      \n",
      " |      Args:\n",
      " |          obj: The object containing string data to validate.\n",
      " |          strict: Whether to enforce types strictly.\n",
      " |          extra: Whether to ignore, allow, or forbid extra data during model validation.\n",
      " |              See the [`extra` configuration value][pydantic.ConfigDict.extra] for details.\n",
      " |          context: Extra variables to pass to the validator.\n",
      " |          by_alias: Whether to use the field's alias when validating against the provided input data.\n",
      " |          by_name: Whether to use the field's name when validating against the provided input data.\n",
      " |      \n",
      " |      Returns:\n",
      " |          The validated Pydantic model.\n",
      " |  \n",
      " |  parse_file(path: 'str | Path', *, content_type: 'str | None' = None, encoding: 'str' = 'utf8', proto: 'DeprecatedParseProtocol | None' = None, allow_pickle: 'bool' = False) -> 'Self' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  parse_obj(obj: 'Any') -> 'Self' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  parse_raw(b: 'str | bytes', *, content_type: 'str | None' = None, encoding: 'str' = 'utf8', proto: 'DeprecatedParseProtocol | None' = None, allow_pickle: 'bool' = False) -> 'Self' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  schema(by_alias: 'bool' = True, ref_template: 'str' = '#/$defs/{model}') -> 'Dict[str, Any]' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  schema_json(*, by_alias: 'bool' = True, ref_template: 'str' = '#/$defs/{model}', **dumps_kwargs: 'Any') -> 'str' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  update_forward_refs(**localns: 'Any') -> 'None' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  validate(value: 'Any') -> 'Self' from pydantic._internal._model_construction.ModelMetaclass\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Readonly properties inherited from pydantic.main.BaseModel:\n",
      " |  \n",
      " |  __fields_set__\n",
      " |  \n",
      " |  model_extra\n",
      " |      Get extra fields set during validation.\n",
      " |      \n",
      " |      Returns:\n",
      " |          A dictionary of extra fields, or `None` if `config.extra` is not set to `\"allow\"`.\n",
      " |  \n",
      " |  model_fields_set\n",
      " |      Returns the set of fields that have been explicitly set on this model instance.\n",
      " |      \n",
      " |      Returns:\n",
      " |          A set of strings representing the fields that have been set,\n",
      " |              i.e. that were not filled from defaults.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors inherited from pydantic.main.BaseModel:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __pydantic_extra__\n",
      " |  \n",
      " |  __pydantic_fields_set__\n",
      " |  \n",
      " |  __pydantic_private__\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data and other attributes inherited from pydantic.main.BaseModel:\n",
      " |  \n",
      " |  __hash__ = None\n",
      " |  \n",
      " |  __pydantic_root_model__ = False\n",
      " |  \n",
      " |  model_computed_fields = {}\n",
      " |  \n",
      " |  model_fields = {'id': FieldInfo(annotation=Union[str, NoneType], requi...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(results[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d9d596b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on FAISS in module langchain_community.vectorstores.faiss object:\n",
      "\n",
      "class FAISS(langchain_core.vectorstores.base.VectorStore)\n",
      " |  FAISS(embedding_function: 'Union[Callable[[str], List[float]], Embeddings]', index: 'Any', docstore: 'Docstore', index_to_docstore_id: 'Dict[int, str]', relevance_score_fn: 'Optional[Callable[[float], float]]' = None, normalize_L2: 'bool' = False, distance_strategy: 'DistanceStrategy' = <DistanceStrategy.EUCLIDEAN_DISTANCE: 'EUCLIDEAN_DISTANCE'>)\n",
      " |  \n",
      " |  FAISS vector store integration.\n",
      " |  \n",
      " |  See [The FAISS Library](https://arxiv.org/pdf/2401.08281) paper.\n",
      " |  \n",
      " |  Setup:\n",
      " |      Install ``langchain_community`` and ``faiss-cpu`` python packages.\n",
      " |  \n",
      " |      .. code-block:: bash\n",
      " |  \n",
      " |          pip install -qU langchain_community faiss-cpu\n",
      " |  \n",
      " |  Key init args — indexing params:\n",
      " |      embedding_function: Embeddings\n",
      " |          Embedding function to use.\n",
      " |  \n",
      " |  Key init args — client params:\n",
      " |      index: Any\n",
      " |          FAISS index to use.\n",
      " |      docstore: Docstore\n",
      " |          Docstore to use.\n",
      " |      index_to_docstore_id: Dict[int, str]\n",
      " |          Mapping of index to docstore id.\n",
      " |  \n",
      " |  Instantiate:\n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          import faiss\n",
      " |          from langchain_community.vectorstores import FAISS\n",
      " |          from langchain_community.docstore.in_memory import InMemoryDocstore\n",
      " |          from langchain_openai import OpenAIEmbeddings\n",
      " |  \n",
      " |          index = faiss.IndexFlatL2(len(OpenAIEmbeddings().embed_query(\"hello world\")))\n",
      " |  \n",
      " |          vector_store = FAISS(\n",
      " |              embedding_function=OpenAIEmbeddings(),\n",
      " |              index=index,\n",
      " |              docstore= InMemoryDocstore(),\n",
      " |              index_to_docstore_id={}\n",
      " |          )\n",
      " |  \n",
      " |  Add Documents:\n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          from langchain_core.documents import Document\n",
      " |  \n",
      " |          document_1 = Document(page_content=\"foo\", metadata={\"baz\": \"bar\"})\n",
      " |          document_2 = Document(page_content=\"thud\", metadata={\"bar\": \"baz\"})\n",
      " |          document_3 = Document(page_content=\"i will be deleted :(\")\n",
      " |  \n",
      " |          documents = [document_1, document_2, document_3]\n",
      " |          ids = [\"1\", \"2\", \"3\"]\n",
      " |          vector_store.add_documents(documents=documents, ids=ids)\n",
      " |  \n",
      " |  Delete Documents:\n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          vector_store.delete(ids=[\"3\"])\n",
      " |  \n",
      " |  Search:\n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          results = vector_store.similarity_search(query=\"thud\",k=1)\n",
      " |          for doc in results:\n",
      " |              print(f\"* {doc.page_content} [{doc.metadata}]\")\n",
      " |  \n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          * thud [{'bar': 'baz'}]\n",
      " |  \n",
      " |  Search with filter:\n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          results = vector_store.similarity_search(query=\"thud\",k=1,filter={\"bar\": \"baz\"})\n",
      " |          for doc in results:\n",
      " |              print(f\"* {doc.page_content} [{doc.metadata}]\")\n",
      " |  \n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          * thud [{'bar': 'baz'}]\n",
      " |  \n",
      " |  Search with score:\n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          results = vector_store.similarity_search_with_score(query=\"qux\",k=1)\n",
      " |          for doc, score in results:\n",
      " |              print(f\"* [SIM={score:3f}] {doc.page_content} [{doc.metadata}]\")\n",
      " |  \n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          * [SIM=0.335304] foo [{'baz': 'bar'}]\n",
      " |  \n",
      " |  Async:\n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          # add documents\n",
      " |          # await vector_store.aadd_documents(documents=documents, ids=ids)\n",
      " |  \n",
      " |          # delete documents\n",
      " |          # await vector_store.adelete(ids=[\"3\"])\n",
      " |  \n",
      " |          # search\n",
      " |          # results = vector_store.asimilarity_search(query=\"thud\",k=1)\n",
      " |  \n",
      " |          # search with score\n",
      " |          results = await vector_store.asimilarity_search_with_score(query=\"qux\",k=1)\n",
      " |          for doc,score in results:\n",
      " |              print(f\"* [SIM={score:3f}] {doc.page_content} [{doc.metadata}]\")\n",
      " |  \n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          * [SIM=0.335304] foo [{'baz': 'bar'}]\n",
      " |  \n",
      " |  Use as Retriever:\n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          retriever = vector_store.as_retriever(\n",
      " |              search_type=\"mmr\",\n",
      " |              search_kwargs={\"k\": 1, \"fetch_k\": 2, \"lambda_mult\": 0.5},\n",
      " |          )\n",
      " |          retriever.invoke(\"thud\")\n",
      " |  \n",
      " |      .. code-block:: python\n",
      " |  \n",
      " |          [Document(metadata={'bar': 'baz'}, page_content='thud')]\n",
      " |  \n",
      " |  Method resolution order:\n",
      " |      FAISS\n",
      " |      langchain_core.vectorstores.base.VectorStore\n",
      " |      abc.ABC\n",
      " |      builtins.object\n",
      " |  \n",
      " |  Methods defined here:\n",
      " |  \n",
      " |  __init__(self, embedding_function: 'Union[Callable[[str], List[float]], Embeddings]', index: 'Any', docstore: 'Docstore', index_to_docstore_id: 'Dict[int, str]', relevance_score_fn: 'Optional[Callable[[float], float]]' = None, normalize_L2: 'bool' = False, distance_strategy: 'DistanceStrategy' = <DistanceStrategy.EUCLIDEAN_DISTANCE: 'EUCLIDEAN_DISTANCE'>)\n",
      " |      Initialize with necessary components.\n",
      " |  \n",
      " |  async aadd_texts(self, texts: 'Iterable[str]', metadatas: 'Optional[List[dict]]' = None, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'List[str]'\n",
      " |      Run more texts through the embeddings and add to the vectorstore\n",
      " |          asynchronously.\n",
      " |      \n",
      " |      Args:\n",
      " |          texts: Iterable of strings to add to the vectorstore.\n",
      " |          metadatas: Optional list of metadatas associated with the texts.\n",
      " |          ids: Optional list of unique IDs.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of ids from adding the texts into the vectorstore.\n",
      " |  \n",
      " |  add_embeddings(self, text_embeddings: 'Iterable[Tuple[str, List[float]]]', metadatas: 'Optional[List[dict]]' = None, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'List[str]'\n",
      " |      Add the given texts and embeddings to the vectorstore.\n",
      " |      \n",
      " |      Args:\n",
      " |          text_embeddings: Iterable pairs of string and embedding to\n",
      " |              add to the vectorstore.\n",
      " |          metadatas: Optional list of metadatas associated with the texts.\n",
      " |          ids: Optional list of unique IDs.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of ids from adding the texts into the vectorstore.\n",
      " |  \n",
      " |  add_texts(self, texts: 'Iterable[str]', metadatas: 'Optional[List[dict]]' = None, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'List[str]'\n",
      " |      Run more texts through the embeddings and add to the vectorstore.\n",
      " |      \n",
      " |      Args:\n",
      " |          texts: Iterable of strings to add to the vectorstore.\n",
      " |          metadatas: Optional list of metadatas associated with the texts.\n",
      " |          ids: Optional list of unique IDs.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of ids from adding the texts into the vectorstore.\n",
      " |  \n",
      " |  async amax_marginal_relevance_search(self, query: 'str', k: 'int' = 4, fetch_k: 'int' = 20, lambda_mult: 'float' = 0.5, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs selected using the maximal marginal relevance asynchronously.\n",
      " |      \n",
      " |      Maximal marginal relevance optimizes for similarity to query AND diversity\n",
      " |      among selected documents.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Text to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          fetch_k: Number of Documents to fetch before filtering (if needed) to\n",
      " |                   pass to MMR algorithm.\n",
      " |          lambda_mult: Number between 0 and 1 that determines the degree\n",
      " |                      of diversity among the results with 0 corresponding\n",
      " |                      to maximum diversity and 1 to minimum diversity.\n",
      " |                      Defaults to 0.5.\n",
      " |      Returns:\n",
      " |          List of Documents selected by maximal marginal relevance.\n",
      " |  \n",
      " |  async amax_marginal_relevance_search_by_vector(self, embedding: 'List[float]', k: 'int' = 4, fetch_k: 'int' = 20, lambda_mult: 'float' = 0.5, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs selected using the maximal marginal relevance asynchronously.\n",
      " |      \n",
      " |      Maximal marginal relevance optimizes for similarity to query AND diversity\n",
      " |      among selected documents.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          fetch_k: Number of Documents to fetch before filtering to\n",
      " |                   pass to MMR algorithm.\n",
      " |          lambda_mult: Number between 0 and 1 that determines the degree\n",
      " |                      of diversity among the results with 0 corresponding\n",
      " |                      to maximum diversity and 1 to minimum diversity.\n",
      " |                      Defaults to 0.5.\n",
      " |      Returns:\n",
      " |          List of Documents selected by maximal marginal relevance.\n",
      " |  \n",
      " |  async amax_marginal_relevance_search_with_score_by_vector(self, embedding: 'List[float]', *, k: 'int' = 4, fetch_k: 'int' = 20, lambda_mult: 'float' = 0.5, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None) -> 'List[Tuple[Document, float]]'\n",
      " |      Return docs and their similarity scores selected using the maximal marginal\n",
      " |          relevance asynchronously.\n",
      " |      \n",
      " |      Maximal marginal relevance optimizes for similarity to query AND diversity\n",
      " |      among selected documents.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          fetch_k: Number of Documents to fetch before filtering to\n",
      " |                   pass to MMR algorithm.\n",
      " |          lambda_mult: Number between 0 and 1 that determines the degree\n",
      " |                      of diversity among the results with 0 corresponding\n",
      " |                      to maximum diversity and 1 to minimum diversity.\n",
      " |                      Defaults to 0.5.\n",
      " |      Returns:\n",
      " |          List of Documents and similarity scores selected by maximal marginal\n",
      " |              relevance and score for each.\n",
      " |  \n",
      " |  async asimilarity_search(self, query: 'str', k: 'int' = 4, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, fetch_k: 'int' = 20, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs most similar to query asynchronously.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Text to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          filter: (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.\n",
      " |          fetch_k: (Optional[int]) Number of Documents to fetch before filtering.\n",
      " |                    Defaults to 20.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents most similar to the query.\n",
      " |  \n",
      " |  async asimilarity_search_by_vector(self, embedding: 'List[float]', k: 'int' = 4, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, fetch_k: 'int' = 20, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs most similar to embedding vector asynchronously.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          filter (Optional[Dict[str, str]]): Filter by metadata.\n",
      " |              Defaults to None. If a callable, it must take as input the\n",
      " |              metadata dict of Document and return a bool.\n",
      " |      \n",
      " |          fetch_k: (Optional[int]) Number of Documents to fetch before filtering.\n",
      " |                    Defaults to 20.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents most similar to the embedding.\n",
      " |  \n",
      " |  async asimilarity_search_with_score(self, query: 'str', k: 'int' = 4, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, fetch_k: 'int' = 20, **kwargs: 'Any') -> 'List[Tuple[Document, float]]'\n",
      " |      Return docs most similar to query asynchronously.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Text to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          filter (Optional[Dict[str, str]]): Filter by metadata.\n",
      " |              Defaults to None. If a callable, it must take as input the\n",
      " |              metadata dict of Document and return a bool.\n",
      " |      \n",
      " |          fetch_k: (Optional[int]) Number of Documents to fetch before filtering.\n",
      " |                    Defaults to 20.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of documents most similar to the query text with\n",
      " |          L2 distance in float. Lower score represents more similarity.\n",
      " |  \n",
      " |  async asimilarity_search_with_score_by_vector(self, embedding: 'List[float]', k: 'int' = 4, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, fetch_k: 'int' = 20, **kwargs: 'Any') -> 'List[Tuple[Document, float]]'\n",
      " |      Return docs most similar to query asynchronously.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding vector to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          filter (Optional[Dict[str, Any]]): Filter by metadata.\n",
      " |              Defaults to None. If a callable, it must take as input the\n",
      " |              metadata dict of Document and return a bool.\n",
      " |      \n",
      " |          fetch_k: (Optional[int]) Number of Documents to fetch before filtering.\n",
      " |                    Defaults to 20.\n",
      " |          **kwargs: kwargs to be passed to similarity search. Can include:\n",
      " |              score_threshold: Optional, a floating point value between 0 to 1 to\n",
      " |                  filter the resulting set of retrieved docs\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of documents most similar to the query text and L2 distance\n",
      " |          in float for each. Lower score represents more similarity.\n",
      " |  \n",
      " |  delete(self, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'Optional[bool]'\n",
      " |      Delete by ID. These are the IDs in the vectorstore.\n",
      " |      \n",
      " |      Args:\n",
      " |          ids: List of ids to delete.\n",
      " |      \n",
      " |      Returns:\n",
      " |          Optional[bool]: True if deletion is successful,\n",
      " |          False otherwise, None if not implemented.\n",
      " |  \n",
      " |  get_by_ids(self, ids: 'Sequence[str]', /) -> 'list[Document]'\n",
      " |      Get documents by their IDs.\n",
      " |      \n",
      " |      The returned documents are expected to have the ID field set to the ID of the\n",
      " |      document in the vector store.\n",
      " |      \n",
      " |      Fewer documents may be returned than requested if some IDs are not found or\n",
      " |      if there are duplicated IDs.\n",
      " |      \n",
      " |      Users should not assume that the order of the returned documents matches\n",
      " |      the order of the input IDs. Instead, users should rely on the ID field of the\n",
      " |      returned documents.\n",
      " |      \n",
      " |      This method should **NOT** raise exceptions if no documents are found for\n",
      " |      some IDs.\n",
      " |      \n",
      " |      Args:\n",
      " |          ids: List of IDs to retrieve.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of `Document` objects.\n",
      " |  \n",
      " |  max_marginal_relevance_search(self, query: 'str', k: 'int' = 4, fetch_k: 'int' = 20, lambda_mult: 'float' = 0.5, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs selected using the maximal marginal relevance.\n",
      " |      \n",
      " |      Maximal marginal relevance optimizes for similarity to query AND diversity\n",
      " |      among selected documents.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Text to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          fetch_k: Number of Documents to fetch before filtering (if needed) to\n",
      " |                   pass to MMR algorithm.\n",
      " |          lambda_mult: Number between 0 and 1 that determines the degree\n",
      " |                      of diversity among the results with 0 corresponding\n",
      " |                      to maximum diversity and 1 to minimum diversity.\n",
      " |                      Defaults to 0.5.\n",
      " |      Returns:\n",
      " |          List of Documents selected by maximal marginal relevance.\n",
      " |  \n",
      " |  max_marginal_relevance_search_by_vector(self, embedding: 'List[float]', k: 'int' = 4, fetch_k: 'int' = 20, lambda_mult: 'float' = 0.5, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs selected using the maximal marginal relevance.\n",
      " |      \n",
      " |      Maximal marginal relevance optimizes for similarity to query AND diversity\n",
      " |      among selected documents.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          fetch_k: Number of Documents to fetch before filtering to\n",
      " |                   pass to MMR algorithm.\n",
      " |          lambda_mult: Number between 0 and 1 that determines the degree\n",
      " |                      of diversity among the results with 0 corresponding\n",
      " |                      to maximum diversity and 1 to minimum diversity.\n",
      " |                      Defaults to 0.5.\n",
      " |      Returns:\n",
      " |          List of Documents selected by maximal marginal relevance.\n",
      " |  \n",
      " |  max_marginal_relevance_search_with_score_by_vector(self, embedding: 'List[float]', *, k: 'int' = 4, fetch_k: 'int' = 20, lambda_mult: 'float' = 0.5, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None) -> 'List[Tuple[Document, float]]'\n",
      " |      Return docs and their similarity scores selected using the maximal marginal\n",
      " |          relevance.\n",
      " |      \n",
      " |      Maximal marginal relevance optimizes for similarity to query AND diversity\n",
      " |      among selected documents.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          fetch_k: Number of Documents to fetch before filtering to\n",
      " |                   pass to MMR algorithm.\n",
      " |          lambda_mult: Number between 0 and 1 that determines the degree\n",
      " |                      of diversity among the results with 0 corresponding\n",
      " |                      to maximum diversity and 1 to minimum diversity.\n",
      " |                      Defaults to 0.5.\n",
      " |      Returns:\n",
      " |          List of Documents and similarity scores selected by maximal marginal\n",
      " |              relevance and score for each.\n",
      " |  \n",
      " |  merge_from(self, target: 'FAISS') -> 'None'\n",
      " |      Merge another FAISS object with the current one.\n",
      " |      \n",
      " |      Add the target FAISS to the current one.\n",
      " |      \n",
      " |      Args:\n",
      " |          target: FAISS object you wish to merge into the current one\n",
      " |      \n",
      " |      Returns:\n",
      " |          None.\n",
      " |  \n",
      " |  save_local(self, folder_path: 'str', index_name: 'str' = 'index') -> 'None'\n",
      " |      Save FAISS index, docstore, and index_to_docstore_id to disk.\n",
      " |      \n",
      " |      Args:\n",
      " |          folder_path: folder path to save index, docstore,\n",
      " |              and index_to_docstore_id to.\n",
      " |          index_name: for saving with a specific index file name\n",
      " |  \n",
      " |  serialize_to_bytes(self) -> 'bytes'\n",
      " |      Serialize FAISS index, docstore, and index_to_docstore_id to bytes.\n",
      " |  \n",
      " |  similarity_search(self, query: 'str', k: 'int' = 4, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, fetch_k: 'int' = 20, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs most similar to query.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Text to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          filter: (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.\n",
      " |          fetch_k: (Optional[int]) Number of Documents to fetch before filtering.\n",
      " |                    Defaults to 20.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents most similar to the query.\n",
      " |  \n",
      " |  similarity_search_by_vector(self, embedding: 'List[float]', k: 'int' = 4, filter: 'Optional[Dict[str, Any]]' = None, fetch_k: 'int' = 20, **kwargs: 'Any') -> 'List[Document]'\n",
      " |      Return docs most similar to embedding vector.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          filter (Optional[Dict[str, str]]): Filter by metadata.\n",
      " |              Defaults to None. If a callable, it must take as input the\n",
      " |              metadata dict of Document and return a bool.\n",
      " |      \n",
      " |          fetch_k: (Optional[int]) Number of Documents to fetch before filtering.\n",
      " |                    Defaults to 20.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of Documents most similar to the embedding.\n",
      " |  \n",
      " |  similarity_search_with_score(self, query: 'str', k: 'int' = 4, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, fetch_k: 'int' = 20, **kwargs: 'Any') -> 'List[Tuple[Document, float]]'\n",
      " |      Return docs most similar to query.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Text to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          filter (Optional[Dict[str, str]]): Filter by metadata.\n",
      " |              Defaults to None. If a callable, it must take as input the\n",
      " |              metadata dict of Document and return a bool.\n",
      " |      \n",
      " |          fetch_k: (Optional[int]) Number of Documents to fetch before filtering.\n",
      " |                    Defaults to 20.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of documents most similar to the query text with\n",
      " |          L2 distance in float. Lower score represents more similarity.\n",
      " |  \n",
      " |  similarity_search_with_score_by_vector(self, embedding: 'List[float]', k: 'int' = 4, filter: 'Optional[Union[Callable, Dict[str, Any]]]' = None, fetch_k: 'int' = 20, **kwargs: 'Any') -> 'List[Tuple[Document, float]]'\n",
      " |      Return docs most similar to query.\n",
      " |      \n",
      " |      Args:\n",
      " |          embedding: Embedding vector to look up documents similar to.\n",
      " |          k: Number of Documents to return. Defaults to 4.\n",
      " |          filter (Optional[Union[Callable, Dict[str, Any]]]): Filter by metadata.\n",
      " |              Defaults to None. If a callable, it must take as input the\n",
      " |              metadata dict of Document and return a bool.\n",
      " |          fetch_k: (Optional[int]) Number of Documents to fetch before filtering.\n",
      " |                    Defaults to 20.\n",
      " |          **kwargs: kwargs to be passed to similarity search. Can include:\n",
      " |              score_threshold: Optional, a floating point value between 0 to 1 to\n",
      " |                  filter the resulting set of retrieved docs\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of documents most similar to the query text and L2 distance\n",
      " |          in float for each. Lower score represents more similarity.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Class methods defined here:\n",
      " |  \n",
      " |  async afrom_embeddings(text_embeddings: 'Iterable[Tuple[str, List[float]]]', embedding: 'Embeddings', metadatas: 'Optional[Iterable[dict]]' = None, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'FAISS' from abc.ABCMeta\n",
      " |      Construct FAISS wrapper from raw documents asynchronously.\n",
      " |  \n",
      " |  async afrom_texts(texts: 'list[str]', embedding: 'Embeddings', metadatas: 'Optional[List[dict]]' = None, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'FAISS' from abc.ABCMeta\n",
      " |      Construct FAISS wrapper from raw documents asynchronously.\n",
      " |      \n",
      " |      This is a user friendly interface that:\n",
      " |          1. Embeds documents.\n",
      " |          2. Creates an in memory docstore\n",
      " |          3. Initializes the FAISS database\n",
      " |      \n",
      " |      This is intended to be a quick way to get started.\n",
      " |      \n",
      " |      Example:\n",
      " |          .. code-block:: python\n",
      " |      \n",
      " |              from langchain_community.vectorstores import FAISS\n",
      " |              from langchain_community.embeddings import OpenAIEmbeddings\n",
      " |      \n",
      " |              embeddings = OpenAIEmbeddings()\n",
      " |              faiss = await FAISS.afrom_texts(texts, embeddings)\n",
      " |  \n",
      " |  deserialize_from_bytes(serialized: 'bytes', embeddings: 'Embeddings', *, allow_dangerous_deserialization: 'bool' = False, **kwargs: 'Any') -> 'FAISS' from abc.ABCMeta\n",
      " |      Deserialize FAISS index, docstore, and index_to_docstore_id from bytes.\n",
      " |  \n",
      " |  from_embeddings(text_embeddings: 'Iterable[Tuple[str, List[float]]]', embedding: 'Embeddings', metadatas: 'Optional[Iterable[dict]]' = None, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'FAISS' from abc.ABCMeta\n",
      " |      Construct FAISS wrapper from raw documents.\n",
      " |      \n",
      " |      This is a user friendly interface that:\n",
      " |          1. Embeds documents.\n",
      " |          2. Creates an in memory docstore\n",
      " |          3. Initializes the FAISS database\n",
      " |      \n",
      " |      This is intended to be a quick way to get started.\n",
      " |      \n",
      " |      Example:\n",
      " |          .. code-block:: python\n",
      " |      \n",
      " |              from langchain_community.vectorstores import FAISS\n",
      " |              from langchain_community.embeddings import OpenAIEmbeddings\n",
      " |      \n",
      " |              embeddings = OpenAIEmbeddings()\n",
      " |              text_embeddings = embeddings.embed_documents(texts)\n",
      " |              text_embedding_pairs = zip(texts, text_embeddings)\n",
      " |              faiss = FAISS.from_embeddings(text_embedding_pairs, embeddings)\n",
      " |  \n",
      " |  from_texts(texts: 'List[str]', embedding: 'Embeddings', metadatas: 'Optional[List[dict]]' = None, ids: 'Optional[List[str]]' = None, **kwargs: 'Any') -> 'FAISS' from abc.ABCMeta\n",
      " |      Construct FAISS wrapper from raw documents.\n",
      " |      \n",
      " |      This is a user friendly interface that:\n",
      " |          1. Embeds documents.\n",
      " |          2. Creates an in memory docstore\n",
      " |          3. Initializes the FAISS database\n",
      " |      \n",
      " |      This is intended to be a quick way to get started.\n",
      " |      \n",
      " |      Example:\n",
      " |          .. code-block:: python\n",
      " |      \n",
      " |              from langchain_community.vectorstores import FAISS\n",
      " |              from langchain_community.embeddings import OpenAIEmbeddings\n",
      " |      \n",
      " |              embeddings = OpenAIEmbeddings()\n",
      " |              faiss = FAISS.from_texts(texts, embeddings)\n",
      " |  \n",
      " |  load_local(folder_path: 'str', embeddings: 'Embeddings', index_name: 'str' = 'index', *, allow_dangerous_deserialization: 'bool' = False, **kwargs: 'Any') -> 'FAISS' from abc.ABCMeta\n",
      " |      Load FAISS index, docstore, and index_to_docstore_id from disk.\n",
      " |      \n",
      " |      Args:\n",
      " |          folder_path: folder path to load index, docstore,\n",
      " |              and index_to_docstore_id from.\n",
      " |          embeddings: Embeddings to use when generating queries\n",
      " |          index_name: for saving with a specific index file name\n",
      " |          allow_dangerous_deserialization: whether to allow deserialization\n",
      " |              of the data which involves loading a pickle file.\n",
      " |              Pickle files can be modified by malicious actors to deliver a\n",
      " |              malicious payload that results in execution of\n",
      " |              arbitrary code on your machine.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Readonly properties defined here:\n",
      " |  \n",
      " |  embeddings\n",
      " |      Access the query embedding object if available.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data and other attributes defined here:\n",
      " |  \n",
      " |  __abstractmethods__ = frozenset()\n",
      " |  \n",
      " |  __annotations__ = {}\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Methods inherited from langchain_core.vectorstores.base.VectorStore:\n",
      " |  \n",
      " |  async aadd_documents(self, documents: 'list[Document]', **kwargs: 'Any') -> 'list[str]'\n",
      " |      Async run more documents through the embeddings and add to the `VectorStore`.\n",
      " |      \n",
      " |      Args:\n",
      " |          documents: Documents to add to the `VectorStore`.\n",
      " |          **kwargs: Additional keyword arguments.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of IDs of the added texts.\n",
      " |  \n",
      " |  add_documents(self, documents: 'list[Document]', **kwargs: 'Any') -> 'list[str]'\n",
      " |      Add or update documents in the `VectorStore`.\n",
      " |      \n",
      " |      Args:\n",
      " |          documents: Documents to add to the `VectorStore`.\n",
      " |          **kwargs: Additional keyword arguments.\n",
      " |      \n",
      " |              If kwargs contains IDs and documents contain ids, the IDs in the kwargs\n",
      " |              will receive precedence.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of IDs of the added texts.\n",
      " |  \n",
      " |  async adelete(self, ids: 'list[str] | None' = None, **kwargs: 'Any') -> 'bool | None'\n",
      " |      Async delete by vector ID or other criteria.\n",
      " |      \n",
      " |      Args:\n",
      " |          ids: List of IDs to delete. If `None`, delete all.\n",
      " |          **kwargs: Other keyword arguments that subclasses might use.\n",
      " |      \n",
      " |      Returns:\n",
      " |          `True` if deletion is successful, `False` otherwise, `None` if not\n",
      " |              implemented.\n",
      " |  \n",
      " |  async aget_by_ids(self, ids: 'Sequence[str]', /) -> 'list[Document]'\n",
      " |      Async get documents by their IDs.\n",
      " |      \n",
      " |      The returned documents are expected to have the ID field set to the ID of the\n",
      " |      document in the vector store.\n",
      " |      \n",
      " |      Fewer documents may be returned than requested if some IDs are not found or\n",
      " |      if there are duplicated IDs.\n",
      " |      \n",
      " |      Users should not assume that the order of the returned documents matches\n",
      " |      the order of the input IDs. Instead, users should rely on the ID field of the\n",
      " |      returned documents.\n",
      " |      \n",
      " |      This method should **NOT** raise exceptions if no documents are found for\n",
      " |      some IDs.\n",
      " |      \n",
      " |      Args:\n",
      " |          ids: List of IDs to retrieve.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of `Document` objects.\n",
      " |  \n",
      " |  as_retriever(self, **kwargs: 'Any') -> 'VectorStoreRetriever'\n",
      " |      Return `VectorStoreRetriever` initialized from this `VectorStore`.\n",
      " |      \n",
      " |      Args:\n",
      " |          **kwargs: Keyword arguments to pass to the search function.\n",
      " |              Can include:\n",
      " |      \n",
      " |              * `search_type`: Defines the type of search that the Retriever should\n",
      " |                  perform. Can be `'similarity'` (default), `'mmr'`, or\n",
      " |                  `'similarity_score_threshold'`.\n",
      " |              * `search_kwargs`: Keyword arguments to pass to the search function. Can\n",
      " |                  include things like:\n",
      " |      \n",
      " |                  * `k`: Amount of documents to return (Default: `4`)\n",
      " |                  * `score_threshold`: Minimum relevance threshold\n",
      " |                      for `similarity_score_threshold`\n",
      " |                  * `fetch_k`: Amount of documents to pass to MMR algorithm\n",
      " |                      (Default: `20`)\n",
      " |                  * `lambda_mult`: Diversity of results returned by MMR;\n",
      " |                      `1` for minimum diversity and 0 for maximum. (Default: `0.5`)\n",
      " |                  * `filter`: Filter by document metadata\n",
      " |      \n",
      " |      Returns:\n",
      " |          Retriever class for `VectorStore`.\n",
      " |      \n",
      " |      Examples:\n",
      " |      ```python\n",
      " |      # Retrieve more documents with higher diversity\n",
      " |      # Useful if your dataset has many similar documents\n",
      " |      docsearch.as_retriever(\n",
      " |          search_type=\"mmr\", search_kwargs={\"k\": 6, \"lambda_mult\": 0.25}\n",
      " |      )\n",
      " |      \n",
      " |      # Fetch more documents for the MMR algorithm to consider\n",
      " |      # But only return the top 5\n",
      " |      docsearch.as_retriever(search_type=\"mmr\", search_kwargs={\"k\": 5, \"fetch_k\": 50})\n",
      " |      \n",
      " |      # Only retrieve documents that have a relevance score\n",
      " |      # Above a certain threshold\n",
      " |      docsearch.as_retriever(\n",
      " |          search_type=\"similarity_score_threshold\",\n",
      " |          search_kwargs={\"score_threshold\": 0.8},\n",
      " |      )\n",
      " |      \n",
      " |      # Only get the single most similar document from the dataset\n",
      " |      docsearch.as_retriever(search_kwargs={\"k\": 1})\n",
      " |      \n",
      " |      # Use a filter to only retrieve documents from a specific paper\n",
      " |      docsearch.as_retriever(\n",
      " |          search_kwargs={\"filter\": {\"paper_title\": \"GPT-4 Technical Report\"}}\n",
      " |      )\n",
      " |      ```\n",
      " |  \n",
      " |  async asearch(self, query: 'str', search_type: 'str', **kwargs: 'Any') -> 'list[Document]'\n",
      " |      Async return docs most similar to query using a specified search type.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Input text.\n",
      " |          search_type: Type of search to perform. Can be `'similarity'`, `'mmr'`, or\n",
      " |              `'similarity_score_threshold'`.\n",
      " |          **kwargs: Arguments to pass to the search method.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of `Document` objects most similar to the query.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValueError: If `search_type` is not one of `'similarity'`,\n",
      " |              `'mmr'`, or `'similarity_score_threshold'`.\n",
      " |  \n",
      " |  async asimilarity_search_with_relevance_scores(self, query: 'str', k: 'int' = 4, **kwargs: 'Any') -> 'list[tuple[Document, float]]'\n",
      " |      Async return docs and relevance scores in the range `[0, 1]`.\n",
      " |      \n",
      " |      `0` is dissimilar, `1` is most similar.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Input text.\n",
      " |          k: Number of `Document` objects to return.\n",
      " |          **kwargs: kwargs to be passed to similarity search. Should include\n",
      " |              `score_threshold`, An optional floating point value between `0` to `1`\n",
      " |              to filter the resulting set of retrieved docs\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of tuples of `(doc, similarity_score)`\n",
      " |  \n",
      " |  search(self, query: 'str', search_type: 'str', **kwargs: 'Any') -> 'list[Document]'\n",
      " |      Return docs most similar to query using a specified search type.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Input text.\n",
      " |          search_type: Type of search to perform. Can be `'similarity'`, `'mmr'`, or\n",
      " |              `'similarity_score_threshold'`.\n",
      " |          **kwargs: Arguments to pass to the search method.\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of `Document` objects most similar to the query.\n",
      " |      \n",
      " |      Raises:\n",
      " |          ValueError: If `search_type` is not one of `'similarity'`,\n",
      " |              `'mmr'`, or `'similarity_score_threshold'`.\n",
      " |  \n",
      " |  similarity_search_with_relevance_scores(self, query: 'str', k: 'int' = 4, **kwargs: 'Any') -> 'list[tuple[Document, float]]'\n",
      " |      Return docs and relevance scores in the range `[0, 1]`.\n",
      " |      \n",
      " |      `0` is dissimilar, `1` is most similar.\n",
      " |      \n",
      " |      Args:\n",
      " |          query: Input text.\n",
      " |          k: Number of `Document` objects to return.\n",
      " |          **kwargs: kwargs to be passed to similarity search. Should include\n",
      " |              `score_threshold`, An optional floating point value between `0` to `1`\n",
      " |              to filter the resulting set of retrieved docs\n",
      " |      \n",
      " |      Returns:\n",
      " |          List of tuples of `(doc, similarity_score)`.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Class methods inherited from langchain_core.vectorstores.base.VectorStore:\n",
      " |  \n",
      " |  async afrom_documents(documents: 'list[Document]', embedding: 'Embeddings', **kwargs: 'Any') -> 'Self' from abc.ABCMeta\n",
      " |      Async return `VectorStore` initialized from documents and embeddings.\n",
      " |      \n",
      " |      Args:\n",
      " |          documents: List of `Document` objects to add to the `VectorStore`.\n",
      " |          embedding: Embedding function to use.\n",
      " |          **kwargs: Additional keyword arguments.\n",
      " |      \n",
      " |      Returns:\n",
      " |          `VectorStore` initialized from documents and embeddings.\n",
      " |  \n",
      " |  from_documents(documents: 'list[Document]', embedding: 'Embeddings', **kwargs: 'Any') -> 'Self' from abc.ABCMeta\n",
      " |      Return `VectorStore` initialized from documents and embeddings.\n",
      " |      \n",
      " |      Args:\n",
      " |          documents: List of `Document` objects to add to the `VectorStore`.\n",
      " |          embedding: Embedding function to use.\n",
      " |          **kwargs: Additional keyword arguments.\n",
      " |      \n",
      " |      Returns:\n",
      " |          `VectorStore` initialized from documents and embeddings.\n",
      " |  \n",
      " |  ----------------------------------------------------------------------\n",
      " |  Data descriptors inherited from langchain_core.vectorstores.base.VectorStore:\n",
      " |  \n",
      " |  __dict__\n",
      " |      dictionary for instance variables (if defined)\n",
      " |  \n",
      " |  __weakref__\n",
      " |      list of weak references to the object (if defined)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(vector_store)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f29cd19d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(Document(id='6987e0fd-61d2-4a49-ad72-2b29cd5978b1', metadata={'source': '27578435', 'start_index': 62298}, page_content='. [PubMed: 19901337] \\n84. Reguera J, Santiago C, Mudgal G, Ordono D, Enjuanes L, Casasnovas JM. Structural bases of \\ncoronavirus attachment to host aminopeptidase N and its inhibition by neutralizing antibodies. \\nPLOS Pathog. 2012; 8:e1002859. [PubMed: 22876187] \\n85. Chen L, Lin YL, Peng G, Li F. Structural basis for multifunctional roles of mammalian aminopep-\\ntidase N. PNAS. 2012; 109:17966–71. [PubMed: 23071329] \\n86. Wong AH, Zhou D, Rini JM. The X-ray crystal structure of human aminopeptidase N reveals a \\nnovel dimer and the basis for peptide processing. J Biol Chem. 2012; 287:36804–13. [PubMed: \\n22932899] \\n87. Tusell SM, Schittone SA, Holmes KV . Mutational analysis of aminopeptidase N, a receptor for \\nseveral group 1 coronaviruses, identifies key determinants of viral host range. J Virol. 2007; \\n81:1261–73. [PubMed: 17093189] \\n88. Peng GQ, Sun DW, Rajashankar KR, Qian ZH, Holmes KV , Li F. Crystal structure of mouse \\ncoronavirus receptor-binding domain complexed with its murine receptor. PNAS. 2011; \\n108:10696–701. [PubMed: 21670291] \\n89. Beauchemin N, Draber P, Dveksler G, Gold P, Gray-Owen S, et al. Redefined nomenclature for \\nmembers of the carcinoembryonic antigen family. Exp Cell Res. 1999; 252:243–49. [PubMed: \\n11501563] \\n90. Tan KM, Zelus BD, Meijers R, Liu JH, Bergelson JM, et al. Crystal structure of murine \\nsCEACAM1a[1,4]: a coronavirus receptor in the CEA family. EMBOJ. 2002; 21:2076–86.\\n91. Wessner DR, Shick PC, Lu JH, Cardellichio CB, Gagneten SE, et al. Mutational analysis of the \\nvirus and monoclonal antibody binding sites in MHVR, the cellular receptor of the murine \\ncoronavirus mouse hepatitis virus strain A59. J Virol. 1998; 72:1941–48. [PubMed: 9499047] \\n92. Thackray LB, Turner BC, Holmes KV . Substitutions of conserved amino acids in the \\nreceptorbinding domain of the spike glycoprotein affect utilization of murine CEACAM1a by the \\nmurine coronavirus MHV-A59. Virology. 2005; 334:98–110. [PubMed: 15749126] \\n93. Leparc-Goffart I, Hingley ST, Chua MM, Jiang XH, Lavi E, Weiss SR. Altered pathogenesis of a \\nmutant of the murine coronavirus MHV-A59 is associated with a Q159L amino acid substitution in \\nthe spike protein. Virology. 1997; 239:1–10. [PubMed: 9426441] \\n94. Tsai JC, Zelus BD, Holmes KV , Weiss SR. The N-terminal domain of the murine coronavirus spike \\nglycoprotein determines the CEACAM1 receptor specificity of the virus strain. J Virol. 2003; \\n77:841–50. [PubMed: 12502800] \\n95. Ohtsuka N, Taguchi F. Mouse susceptibility to mouse hepatitis virus infection is linked to viral \\nreceptor genotype. J Virol. 1997; 71:8860–63. [PubMed: 9343248] \\n96. Ohtsuka N, Yamada YK, Taguchi F. Difference in virus-binding activity of two distinct receptor \\nproteins for mouse hepatitis virus. J Gen Virol. 1996; 77:1683–92. [PubMed: 8760415] \\n97. Hirai A, Ohtsuka N, Ikeda T, Taniguchi R, Blau D, et al. Role of mouse hepatitis virus (MHV) \\nreceptor murine CEACAM1 in the resistance of mice to MHV infection: studies of mice with \\nchimeric mCEACAMla and mCEACAMlb. J Virol. 2010; 84:6654–66. [PubMed: 20410265] \\n98. Seetharaman J, Kanigsberg A, Slaaby R, Leffler H, Barondes SH, Rini JM. X-ray crystal structure \\nof the human galectin-3 carbohydrate recognition domain at 2.1-Åresolution. J Biol Chem. 1998; \\n273:13047–52. [PubMed: 9582341] \\n99. Kunkel F, Herrler G. Structural and functional-analysis of the surface protein of human coronavirus \\nOC43. Virology. 1993; 195:195–202. [PubMed: 8317096] \\nLi Page 17\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript100. Vijgen L, Keyaerts E, Lemey P, Maes P, van Reeth K, et al. Evolutionary history of the closely \\nrelated group 2 coronaviruses: porcine hemagglutinating encephalomyelitis virus, bovine \\ncoronavirus, and human coronavirus OC43. J Virol. 2006; 80:7270–74. [PubMed: 16809333] \\n101. Vijgen L, Keyaerts E, Moes E, Thoelen I, Wollants E, et al. Complete genomic sequence of \\nhuman coronavirus OC43: molecular clock analysis suggests a relatively recent zoonotic \\ncoronavirus transmission event. J Virol. 2005; 79:1595–604. [PubMed: 15650185] \\n102. Klein A, Krishna M, Varki NM, Varki A. 9-O-acetylated sialic acids have widespread but \\nselective expression: analysis using a chimeric dual-function probe derived from influenza C \\nhemagglutinin-esterase. PNAS. 1994; 91:7782–86. [PubMed: 8052660] \\n103. Chen L, Li F. Structural analysis of the evolutionary origins of influenza virus hemagglutinin and \\nother viral lectins. J Virol. 2013; 87:4118–20. [PubMed: 23365425] \\n104. Eckert DM, Kim PS. Mechanisms of viral membrane fusion and its inhibition. Annu Rev \\nBiochem. 2001; 70:777–810. [PubMed: 11395423] \\n105. Skehel JJ, Wiley DC. Receptor binding and membrane fusion in virus entry: the influenza \\nhemagglutinin. Annu Rev Biochem. 2000; 69:531–69. [PubMed: 10966468] \\n106. Klenk HD, Garten W. Host cell proteases controlling virus pathogenicity. Trends Microbiol. 1994; \\n2:39–43. [PubMed: 8162439] \\n107. Wilson IA, Skehel JJ, Wiley DC. Structure of the haemagglutinin membrane glycoprotein of \\ninfluenza virus at 3 Å resolution. Nature. 1981; 289:366–73. [PubMed: 7464906] \\n108. Stein BS, Gowda SD, Lifson JD, Penhallow RC, Bensch KG, Engleman EG. pH-independent \\nHIV entry into CD4-positive T cells via virus envelope fusion to the plasma membrane. Cell. \\n1987; 49:659–68. [PubMed: 3107838] \\n109. White J, Matlin K, Helenius A. Cell fusion by Semliki Forest, influenza, and vesicular stomatitis \\nviruses. J Cell Biol. 1981; 89:674–79. [PubMed: 6265470] \\n110. Mothes W, Boerger AL, Narayan S, Cunningham JM, Young JA. Retroviral entry mediated by \\nreceptor priming and low pH triggering of an envelope glycoprotein. Cell. 2000; 103:679–89. \\n[PubMed: 11106737] \\n111. Belouzard S, Millet JK, Licitra BN, Whittaker GR. Mechanisms of coronavirus cell entry \\nmediated by the viral spike protein. Viruses. 2012; 4:1011–33. [PubMed: 22816037] \\n112. Heald-Sargent T, Gallagher T. Ready, set, fuse! The coronavirus spike protein and acquisition of \\nfusion competence. Viruses. 2012; 4:557–80. [PubMed: 22590686] \\n113. Xu YH, Lou ZY , Liu YW, Pang H, Tien P, et al. Crystal structure of severe acute respiratory \\nsyndrome coronavirus spike protein fusion core. J Biol Chem. 2004; 279:49414–19. [PubMed: \\n15345712] \\n114. Lu L, Liu Q, Zhu Y , Chan KH, Qin L, et al. Structure-based discovery of Middle East respiratory \\nsyndrome coronavirus fusion inhibitor. Nat Commun. 2014; 5:3067. [PubMed: 24473083] \\n115. Zheng Q, Deng Y , Liu J, van der Hoek L, Berkhout B, Lu M. Core structure of S2 from the \\nhuman coronavirus NL63 spike glycoprotein. Biochemistry. 2006; 45:15205–15. [PubMed: \\n17176042] \\n116. Xu Y , Liu Y , Lou Z, Qin L, Li X, et al. Structural basis for coronavirus-mediated membrane \\nfusion. Crystal structure of mouse hepatitis virus spike protein fusion core. J Biol Chem. 2004; \\n279:30514–22. [PubMed: 15123674] \\n117. Duquerroy S, Vigouroux AN, Rottier PJM, Rey FA, Bosch BJ. Central ions and lateral \\nasparagine/glutamine zippers stabilize the post-fusion hairpin conformation of the SARS \\ncoronavirus spike glycoprotein. Virology. 2005; 335:276–85. [PubMed: 15840526] \\n118. Gao J, Lu G, Qi J, Li Y , Wu Y , et al. Structure of the fusion core and inhibition of fusion by a \\nheptad repeat peptide derived from the S protein of Middle East respiratory syndrome \\ncoronavirus. J Virol. 2013; 87:13134–40. [PubMed: 24067982] \\n119. Supekar VM, Bruckmann C, Ingallinella P, Bianchi E, Pessi A, Carfi A. Structure of a \\nproteolytically resistant core from the severe acute respiratory syndrome coronavirus S2 fusion \\nprotein. PNAS. 2004; 101:17958–63. [PubMed: 15604146] \\n120. Millet JK, Whittaker GR. Host cell proteases: critical determinants of coronavirus tropism and \\npathogenesis. Virus Res. 2015; 202:120–34. [PubMed: 25445340] \\nLi Page 18\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript121. Spaan W, Cavanagh D, Horzinek MC. Coronaviruses: structure and genome expression. J Gen \\nVirol. 1988; 69:2939–52. [PubMed: 3058868] \\n122. Matsuyama S, Taguchi F. Receptor-induced conformational changes of murine coronavirus spike \\nprotein. J Virol. 2002; 76:11819–26. [PubMed: 12414924] \\n123. Zelus BD, Schickli JH, Blau DM, Weiss SR, Holmes KV . Conformational changes in the spike \\nglycoprotein of murine coronavirus are induced at 37° C either by soluble murine CEACAM1 \\nreceptors or by pH 8. J Virol. 2003; 77:830–40. [PubMed: 12502799] \\n124. Qiu Z, Hingley ST, Simmons G, Yu C, Das Sarma J, et al. Endosomal proteolysis by cathepsins is \\nnecessary for murine coronavirus mouse hepatitis virus type 2 spike-mediated entry. J Virol. \\n2006; 80:5768–76. [PubMed: 16731916] \\n125. Eifart P, Ludwig K, Bottcher C, de Haan CA, Rottier PJ, et al. Role of endocytosis and low pH in \\nmurine hepatitis virus strain A59 cell entry. J Virol. 2007; 81:10758–68. [PubMed: 17626088] \\n126. Nakagaki K, Nakagaki K, Taguchi F. Receptor-independent spread of a highly neurotropic murine \\ncoronavirus JHMV strain from initially infected microglial cells in mixed neural cultures. J Virol. \\n2005; 79:6102–10. [PubMed: 15857995] \\n127. Gallagher TM, Buchmeier MJ, Perlman S. Cell receptor-independent infection by a neurotropic \\nmurine coronavirus. Virology. 1992; 191:517–22. [PubMed: 1413526] \\n128. Taguchi F, Matsuyama S, Saeki K. Difference in Bgp-independent fusion activity among mouse \\nhepatitis viruses. Arch Virol. 1999; 144:2041–49'),\n",
       "  0.83742017),\n",
       " (Document(id='d05cb720-1f6f-41c2-bdc9-67ae596f353a', metadata={'source': '39034968', 'start_index': -1}, page_content='.Oncoimmunology. 2021;10:1951019.\\n25. Vladimer GI, Snijder B, Krall N, et al. Global survey of\\nthe immunomodulatory potential of common drugs.Nat\\nChem Biol. 2017;13:681–690\\n.\\n26. Zhou P , Shaffer DR, Alvarez Arias DA, et al. In vivo dis-\\ncovery of immunotherapy targets in the tumour micro-\\nenvironment. Nature. 2014;506:52–57.\\n27. Spigel DR, Reynolds C, Waterhouse D, et al. Phase 1/2\\nstudy of the safety and tolerability of nivolumab plus\\ncrizotinib for theﬁrst-line treatment of anaplastic lym-\\nphoma kinase translocation - positive advanced non-\\nsmall cell lung cancer (CheckMate 370). J Thorac\\nOncol. 2018;13:682–688.\\n28. Felip E, de Braud FG, Maur M, et al. Ceritinib plus\\nnivolumab in patients with advanced ALK-rearranged\\nnon-small cell lung cancer: results of an open-label,\\nmulticenter , phase 1b study. J Thorac Oncol .\\n2020;15:392–403.\\n29. Kim DW, Gadgeel S, Gettinger SN, et al. Brief report:\\nsafety and antitumor activity of alectinib plus atezoli-\\nzumab from a phase 1b study in advanced ALK-positive\\nNSCLC. JTO Clin Res Rep. 2022;3:100367.\\n30. Patel SP , Pakkala S, Pennell NA, et al. Phase Ib study of\\ncrizotinib plus pembrolizumab in patients with previ-\\nously untreated advanced non-small cell lung cancer\\nwith ALK translocation. Oncologist. 2020;25:e562 –\\ne1012.\\n31. Brahmer J, Reckamp KL, Baas P , et al. Nivolumab versus\\ndocetaxel in advanced squamous-cell non-small-cell lung\\ncancer. N Engl J Med. 2015;373:123–135.\\n32. Borghaei H, Paz-Ares L, Horn L, et al. Nivolumab versus\\ndocetaxel in advanced nonsquamous non-small-cell lung\\ncancer. N Engl J Med. 2015;373:1627–1639.\\n33. Solomon BJ, Mok T , Kim DW, et al. First-line crizotinib\\nversus chemotherapy in ALK-positive lung cancer.N Engl\\nJ Med. 2014;371:2167–2177.\\n34. Shaw AT , Kim TM, Crinò L, et al. Ceritinib versus chemo-\\ntherapy in patients with ALK-rearranged non-small-cell\\nlung cancer previously given chemotherapy and crizotinib\\n(ASCEND-5): a randomised, controlled, open-label, phase\\n3 trial.Lancet Oncol. 2017;18:874–886.\\n35. Rittmeyer A, Barlesi F , Waterkamp D, et al. Atezolizu-\\nmab versus docetaxel in patients with previously treated\\nnon-small-cell lung cancer (OAK): a phase 3, open-label,\\nmulticentre randomised controlled trial. Lancet.\\n2017;389:255–265.\\n36. Herbst RS, Giaccone G, de Marinis F , et al. Atezolizumab\\nfor ﬁ\\nrst-line treatment of PD-L1-selected patients with\\nNSCLC. N Engl J Med. 2020;383:1328–1339.\\n37. Peters S, Camidge DR, Shaw AT , et al. Alectinib versus\\ncrizotinib in untreated ALK-positive non-small-cell lung\\ncancer. N Engl J Med. 2017;377:829–838.\\n38. Reck M, Rodriguez-Abreu D, Robinson AG, et al. Pem-\\nbrolizumab versus chemotherapy for PD-L1-positive non-\\nsmall-cell lung cancer .N Engl J Med. 2016;375:1823–1833.\\n39. Dempke WCM, Fenchel K. Has programmed cell death\\nligand-1 MET an accomplice in non-small cell lung\\ncancer?-a narrative review. Transl Lung Cancer Res .\\n2021;10:2667–2682.\\n40. Johnson TR, Tan W, Goulet L, et al. Metabolism, excre-\\ntion and pharmacokinetics of [14C]crizotinib following\\noral administration to healthy subjects. Xenobiotica.\\n2015;45:45–59.\\n41. Xu H, O’Gorman M, Boutros T , et al. Evaluation of cri-\\nzotinib absolute bioavailability, the bioequivalence of\\nthree oral formulations, and the effect of food on cri-\\nzotinib pharmacokinetics in healthy subjects. J Clin\\nPharmacol. 2015;55:104–113.\\n42. Gulley JL, Rajan A, Spigel DR, et al. Avelumab for pa-\\ntients with previously treated metastatic or recurrent\\nnon-small-cell lung cancer (JAVELIN Solid T umor): dose-\\nexpansion cohort of a multicentre, open-label, phase\\n1b trial. Lancet Oncol. 2017;18:599–610.\\nJuly 2024 Avelumab + Lorlatinib or Crizotinib in NSCLC 11'),\n",
       "  0.8760837)]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res = vector_store.similarity_search_with_score(\"molecule\",2)\n",
    "res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "4ad1c5f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Document(id='6987e0fd-61d2-4a49-ad72-2b29cd5978b1', metadata={'source': '27578435', 'start_index': 62298}, page_content='. [PubMed: 19901337] \\n84. Reguera J, Santiago C, Mudgal G, Ordono D, Enjuanes L, Casasnovas JM. Structural bases of \\ncoronavirus attachment to host aminopeptidase N and its inhibition by neutralizing antibodies. \\nPLOS Pathog. 2012; 8:e1002859. [PubMed: 22876187] \\n85. Chen L, Lin YL, Peng G, Li F. Structural basis for multifunctional roles of mammalian aminopep-\\ntidase N. PNAS. 2012; 109:17966–71. [PubMed: 23071329] \\n86. Wong AH, Zhou D, Rini JM. The X-ray crystal structure of human aminopeptidase N reveals a \\nnovel dimer and the basis for peptide processing. J Biol Chem. 2012; 287:36804–13. [PubMed: \\n22932899] \\n87. Tusell SM, Schittone SA, Holmes KV . Mutational analysis of aminopeptidase N, a receptor for \\nseveral group 1 coronaviruses, identifies key determinants of viral host range. J Virol. 2007; \\n81:1261–73. [PubMed: 17093189] \\n88. Peng GQ, Sun DW, Rajashankar KR, Qian ZH, Holmes KV , Li F. Crystal structure of mouse \\ncoronavirus receptor-binding domain complexed with its murine receptor. PNAS. 2011; \\n108:10696–701. [PubMed: 21670291] \\n89. Beauchemin N, Draber P, Dveksler G, Gold P, Gray-Owen S, et al. Redefined nomenclature for \\nmembers of the carcinoembryonic antigen family. Exp Cell Res. 1999; 252:243–49. [PubMed: \\n11501563] \\n90. Tan KM, Zelus BD, Meijers R, Liu JH, Bergelson JM, et al. Crystal structure of murine \\nsCEACAM1a[1,4]: a coronavirus receptor in the CEA family. EMBOJ. 2002; 21:2076–86.\\n91. Wessner DR, Shick PC, Lu JH, Cardellichio CB, Gagneten SE, et al. Mutational analysis of the \\nvirus and monoclonal antibody binding sites in MHVR, the cellular receptor of the murine \\ncoronavirus mouse hepatitis virus strain A59. J Virol. 1998; 72:1941–48. [PubMed: 9499047] \\n92. Thackray LB, Turner BC, Holmes KV . Substitutions of conserved amino acids in the \\nreceptorbinding domain of the spike glycoprotein affect utilization of murine CEACAM1a by the \\nmurine coronavirus MHV-A59. Virology. 2005; 334:98–110. [PubMed: 15749126] \\n93. Leparc-Goffart I, Hingley ST, Chua MM, Jiang XH, Lavi E, Weiss SR. Altered pathogenesis of a \\nmutant of the murine coronavirus MHV-A59 is associated with a Q159L amino acid substitution in \\nthe spike protein. Virology. 1997; 239:1–10. [PubMed: 9426441] \\n94. Tsai JC, Zelus BD, Holmes KV , Weiss SR. The N-terminal domain of the murine coronavirus spike \\nglycoprotein determines the CEACAM1 receptor specificity of the virus strain. J Virol. 2003; \\n77:841–50. [PubMed: 12502800] \\n95. Ohtsuka N, Taguchi F. Mouse susceptibility to mouse hepatitis virus infection is linked to viral \\nreceptor genotype. J Virol. 1997; 71:8860–63. [PubMed: 9343248] \\n96. Ohtsuka N, Yamada YK, Taguchi F. Difference in virus-binding activity of two distinct receptor \\nproteins for mouse hepatitis virus. J Gen Virol. 1996; 77:1683–92. [PubMed: 8760415] \\n97. Hirai A, Ohtsuka N, Ikeda T, Taniguchi R, Blau D, et al. Role of mouse hepatitis virus (MHV) \\nreceptor murine CEACAM1 in the resistance of mice to MHV infection: studies of mice with \\nchimeric mCEACAMla and mCEACAMlb. J Virol. 2010; 84:6654–66. [PubMed: 20410265] \\n98. Seetharaman J, Kanigsberg A, Slaaby R, Leffler H, Barondes SH, Rini JM. X-ray crystal structure \\nof the human galectin-3 carbohydrate recognition domain at 2.1-Åresolution. J Biol Chem. 1998; \\n273:13047–52. [PubMed: 9582341] \\n99. Kunkel F, Herrler G. Structural and functional-analysis of the surface protein of human coronavirus \\nOC43. Virology. 1993; 195:195–202. [PubMed: 8317096] \\nLi Page 17\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript100. Vijgen L, Keyaerts E, Lemey P, Maes P, van Reeth K, et al. Evolutionary history of the closely \\nrelated group 2 coronaviruses: porcine hemagglutinating encephalomyelitis virus, bovine \\ncoronavirus, and human coronavirus OC43. J Virol. 2006; 80:7270–74. [PubMed: 16809333] \\n101. Vijgen L, Keyaerts E, Moes E, Thoelen I, Wollants E, et al. Complete genomic sequence of \\nhuman coronavirus OC43: molecular clock analysis suggests a relatively recent zoonotic \\ncoronavirus transmission event. J Virol. 2005; 79:1595–604. [PubMed: 15650185] \\n102. Klein A, Krishna M, Varki NM, Varki A. 9-O-acetylated sialic acids have widespread but \\nselective expression: analysis using a chimeric dual-function probe derived from influenza C \\nhemagglutinin-esterase. PNAS. 1994; 91:7782–86. [PubMed: 8052660] \\n103. Chen L, Li F. Structural analysis of the evolutionary origins of influenza virus hemagglutinin and \\nother viral lectins. J Virol. 2013; 87:4118–20. [PubMed: 23365425] \\n104. Eckert DM, Kim PS. Mechanisms of viral membrane fusion and its inhibition. Annu Rev \\nBiochem. 2001; 70:777–810. [PubMed: 11395423] \\n105. Skehel JJ, Wiley DC. Receptor binding and membrane fusion in virus entry: the influenza \\nhemagglutinin. Annu Rev Biochem. 2000; 69:531–69. [PubMed: 10966468] \\n106. Klenk HD, Garten W. Host cell proteases controlling virus pathogenicity. Trends Microbiol. 1994; \\n2:39–43. [PubMed: 8162439] \\n107. Wilson IA, Skehel JJ, Wiley DC. Structure of the haemagglutinin membrane glycoprotein of \\ninfluenza virus at 3 Å resolution. Nature. 1981; 289:366–73. [PubMed: 7464906] \\n108. Stein BS, Gowda SD, Lifson JD, Penhallow RC, Bensch KG, Engleman EG. pH-independent \\nHIV entry into CD4-positive T cells via virus envelope fusion to the plasma membrane. Cell. \\n1987; 49:659–68. [PubMed: 3107838] \\n109. White J, Matlin K, Helenius A. Cell fusion by Semliki Forest, influenza, and vesicular stomatitis \\nviruses. J Cell Biol. 1981; 89:674–79. [PubMed: 6265470] \\n110. Mothes W, Boerger AL, Narayan S, Cunningham JM, Young JA. Retroviral entry mediated by \\nreceptor priming and low pH triggering of an envelope glycoprotein. Cell. 2000; 103:679–89. \\n[PubMed: 11106737] \\n111. Belouzard S, Millet JK, Licitra BN, Whittaker GR. Mechanisms of coronavirus cell entry \\nmediated by the viral spike protein. Viruses. 2012; 4:1011–33. [PubMed: 22816037] \\n112. Heald-Sargent T, Gallagher T. Ready, set, fuse! The coronavirus spike protein and acquisition of \\nfusion competence. Viruses. 2012; 4:557–80. [PubMed: 22590686] \\n113. Xu YH, Lou ZY , Liu YW, Pang H, Tien P, et al. Crystal structure of severe acute respiratory \\nsyndrome coronavirus spike protein fusion core. J Biol Chem. 2004; 279:49414–19. [PubMed: \\n15345712] \\n114. Lu L, Liu Q, Zhu Y , Chan KH, Qin L, et al. Structure-based discovery of Middle East respiratory \\nsyndrome coronavirus fusion inhibitor. Nat Commun. 2014; 5:3067. [PubMed: 24473083] \\n115. Zheng Q, Deng Y , Liu J, van der Hoek L, Berkhout B, Lu M. Core structure of S2 from the \\nhuman coronavirus NL63 spike glycoprotein. Biochemistry. 2006; 45:15205–15. [PubMed: \\n17176042] \\n116. Xu Y , Liu Y , Lou Z, Qin L, Li X, et al. Structural basis for coronavirus-mediated membrane \\nfusion. Crystal structure of mouse hepatitis virus spike protein fusion core. J Biol Chem. 2004; \\n279:30514–22. [PubMed: 15123674] \\n117. Duquerroy S, Vigouroux AN, Rottier PJM, Rey FA, Bosch BJ. Central ions and lateral \\nasparagine/glutamine zippers stabilize the post-fusion hairpin conformation of the SARS \\ncoronavirus spike glycoprotein. Virology. 2005; 335:276–85. [PubMed: 15840526] \\n118. Gao J, Lu G, Qi J, Li Y , Wu Y , et al. Structure of the fusion core and inhibition of fusion by a \\nheptad repeat peptide derived from the S protein of Middle East respiratory syndrome \\ncoronavirus. J Virol. 2013; 87:13134–40. [PubMed: 24067982] \\n119. Supekar VM, Bruckmann C, Ingallinella P, Bianchi E, Pessi A, Carfi A. Structure of a \\nproteolytically resistant core from the severe acute respiratory syndrome coronavirus S2 fusion \\nprotein. PNAS. 2004; 101:17958–63. [PubMed: 15604146] \\n120. Millet JK, Whittaker GR. Host cell proteases: critical determinants of coronavirus tropism and \\npathogenesis. Virus Res. 2015; 202:120–34. [PubMed: 25445340] \\nLi Page 18\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript121. Spaan W, Cavanagh D, Horzinek MC. Coronaviruses: structure and genome expression. J Gen \\nVirol. 1988; 69:2939–52. [PubMed: 3058868] \\n122. Matsuyama S, Taguchi F. Receptor-induced conformational changes of murine coronavirus spike \\nprotein. J Virol. 2002; 76:11819–26. [PubMed: 12414924] \\n123. Zelus BD, Schickli JH, Blau DM, Weiss SR, Holmes KV . Conformational changes in the spike \\nglycoprotein of murine coronavirus are induced at 37° C either by soluble murine CEACAM1 \\nreceptors or by pH 8. J Virol. 2003; 77:830–40. [PubMed: 12502799] \\n124. Qiu Z, Hingley ST, Simmons G, Yu C, Das Sarma J, et al. Endosomal proteolysis by cathepsins is \\nnecessary for murine coronavirus mouse hepatitis virus type 2 spike-mediated entry. J Virol. \\n2006; 80:5768–76. [PubMed: 16731916] \\n125. Eifart P, Ludwig K, Bottcher C, de Haan CA, Rottier PJ, et al. Role of endocytosis and low pH in \\nmurine hepatitis virus strain A59 cell entry. J Virol. 2007; 81:10758–68. [PubMed: 17626088] \\n126. Nakagaki K, Nakagaki K, Taguchi F. Receptor-independent spread of a highly neurotropic murine \\ncoronavirus JHMV strain from initially infected microglial cells in mixed neural cultures. J Virol. \\n2005; 79:6102–10. [PubMed: 15857995] \\n127. Gallagher TM, Buchmeier MJ, Perlman S. Cell receptor-independent infection by a neurotropic \\nmurine coronavirus. Virology. 1992; 191:517–22. [PubMed: 1413526] \\n128. Taguchi F, Matsuyama S, Saeki K. Difference in Bgp-independent fusion activity among mouse \\nhepatitis viruses. Arch Virol. 1999; 144:2041–49')"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res[0][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f6628d1c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"source:27578435\\ncontent:. [PubMed: 19901337] \\n84. Reguera J, Santiago C, Mudgal G, Ordono D, Enjuanes L, Casasnovas JM. Structural bases of \\ncoronavirus attachment to host aminopeptidase N and its inhibition by neutralizing antibodies. \\nPLOS Pathog. 2012; 8:e1002859. [PubMed: 22876187] \\n85. Chen L, Lin YL, Peng G, Li F. Structural basis for multifunctional roles of mammalian aminopep-\\ntidase N. PNAS. 2012; 109:17966–71. [PubMed: 23071329] \\n86. Wong AH, Zhou D, Rini JM. The X-ray crystal structure of human aminopeptidase N reveals a \\nnovel dimer and the basis for peptide processing. J Biol Chem. 2012; 287:36804–13. [PubMed: \\n22932899] \\n87. Tusell SM, Schittone SA, Holmes KV . Mutational analysis of aminopeptidase N, a receptor for \\nseveral group 1 coronaviruses, identifies key determinants of viral host range. J Virol. 2007; \\n81:1261–73. [PubMed: 17093189] \\n88. Peng GQ, Sun DW, Rajashankar KR, Qian ZH, Holmes KV , Li F. Crystal structure of mouse \\ncoronavirus receptor-binding domain complexed with its murine receptor. PNAS. 2011; \\n108:10696–701. [PubMed: 21670291] \\n89. Beauchemin N, Draber P, Dveksler G, Gold P, Gray-Owen S, et al. Redefined nomenclature for \\nmembers of the carcinoembryonic antigen family. Exp Cell Res. 1999; 252:243–49. [PubMed: \\n11501563] \\n90. Tan KM, Zelus BD, Meijers R, Liu JH, Bergelson JM, et al. Crystal structure of murine \\nsCEACAM1a[1,4]: a coronavirus receptor in the CEA family. EMBOJ. 2002; 21:2076–86.\\n91. Wessner DR, Shick PC, Lu JH, Cardellichio CB, Gagneten SE, et al. Mutational analysis of the \\nvirus and monoclonal antibody binding sites in MHVR, the cellular receptor of the murine \\ncoronavirus mouse hepatitis virus strain A59. J Virol. 1998; 72:1941–48. [PubMed: 9499047] \\n92. Thackray LB, Turner BC, Holmes KV . Substitutions of conserved amino acids in the \\nreceptorbinding domain of the spike glycoprotein affect utilization of murine CEACAM1a by the \\nmurine coronavirus MHV-A59. Virology. 2005; 334:98–110. [PubMed: 15749126] \\n93. Leparc-Goffart I, Hingley ST, Chua MM, Jiang XH, Lavi E, Weiss SR. Altered pathogenesis of a \\nmutant of the murine coronavirus MHV-A59 is associated with a Q159L amino acid substitution in \\nthe spike protein. Virology. 1997; 239:1–10. [PubMed: 9426441] \\n94. Tsai JC, Zelus BD, Holmes KV , Weiss SR. The N-terminal domain of the murine coronavirus spike \\nglycoprotein determines the CEACAM1 receptor specificity of the virus strain. J Virol. 2003; \\n77:841–50. [PubMed: 12502800] \\n95. Ohtsuka N, Taguchi F. Mouse susceptibility to mouse hepatitis virus infection is linked to viral \\nreceptor genotype. J Virol. 1997; 71:8860–63. [PubMed: 9343248] \\n96. Ohtsuka N, Yamada YK, Taguchi F. Difference in virus-binding activity of two distinct receptor \\nproteins for mouse hepatitis virus. J Gen Virol. 1996; 77:1683–92. [PubMed: 8760415] \\n97. Hirai A, Ohtsuka N, Ikeda T, Taniguchi R, Blau D, et al. Role of mouse hepatitis virus (MHV) \\nreceptor murine CEACAM1 in the resistance of mice to MHV infection: studies of mice with \\nchimeric mCEACAMla and mCEACAMlb. J Virol. 2010; 84:6654–66. [PubMed: 20410265] \\n98. Seetharaman J, Kanigsberg A, Slaaby R, Leffler H, Barondes SH, Rini JM. X-ray crystal structure \\nof the human galectin-3 carbohydrate recognition domain at 2.1-Åresolution. J Biol Chem. 1998; \\n273:13047–52. [PubMed: 9582341] \\n99. Kunkel F, Herrler G. Structural and functional-analysis of the surface protein of human coronavirus \\nOC43. Virology. 1993; 195:195–202. [PubMed: 8317096] \\nLi Page 17\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript100. Vijgen L, Keyaerts E, Lemey P, Maes P, van Reeth K, et al. Evolutionary history of the closely \\nrelated group 2 coronaviruses: porcine hemagglutinating encephalomyelitis virus, bovine \\ncoronavirus, and human coronavirus OC43. J Virol. 2006; 80:7270–74. [PubMed: 16809333] \\n101. Vijgen L, Keyaerts E, Moes E, Thoelen I, Wollants E, et al. Complete genomic sequence of \\nhuman coronavirus OC43: molecular clock analysis suggests a relatively recent zoonotic \\ncoronavirus transmission event. J Virol. 2005; 79:1595–604. [PubMed: 15650185] \\n102. Klein A, Krishna M, Varki NM, Varki A. 9-O-acetylated sialic acids have widespread but \\nselective expression: analysis using a chimeric dual-function probe derived from influenza C \\nhemagglutinin-esterase. PNAS. 1994; 91:7782–86. [PubMed: 8052660] \\n103. Chen L, Li F. Structural analysis of the evolutionary origins of influenza virus hemagglutinin and \\nother viral lectins. J Virol. 2013; 87:4118–20. [PubMed: 23365425] \\n104. Eckert DM, Kim PS. Mechanisms of viral membrane fusion and its inhibition. Annu Rev \\nBiochem. 2001; 70:777–810. [PubMed: 11395423] \\n105. Skehel JJ, Wiley DC. Receptor binding and membrane fusion in virus entry: the influenza \\nhemagglutinin. Annu Rev Biochem. 2000; 69:531–69. [PubMed: 10966468] \\n106. Klenk HD, Garten W. Host cell proteases controlling virus pathogenicity. Trends Microbiol. 1994; \\n2:39–43. [PubMed: 8162439] \\n107. Wilson IA, Skehel JJ, Wiley DC. Structure of the haemagglutinin membrane glycoprotein of \\ninfluenza virus at 3 Å resolution. Nature. 1981; 289:366–73. [PubMed: 7464906] \\n108. Stein BS, Gowda SD, Lifson JD, Penhallow RC, Bensch KG, Engleman EG. pH-independent \\nHIV entry into CD4-positive T cells via virus envelope fusion to the plasma membrane. Cell. \\n1987; 49:659–68. [PubMed: 3107838] \\n109. White J, Matlin K, Helenius A. Cell fusion by Semliki Forest, influenza, and vesicular stomatitis \\nviruses. J Cell Biol. 1981; 89:674–79. [PubMed: 6265470] \\n110. Mothes W, Boerger AL, Narayan S, Cunningham JM, Young JA. Retroviral entry mediated by \\nreceptor priming and low pH triggering of an envelope glycoprotein. Cell. 2000; 103:679–89. \\n[PubMed: 11106737] \\n111. Belouzard S, Millet JK, Licitra BN, Whittaker GR. Mechanisms of coronavirus cell entry \\nmediated by the viral spike protein. Viruses. 2012; 4:1011–33. [PubMed: 22816037] \\n112. Heald-Sargent T, Gallagher T. Ready, set, fuse! The coronavirus spike protein and acquisition of \\nfusion competence. Viruses. 2012; 4:557–80. [PubMed: 22590686] \\n113. Xu YH, Lou ZY , Liu YW, Pang H, Tien P, et al. Crystal structure of severe acute respiratory \\nsyndrome coronavirus spike protein fusion core. J Biol Chem. 2004; 279:49414–19. [PubMed: \\n15345712] \\n114. Lu L, Liu Q, Zhu Y , Chan KH, Qin L, et al. Structure-based discovery of Middle East respiratory \\nsyndrome coronavirus fusion inhibitor. Nat Commun. 2014; 5:3067. [PubMed: 24473083] \\n115. Zheng Q, Deng Y , Liu J, van der Hoek L, Berkhout B, Lu M. Core structure of S2 from the \\nhuman coronavirus NL63 spike glycoprotein. Biochemistry. 2006; 45:15205–15. [PubMed: \\n17176042] \\n116. Xu Y , Liu Y , Lou Z, Qin L, Li X, et al. Structural basis for coronavirus-mediated membrane \\nfusion. Crystal structure of mouse hepatitis virus spike protein fusion core. J Biol Chem. 2004; \\n279:30514–22. [PubMed: 15123674] \\n117. Duquerroy S, Vigouroux AN, Rottier PJM, Rey FA, Bosch BJ. Central ions and lateral \\nasparagine/glutamine zippers stabilize the post-fusion hairpin conformation of the SARS \\ncoronavirus spike glycoprotein. Virology. 2005; 335:276–85. [PubMed: 15840526] \\n118. Gao J, Lu G, Qi J, Li Y , Wu Y , et al. Structure of the fusion core and inhibition of fusion by a \\nheptad repeat peptide derived from the S protein of Middle East respiratory syndrome \\ncoronavirus. J Virol. 2013; 87:13134–40. [PubMed: 24067982] \\n119. Supekar VM, Bruckmann C, Ingallinella P, Bianchi E, Pessi A, Carfi A. Structure of a \\nproteolytically resistant core from the severe acute respiratory syndrome coronavirus S2 fusion \\nprotein. PNAS. 2004; 101:17958–63. [PubMed: 15604146] \\n120. Millet JK, Whittaker GR. Host cell proteases: critical determinants of coronavirus tropism and \\npathogenesis. Virus Res. 2015; 202:120–34. [PubMed: 25445340] \\nLi Page 18\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript121. Spaan W, Cavanagh D, Horzinek MC. Coronaviruses: structure and genome expression. J Gen \\nVirol. 1988; 69:2939–52. [PubMed: 3058868] \\n122. Matsuyama S, Taguchi F. Receptor-induced conformational changes of murine coronavirus spike \\nprotein. J Virol. 2002; 76:11819–26. [PubMed: 12414924] \\n123. Zelus BD, Schickli JH, Blau DM, Weiss SR, Holmes KV . Conformational changes in the spike \\nglycoprotein of murine coronavirus are induced at 37° C either by soluble murine CEACAM1 \\nreceptors or by pH 8. J Virol. 2003; 77:830–40. [PubMed: 12502799] \\n124. Qiu Z, Hingley ST, Simmons G, Yu C, Das Sarma J, et al. Endosomal proteolysis by cathepsins is \\nnecessary for murine coronavirus mouse hepatitis virus type 2 spike-mediated entry. J Virol. \\n2006; 80:5768–76. [PubMed: 16731916] \\n125. Eifart P, Ludwig K, Bottcher C, de Haan CA, Rottier PJ, et al. Role of endocytosis and low pH in \\nmurine hepatitis virus strain A59 cell entry. J Virol. 2007; 81:10758–68. [PubMed: 17626088] \\n126. Nakagaki K, Nakagaki K, Taguchi F. Receptor-independent spread of a highly neurotropic murine \\ncoronavirus JHMV strain from initially infected microglial cells in mixed neural cultures. J Virol. \\n2005; 79:6102–10. [PubMed: 15857995] \\n127. Gallagher TM, Buchmeier MJ, Perlman S. Cell receptor-independent infection by a neurotropic \\nmurine coronavirus. Virology. 1992; 191:517–22. [PubMed: 1413526] \\n128. Taguchi F, Matsuyama S, Saeki K. Difference in Bgp-independent fusion activity among mouse \\nhepatitis viruses. Arch Virol. 1999; 144:2041–49\\n\\nsource:39034968\\ncontent:.Oncoimmunology. 2021;10:1951019.\\n25. Vladimer GI, Snijder B, Krall N, et al. Global survey of\\nthe immunomodulatory potential of common drugs.Nat\\nChem Biol. 2017;13:681–690\\n.\\n26. Zhou P , Shaffer DR, Alvarez Arias DA, et al. In vivo dis-\\ncovery of immunotherapy targets in the tumour micro-\\nenvironment. Nature. 2014;506:52–57.\\n27. Spigel DR, Reynolds C, Waterhouse D, et al. Phase 1/2\\nstudy of the safety and tolerability of nivolumab plus\\ncrizotinib for theﬁrst-line treatment of anaplastic lym-\\nphoma kinase translocation - positive advanced non-\\nsmall cell lung cancer (CheckMate 370). J Thorac\\nOncol. 2018;13:682–688.\\n28. Felip E, de Braud FG, Maur M, et al. Ceritinib plus\\nnivolumab in patients with advanced ALK-rearranged\\nnon-small cell lung cancer: results of an open-label,\\nmulticenter , phase 1b study. J Thorac Oncol .\\n2020;15:392–403.\\n29. Kim DW, Gadgeel S, Gettinger SN, et al. Brief report:\\nsafety and antitumor activity of alectinib plus atezoli-\\nzumab from a phase 1b study in advanced ALK-positive\\nNSCLC. JTO Clin Res Rep. 2022;3:100367.\\n30. Patel SP , Pakkala S, Pennell NA, et al. Phase Ib study of\\ncrizotinib plus pembrolizumab in patients with previ-\\nously untreated advanced non-small cell lung cancer\\nwith ALK translocation. Oncologist. 2020;25:e562 –\\ne1012.\\n31. Brahmer J, Reckamp KL, Baas P , et al. Nivolumab versus\\ndocetaxel in advanced squamous-cell non-small-cell lung\\ncancer. N Engl J Med. 2015;373:123–135.\\n32. Borghaei H, Paz-Ares L, Horn L, et al. Nivolumab versus\\ndocetaxel in advanced nonsquamous non-small-cell lung\\ncancer. N Engl J Med. 2015;373:1627–1639.\\n33. Solomon BJ, Mok T , Kim DW, et al. First-line crizotinib\\nversus chemotherapy in ALK-positive lung cancer.N Engl\\nJ Med. 2014;371:2167–2177.\\n34. Shaw AT , Kim TM, Crinò L, et al. Ceritinib versus chemo-\\ntherapy in patients with ALK-rearranged non-small-cell\\nlung cancer previously given chemotherapy and crizotinib\\n(ASCEND-5): a randomised, controlled, open-label, phase\\n3 trial.Lancet Oncol. 2017;18:874–886.\\n35. Rittmeyer A, Barlesi F , Waterkamp D, et al. Atezolizu-\\nmab versus docetaxel in patients with previously treated\\nnon-small-cell lung cancer (OAK): a phase 3, open-label,\\nmulticentre randomised controlled trial. Lancet.\\n2017;389:255–265.\\n36. Herbst RS, Giaccone G, de Marinis F , et al. Atezolizumab\\nfor ﬁ\\nrst-line treatment of PD-L1-selected patients with\\nNSCLC. N Engl J Med. 2020;383:1328–1339.\\n37. Peters S, Camidge DR, Shaw AT , et al. Alectinib versus\\ncrizotinib in untreated ALK-positive non-small-cell lung\\ncancer. N Engl J Med. 2017;377:829–838.\\n38. Reck M, Rodriguez-Abreu D, Robinson AG, et al. Pem-\\nbrolizumab versus chemotherapy for PD-L1-positive non-\\nsmall-cell lung cancer .N Engl J Med. 2016;375:1823–1833.\\n39. Dempke WCM, Fenchel K. Has programmed cell death\\nligand-1 MET an accomplice in non-small cell lung\\ncancer?-a narrative review. Transl Lung Cancer Res .\\n2021;10:2667–2682.\\n40. Johnson TR, Tan W, Goulet L, et al. Metabolism, excre-\\ntion and pharmacokinetics of [14C]crizotinib following\\noral administration to healthy subjects. Xenobiotica.\\n2015;45:45–59.\\n41. Xu H, O’Gorman M, Boutros T , et al. Evaluation of cri-\\nzotinib absolute bioavailability, the bioequivalence of\\nthree oral formulations, and the effect of food on cri-\\nzotinib pharmacokinetics in healthy subjects. J Clin\\nPharmacol. 2015;55:104–113.\\n42. Gulley JL, Rajan A, Spigel DR, et al. Avelumab for pa-\\ntients with previously treated metastatic or recurrent\\nnon-small-cell lung cancer (JAVELIN Solid T umor): dose-\\nexpansion cohort of a multicentre, open-label, phase\\n1b trial. Lancet Oncol. 2017;18:599–610.\\nJuly 2024 Avelumab + Lorlatinib or Crizotinib in NSCLC 11\\n\\nsource:39034968\\ncontent:.5 /C2 10\\n9/L, platelet\\ncount /C21 100 /C2 109/L, hemoglobin /C21 109 g/dL, esti-\\nmated creatinine clearance /C21 30 mL/min according to\\nthe Cockcroft-Gault formula, total bilirubin level /C20\\n1.5/C2 the upper limit of norm al [ULN], aspartate\\naminotransferase [AST] or alanine aminotransferase\\n[ALT] /C20 2.5 /C2 ULN, and serum amylase or lipase <\\n1.5 /C2 ULN); no prior immunotherapy; and available\\ntumor tissue (archived tissue or fresh biopsy).\\nIn the phase 1b portion, patients in the avelumab plus\\nlorlatinib group hadALK-positive tumors and any num-\\nber of prior regimens (including zero); prior treatment\\nwith lorlatinib was permitted. In the phase 2 portion, the\\navelumab plus lorlatinib group was limited to patients\\nwith no prior systemic treatment for advanced or met-\\nastatic disease (adjuvant or neoadjuvant treatment was\\npermitted if completed/C21 6 mo before study entry) and\\nno prior TKI treatment at any time. PD-L1 status was not\\nconsidered in eligibility criteria. Patients in the avelumab\\nplus crizotinib group were previously treated and had\\nALK-negative tumors with noROS1 translocations or c-\\nMET ampliﬁcation or exon 14 skipping mutations; pa-\\ntients with EGFR mutations were permitted if they had\\ndisease progression after prior EGFR inhibitor treatment.\\nExclusion criteria included major surgery within 4\\nweeks or radiation therapy within 2 weeks before study\\nentry (prior palliative radiotherapy was permitted if\\ncompleted /C21 48 h before patient registration); systemic\\ncytotoxic anticancer therapy within 2 weeks or TKI\\ntherapy within 5 half-lives before study entry; brain\\nmetastases with some exceptions (patients with brain\\nmetastases were eligible for the avelumab plus crizotinib\\ngroup if they had completed treatment and recovered\\nfrom acute effects of radiotherapy or surgery before\\nstudy entry, had discontinued corticosteroid treatment\\nfor /C21 4 wk, and were neurologically stable; patients with\\nbrain metastases were eligible for the avelumab plus\\nlorlatinib group if they had asymptomatic metastases\\nrequiring <10 mg/d prednisone or equivalent); persis-\\ntent toxicity greater than grade 1 related to previous\\ntherapy (excluding grade 2 alopecia); diagnosis of any\\nother malignancy within 3 years before study entry\\n(except for adequately treated basal or squamous cell\\nskin cancer, carcinoma in situ of the breast or cervix, or\\nlow-grade prostate cancer [Gleason 6 or below] without\\nany plans for treatment intervention); and prior use of\\nimmunosuppressive medication within 7 days before\\nrandomization (except for intranasal, inhaled, or topical\\nsteroids or local steroid injections; systemic corticoste-\\nroids at physiological doses; or steroids as premed-\\nication for hypersensitivity).\\nThis trial was conducted in accordance with the\\nethical principles of the Declaration of Helsinki and the\\nGood Clinical Practice guidelines deﬁned by the Inter-\\nnational Council for Harmonisation. All participating\\npatients provided written informed consent. The proto-\\ncol was approved by the institutional review board or\\nindependent ethics committee at each participating\\ncenter.\\nProcedures\\nAvelumab 10 mg/kg was administered every 2 weeks\\nby intravenous (IV) infusion in both treatment groups.\\nLorlatinib was administered orally as 25 mg tablets, with\\na starting dose of 100 mg once daily. Crizotinib was\\nadministered orally as 250 mg capsules, with a starting\\ndose of 250 mg twice daily. Patients were treated until\\ndisease progression, unacceptable toxicity, or other\\nprotocol-speciﬁed criteria for withdrawal occurred.\\nDiscontinuation of one drug due to treatment-related\\ntoxicity with continuation of treatment as monotherapy\\nwas permitted outside of the dose-limiting toxicity (DLT)\\nperiod. Follow-up periods were 30, 60, and 90 days after\\nthe last study treatment.\\nEnd Points and Assessments\\nThe primary objective in phase 1b was to determine\\nthe maximum tolerated dose (MTD) and recommended\\nphase 2 dose (RP2D) for avelumab in combination with\\neither lorlatinib or crizotinib. The primary objective in\\nphase 2 was to assess the objective response rate (ORR)\\nin all patients and the complete response (CR) rate in the\\navelumab plus lorlatinib group per RECIST 1.1. The\\nprimary end point in phase 1b was occurrence of DLT\\nwithin the ﬁrst 2 treatment cycles (14-d cycle). The\\nprimary end point in phase 2 was conﬁrmed objective\\nresponse per RECIST 1.1 by investigator assessment\\n(including CR in the avelumab plus lorlatinib group).\\nSecondary end points included safety, OS, PK, tumor\\nJuly 2024 Avelumab + Lorlatinib or Crizotinib in NSCLC 3tissue biomarkers, duration of response (DOR; assessed\\nfrom CR or partial response [PR] until progressive dis-\\nease [PD], death, or last tumor assessment), disease\\ncontrol rate (DCR), time to response (TTR), and PFS, all\\nassessed by investigator per RECIST 1.1.\\nThe MTD was the highest dose of avelumab plus\\nlorlatinib or crizotinib associated with DLT in fewer than\\n33% of patients within theﬁrst 2 cycles of treatment.\\nDLT was de ﬁned as the occurrence of any of the\\nfollowing adverse events (AEs) attributable to one or\\nmore drug during the primary DLT observation period:\\n(1) hematologic AEs comprising grade 4 neutropenia\\nlasting more than 7 days; febrile neutropenia; grade 3 or\\nhigher neutropenic infection; grade 3 or higher throm-\\nbocytopenia with bleeding; grade 4 thrombocytopenia;\\nand grade 4 anemia; (2) nonhematologic AEs comprising\\nany toxicity of grade 3 or higher except for the following:\\ntransient (/C20 6 h) grade 3 ﬂu-like symptoms or fever\\ncontrolled with medical management; transient (/C20 24 h)\\ngrade 3 fatigue, local reactions, or headache; grade 3\\nnausea or vomiting that resolved to grade 1 or lower\\nwithin 7 days with medical management; grade 3 diar-\\nrhea or skin toxicity that resolved to grade 1 or lower\\nwithin 7 days with medical management; grade 3 or\\nhigher amylase or lipase abnormality not associated with\\npancreatitis; tumor ﬂare phenomenon (local pain, irri-\\ntation, or rash localized at tumor site); or an abnormal\\nlaboratory value that is unlikely related to study treat-\\nment; and (3) inability to complete 75% of lorlatinib or\\ncrizotinib treatment or two infusions of avelumab due to\\ntreatment-related toxicities.\\nAEs and laboratory test abnormalities were graded\\naccording to the National Cancer Institute Common\\nTerminology Criteria for Adverse Events version 4.03\\nand coded using the Medical Dictionary for Regulatory\\nActivities. Infusion-re lated reaction (IRR) was\\nanalyzed as an AE of special interest using a com-\\nposite deﬁnition that included the following preferred\\nterms: infusion-related reaction, chills, pyrexia, and\\ndyspnea. Antitumor activity was assessed radiologi-\\ncally at screening, at week 8, and then every 8 weeks\\nuntil PD. Tumor responses were conﬁrmed at least 4\\nweeks after initial documentation of a CR or PR. PD-\\nL1 status was determined centrally using the Ven-\\ntana PD-L1 (SP263) immunohistochemistry assay\\n(Roche Diagnostics, In dianapolis, IN); PD-L1 –positive\\nstatus was deﬁned as a combined positive score (tu-\\nmor and immune cells) of at least 1%.\\nPK parameters of avelumab, lorlatinib, and crizoti-\\nnib were evaluated using blood samples. For PK ana-\\nlyses of avelumab, 3.5-mL samples were collected\\nbefore infusion and 1 hour after infusion on day 1 and\\nat any time on day 8 of cycles 1 and 2. Additional\\nsamples were collected on day 1 of cycles 3 to 5,\\nf o l l o w e db yd a y1e v e r y6c y c l e st h e r e a f t e r ,a n da tt h e\\nend of treatment. For PK analyses of lorlatinib and\\ncrizotinib, samples (4 mL and 3 mL, respectively) were\\nc o l l e c t e dp r e d o s ea n da t1 ,2 ,4 ,6 ,a n d8h o u r sp o s t d o s e\\non day 1 of cycle 2. Sparse blood samples were\\ncollected predose on day 1 of cycles 4 and 7. For all\\npatients enrolled in phase 2, predose blood samples\\nwere collected on day 1 of cycles 2, 4, and 7.\\nSafety and efﬁcacy were assessed in all patients who\\nreceived at least one dose of the study drug (safety\\nanalysis set and full analysis set). DLT was evaluated in\\nall patients in phase 1b who received at least one dose\\nof the study drug and either had DLT during theﬁrst 2\\ncycles of treatment or completed the observation period\\nfor theﬁrst 2 cycles; patients without DLT who received\\nl e s st h a n7 5 %o ft h ed o s e so fl o r l a t i n i bo rc r i z o t i n i bi n\\nthe ﬁrst 2 cycles or less than two avelumab infusions for\\nreasons other than study drug-related toxicity were\\nreplaced. Response according to PD-L1 status was\\nassessed in the biomarker analysis set, which included\\nall patients in the safety analysis set who had at least\\none baseline biomarker assessment and had received at\\nleast one dose of any study drug. Kaplan-Meier esti-\\nmates were used for time-to-event analyses, and the\\nconﬁdence interval (CI) for the median was calculated\\naccording to Brookmeyer and Crowley methodology.\\nPK parameters were assessed in the PK analysis set,\\nwhich included all patients in the safety analysis set who\\nhad sufﬁcient concentration data to estimate at least one\\nPK parameter of interest. Trough plasma concentration\\n(Ctrough) and maximum plasma concentration (Cmax) for\\navelumab, lorlatinib, and crizotinib were summarized\\ndescriptively (n, mean, standard deviation, percentage of\\ncoefﬁcient of variation [CV], median, minimum,\\nmaximum, geometric mean, CV, and 95% CI) by treat-\\nment group, cycle, and day.\\nOther PK parameters included time to maximum\\nplasma concentration, time of last measurable concen-\\ntration, area under the plasma concentration-time curve\\nduring the dosing interval time course, area under the\\nconcentration-time curve from time of dosing to the last\\ncollection time point, and apparent plasma clearance (CL/\\nF). For the metabolite of crizotinib, metabolite-to-parent\\nratio for area under the plasma concentration versus\\ntime curve during the dosing interval time course and the\\nmetabolite-to-parent ratio for C\\nmax were also determined.\\nDose-normalized parameters for lorlatinib and crizotinib\\n(including its metabolite) were also determined.\\nResults\\nPatient Characteristics and Disposition\\nBetween December 18, 2015, and March 6, 2018, 66\\npatients were screened and 43 were enrolled at 16\\n4 Solomon et al JTO Clinical and Research Reports Vol. 5 No. 7centers across the United States, Australia, France, Spain,\\nSouth Korea, and Japan. Patients received either avelu-\\nmab plus lorlatinib (n¼ 31; 28 in phase 1b and three in\\nphase 2) or avelumab plus crizotinib (n ¼ 12; all in\\nphase 1b). In both treatment groups, all patients\\nreceived starting doses (no other dose levels were\\nadministered). Most patients were Asian, had Eastern\\nCooperative Oncology Group performance status of 1 or\\n2, had PD-L1–positive tumors, had tumors with non-\\nsquamous histology, and had received at least two prior\\nanticancer drug regimens (Table 1). Most patients (27 of\\n31) in the avelumab plus lorlatinib group had received\\nprior therapy with an ALK TKI, and most patients (eight\\nof 12) in the avelumab plus crizotinib group had a his-\\ntory of smoking.\\nOn July 13, 2022 (data cutoff date), the study was\\nstopped, and ﬁve patients with ongoing avelumab plus\\nlorlatinib treatment were moved to a continued access\\nstudy (NCT05059522); no patient had ongoing avelu-\\nmab plus crizotinib treatment. The most common reason\\nfor permanent treatment discontinuation in both treat-\\nment groups was PD (Supplementary Table 1). Median\\nfollow-up for OS was 42 months in the avelumab plus\\nlorlatinib group and 35 months in the avelumab plus\\ncrizotinib group. Median duration of exposure in the\\navelumab plus lorlatinib group was 42 weeks (range, 2–\\n308 wk) for avelumab and 46 weeks (range, 2–308 wk)\\nfor lorlatinib; in the avelumab plus crizotinib group,\\nmedian duration of exposure was 15 weeks (range, 4–\\n126 wk) for avelumab and 6 weeks (range, 2–32 wk) for\\ncrizotinib\\n\\nsource:27578435\\ncontent:.\\nLi Page 22\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 2. \\nCryo-electron microscopy structures of prefusion trimeric coronavirus spikes. (a) Trimeric \\nmouse hepatitis coronavirus (MHV) spike (PDB ID: 3JCL) (16). Three monomers are shown \\n(magenta, cyan, and green). (b) One monomer from the trimeric MHV spike. The important \\nfunctional elements of the spike [the S1 N-terminal domain (S1-NTD), the S1 C-terminal \\ndomain (S1-CTD), the fusion peptide (FP), and the heptad repeat (HR-N)] are colored in the \\nsame way as in Figure 1c. The dotted curve indicates a disordered loop. Scissors indicate \\ntwo critical proteolysis sites.\\nLi Page 23\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 3. \\nCrystal structures of betacoronavirus S1 C-terminal domains (S1-CTDs). (a) Structure of \\nsevere acute respiratory syndrome coronavirus (SARS-CoV) S1-CTD complexed with \\nhuman ACE2 (PDB ID: 2AJF) (52). Shown are the core structure of S1-CTD (cyan), the \\nreceptor-binding motif (red), and ACE2 (green). (b) Interface between human SARS-CoV \\nS1-CTD and human ACE2, showing two virus-binding hot spots on human ACE2. Dashed \\nlines indicate salt bridges. (c) Interface between palm civet SARS-CoV S1-CTD and human \\nACE2. Critical residue changes from human to civet SARS-CoV strains are labeled. (d) \\nInterface between human SARS-CoV S1-CTD and rat or mouse ACE2. Critical residue \\nchanges from human to rat or mouse ACE2 are labeled. (e) Structure of Middle East \\nrespiratory syndrome coronavirus (MERS-CoV) S1-CTD complexed with human DPP4 \\n(PDB ID: 4KR0) (69).\\nLi Page 24\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 4. \\nCrystal structures of alphacoronavirus S1 C-terminal domains (S1-CTDs). (a) Structure of \\nhuman coronavirus NL63 (HCoV-NL63) S1-CTD complexed with human ACE2 (PDB ID: \\n4KBH) (83). (b) Structure of porcine respiratory coronavirus (PRCV) S1-CTD complexed \\nwith porcine APN (PDB ID: 4F5C) (84). (c) Structural topology of alphacoronavirus S1-\\nCTDs. β-Strands are shown as arrows. (d) Structural topology of betacoronavirus S1-CTDs. \\nα-Helices are shown as cylinders. All of the secondary structural elements in panels c and d \\nare connected in the same order, even though strands β4, β1, and β2 in panel c become \\nhelices α4, α1, and a loop, respectively, in panel d. (e) Interface between HCoV-NL63 S1-\\nLi Page 25\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptCTD and human ACE2. Virus-binding motifs (VBMs) on ACE2 are shown in blue. \\nReceptor-binding motifs (RBMs) on S1-CTD are shown in red. (f) Interface between SARS-\\nCoV S1-CTD and human ACE2.\\nLi Page 26\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 5. \\nCrystal structures of betacoronavirus S1 N-terminal domains (S1-NTDs). (a) Structure of \\nmouse hepatitis coronavirus (MHV) S1-NTD complexed with murine CEACAM1 (PDB ID: \\n3R4D) (88). The core structure of MHV S1-NTD is shown in magenta and green, the \\nreceptor-binding motifs in red, and the rest in cyan. The N-terminal immunoglobulin domain \\nof CEACAM1 is shown in yellow and virus-binding motifs in blue. (b) Structure of bovine \\ncoronavirus (BCoV) S1-NTD (PDB ID: 4H14) (43). The asterisk indicates the binding site \\nfor sugar receptor Neu5,9Ac2. (c) Structure of human galectin-3 complexed with galactose \\n(PDB ID: 1A3K). Sugar receptor is shown in blue. (d) Structure of influenza virus HA1 \\n(PDB ID: 1JSO). Sugar receptor is shown in blue. (e–g) Structural topologies of (e) \\nbetacoronavirus S1-NTDs, (f) human galectins, and (g) influenza virus HA1.\\nLi Page 27\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 6. \\nStructural mechanism for membrane fusion by coronavirus spikes. (a) Structural mechanism \\nfor membrane fusion by class I viral membrane fusion proteins. Schematics of these proteins \\nin both prefusion and postfusion conformations are shown. (b) Negative-stain electron \\nmicroscopy images of SARS-CoV spike in both prefusion and postfusion conformations are \\nshown (18). (c) Schematics of SARS-CoV postfusion S2 in solution (left) and in vivo \\n(right). Abbreviations: FP, fusion peptide; HR-N, heptad repeat region N; HR-C, heptad \\nrepeat region C; IC, intracellular tail; SARS-CoV , severe acute respiratory syndrome \\ncoronavirus; TM, transmembrane anchor.\\nLi Page 28\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 7. \\nTriggers for coronavirus spikes to fuse membranes. Scissors indicate potential spike-\\nprocessing host proteases. Shown are virus particles (green spheres), virus surface spikes \\n(blue protrusions), viral genome (magenta coils), cells (large gray circles), endosome/\\nlysosome (small shaded gray circle), and receptor (light green base on cell surface). Spike-\\nprocessing host proteases are labeled for representative coronaviruses: Middle East \\nrespiratory syndrome coronavirus (MERS-CoV), mouse hepatitis coronavirus (MHV), and \\nsevere acute respiratory syndrome coronavirus (SARS-CoV).\\nLi Page 29\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptFigure 8. \\nEvolution of coronavirus spikes. (a) Structural comparison between human galectins and \\nalphacoronavirus HCoV-NL63 S1-CTD. Both the crystal structures and structural topologies \\nof the two proteins are shown. Common subcore structures in the two proteins are \\nhighlighted with gray shading. (b) Hypothesized evolution of coronavirus spike proteins. \\nAbbreviations: HCoV-NL63, human coronavirus NL63; IC, intracellular tail; S1-CTD, S1 \\nC-terminal domain; S1-NTD, S1 N-terminal domain; TM, transmembrane anchor.\\nLi Page 30\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript\\n\\nsource:32007145\\ncontent:.\\nBased on these genomes, we developed a real-time \\nPCR assay and tested the original clinical samples from \\nthe BGI (WH01, \\nWH02, WH03, and WH04) again to \\ndetermine their threshold cycle (Ct) values (table). The \\nremaining samples were tested by a different real-time \\nPCR assay developed by the Chinese CDC, with Ct values \\nranging from 22·85 to 32·41 (table). These results \\nconfirmed the presence of 2019-nCoV in the patients.\\nBronchoalveolar lavage fluid samples or cultured \\nviruses of nine patients were used for next-generation \\nsequencing. After removing host (human) reads, de novo \\nassembly was done and the contigs obtained used as \\nqueries to search the non-redundant protein database. \\nSome contigs identified in all the samples were \\nclosely related to the bat SARS-like betacoronavirus \\nbat-SL-CoVZC45 betacoronavirus.\\n23 Bat-SL-CoVZC45 \\nwas then used as the reference genome and reads from \\neach pool were mapped to it, generating consensus \\nsequences corresponding to all the pools. These \\nconsensus sequences were then used as new reference \\ngenomes. Eight complete genomes and two partial \\ngenomes (from samples WH19002 and WH02; table) \\nwere obtained. The de novo assembly of the clean reads \\nfrom all the pools did not identify any other long contigs \\nthat corresponded to other viruses at high abundance.\\nThe eight complete genomes were nearly identical \\nacross the whole genome, with sequence identity above \\n99·98%, indicative of a very recent emergence into the \\nhuman population (figure 1A). The largest nucleotide \\ndifference was four mutations. Notably, the sequence \\nidentity between the two virus genomes from the same \\npatient (WH19001, from bronchoalveolar lavage fluid, \\nand WH19005, from cell culture) was more than 99·99%, \\nSee Online for appendix\\nFor Genbank see https://www.\\nncbi.nlm.nih.gov/genbank\\nFor the China National \\nMicrobiological Data Center \\nwebsite see http://nmdc.cn/\\nFor the data from BGI on the \\nChina National GeneBank see \\nhttps://db.cngb.org/datamart/\\ndisease/DATAdis19/Articles\\nwww.thelancet.com   Vol 395   February 22, 2020 569\\nFigure 1: Sequence comparison and genomic organisation of 2019-nCoV\\n(A) Sequence alignment of eight full-length genomes of 2019-nCoV, 29 829 base pairs in length, with a few nucleotides truncated at both ends \\nof the genome. \\n(B) Coding regions of 2019-nCoV, bat-SL-CoVZC45, bat-SL-CoVZXC21, SARS-CoV, and MERS-CoV. Only open reading frames of more than 100 nucleotides are \\nshown. 2019-nCoV=2019 novel coronavirus. SARS-CoV=severe acute respiratory syndrome coronavirus. MERS-CoV=Middle East respiratory syndrome coronavirus.\\nE M 7 10bN\\nE M 6 8b 9a\\nMG772933\\nBat_SL-CoVZC45\\n29802 base pairs\\nMG772934\\nBat_SLCoVZXC21\\n29 732 base pairs\\nNC_004718.3\\nSARS-CoV\\n29 751 base pairs 3a 7a 9b\\n8a3b\\n2019-nCoV\\nWH04\\n29844 base pairs\\nNon-structural polyprotein 1ab (7073 amino acids)\\n265 21 486\\nSpike protein (1255 amino acids)\\n21 493 25  260+C, 13 398\\nTTTTTAAACCGGG\\nConsensus\\nWH04\\nWH03\\nWH01\\nYH8011\\nSample\\nWH19001\\nWH19004\\nWH19005\\nWH19008\\nA\\nB\\nNC_019843.3\\nMERS-CoV\\n30 119 base pairs 3\\n54b E\\nM\\nN\\n8b4a\\nNon-structural polyprotein 1ab (7078 amino acids)\\n279 21 514\\nSpike protein (1270 amino acids)\\n21 7052 5 517+C, 13433\\nTTTTTAAACCGGG\\n Non-synonymous mutations\\n Synonymous mutations\\n5’\\n5’\\n5’\\n5’\\n5’\\n5’\\n5’\\n5’\\n5’\\n5’ 3’\\n5’ 3’\\nTTTTTAAACCGGG\\n+C, 13  443\\nCC TG GA C T C\\nA\\nT\\nAA\\nG\\nT\\nC\\nT\\nA\\n3’\\n3’\\n3’\\n3’\\n3’\\n3’\\n3’\\n3’\\n3’\\n1 5000 10 000 15 000 20 000 25 0002 9 829\\n7b\\n3a 8\\nNon-structural polyprotein 1ab (7070 amino acids)\\n265 21 477\\nSpike protein (1245 amino acids)\\n21 4842 5 221+C, 13 389\\nTTTTTAAACCGGG\\n5’ 3’\\n3a 8\\n3b\\nNon-structural polyprotein 1ab  (7092 amino acids)\\n265 21 543\\nSpike protein (1246 amino acids)\\n21 5502 5 290+C, 13455\\nTTTTTAAACCGGG\\n5’ 3’\\n9 13 14\\n3b 91 31 4\\n3\\nE M 7\\n8\\n10b\\nNon-structural polyprotein 1ab (7096 amino acids)\\n251 21 541\\nSpike protein (1273 amino acids)\\n21 5492 5 730+C, 13 443\\nTTTTTAAACCGGG\\n5’ 3’\\nN\\nE M 7 10bN\\n9 13 14\\n6943 8757 11 739 20 64620 655 24 301 27 46928 12028 229\\nBase pairsArticles\\n570 www.thelancet.com   Vol 395   February 22, 2020\\nwith 100% identity at the amino acid level. In addition, \\nthe partial genomes from samples WH02 and WH19002 \\nalso had nearly 100% identity to the complete genomes \\nacross the aligned gene regions.\\nA Blastn search of the complete genomes of 2019-nCoV \\nrevealed that the most closely related viruses available \\non GenBank were bat-SL-CoVZC45 (sequence identity \\n87·99%; query coverage 99%) and another SARS-like \\nbetacoronavirus of bat origin, bat-SL-CoVZXC21 (acces\\n-\\nsion number MG772934;23 87·23%; query coverage 98%). \\nIn five gene regions (E, M, 7, N, and 14), the sequence \\nidentities were greater than 90%, with the highest \\nbeing 98·7% in the E gene (fi gure 2A). The S gene of \\n2019-nCoV exhibited the lowest sequence identity with \\nbat-SL-CoVZC45 and bat-SL-CoVZXC21, at only around \\n75%. In addition, the sequence identity in 1b (about 86%) \\nwas lower than that in 1a (about 90%; figure 2A). Most of \\nthe encoded proteins exhibited high sequence identity \\nbetween 2019-nCoV and the related bat-derived corona\\n-\\nviruses (figure 2a). The notable exception was the spike \\nprotein, with only around 80% sequence identity\\n, and \\nprotein 13, with 73·2% sequence identity. Notably, the \\n2019-nCoV strains were less genetically similar to \\nSARS-CoV (about 79%) and MERS-CoV (about 50%). The \\nsimilarity between 2019-nCoV and related viruses was \\nvisualised using SimPlot software, with the 2019-nCoV \\nconsensus sequence employed as the query (figure 2B).\\nComparison of the predicted coding regions of \\n2019-nCoV showed that they possessed a similar genomic \\norganisation to bat-SL-CoVZC45, bat-SL-CoVZXC21, and \\nSARS-CoV (figure 1B). At least 12 coding regions were \\npredicted, including 1ab, S, 3, E, M, 7, 8, 9, 10b, N, 13, and \\n14 (figure 1B). The lengths of most of the proteins encoded \\nby 2019-nCoV, bat-SL-CoVZC45, and bat-SL-CoVZXC21 \\nwere similar, with only a few minor insertions or dele\\n-\\ntions. A notable difference was a longer spike protein \\nencoded by 2019-nC\\noV compared with the bat SARS-like \\ncoronaviruses, SARS-CoV, and MERS-CoV (figure 1B).\\nPhylogenetic analysis of 2019-nCoV and its closely \\nrelated reference genomes, as well as representative beta-\\ncoronaviruses, revealed that the five subgenera formed \\nfive well supported branches (\\nfigure 3). The subgenus \\nFigure 2: Sequence identity between the consensus of 2019-nCoV and representative betacoronavirus genomes\\n(A) Sequence identities for 2019-nCoV compared with SARS-CoV GZ02 (accession number AY390556) and the bat SARS-like coronaviruses bat-SL-CoVZC45 \\n(MG772933) and bat-SL-CoVZXC21 (MG772934). (B) Similarity between 2019-nCoV and related viruses. 2019-nCoV=2019 novel coronavirus. SARS-CoV=severe \\nacute respiratory syndrome coronavirus.\\n1abS 3\\nE\\nM\\n7\\n8\\n10b\\nN\\n13 14\\nStrain Complete\\ngenome\\n(%)\\nGene region (%)\\nNucleotide\\nsequences\\nAmino acid \\nsequences\\nBat-SL-CoVZC45\\nBat-SL-CoVZXC21\\nSARS-CoVGZ02\\n87·6\\n87·5\\n79·0\\n88·9\\n88·7\\n79·5\\n90·7\\n90·3\\n75·4\\n86·0\\n86·1\\n86·3\\n75·2\\n74·7\\n72·7\\n87·8\\n88·9\\n75·6\\n98·7\\n98·7\\n93·5\\n93·4\\n93·4\\n85·1\\n95·2\\n95·2\\n74·5\\n88·8\\n89·1\\n82·1\\n88·5\\n88·5\\n··\\n91·1\\n91·2\\n88·1\\n89·1\\n89·5\\n··\\n96·7\\n96·7\\n··\\n95·6\\n95·2\\n86·2\\n95·6\\n95·1\\n80·5\\n95·8\\n95·5\\n95·6\\n80·2\\n79·6\\n76·2\\n90·9\\n92·0\\n73·1\\n100·0\\n100·0\\n94·7\\n98·6\\n98·6\\n90·1\\n93·4\\n93·4\\n68·9\\n87·6\\n88·4\\n85·2\\n94·2\\n94·2\\n··\\n94·3\\n94·3\\n90·3\\n73·2\\n73·2\\n ··\\n92·9\\n92·9\\n··\\n··\\n··\\n··\\nBat-SL-CoVZC45\\nBat-SL-CoVZXC21\\nSARS-CoVGZ02\\n1a 1b\\n1ab 1a 1b S3 EM 78 10b N1 31 4\\nA\\nB\\n1000 3000 5000 7000 9000 11 000 13 000 15 000 17 000 19 000 21 000 23 000 27 00025 000 29 000\\n0\\n0·05\\n0·10\\n0·15\\n0·20\\n0·25\\n0·30\\n0·35\\n0·40\\n0·45\\n0·50\\n0·55\\n0·60\\n0·65\\n0·70\\n0·75\\n0·80\\n0·85\\n0·90\\n0·95\\n1·00\\nSimilarity (%)\\nPosition\\nSARS-CoV GZ02\\nBat-SL-CoVZC45\\nBat-SL-CoVZXC21\\nGene\\n2019-nCoV\\nConsensusArticles\\nwww.thelancet.com   Vol 395   February 22, 2020 571\\nSarbecovirus could be classified into three well supported \\nclades: two SARS-CoV-related strains from Rhinolophus sp \\nfrom Bulgaria (accession number GU190215) and Kenya \\n(KY352407) formed clade 1; the ten 2019-nCoV from \\nWuhan and the two bat-derived SARS-like strains from \\nZhoushan in eastern China (bat-SL-CoVZC45 and \\nbat-SL-CoVZXC21) formed clade 2, which was notable for \\nthe long branch separating the human and bat viruses; \\nand SARS-CoV strains from humans and many genetically \\nsimilar SARS-like corona\\n viruses from bats collected \\nfrom southwestern China formed clade 3, with bat-derived\\n \\ncoronaviruses also falling in the basal positions (figure 3). \\nIn addition, 2019-nCoV was distinct from SARS-CoV  \\nin a phylogeny of the complete RNA-dependent RNA \\npolymerase (RdRp) gene (appendix p 2). This evidence \\nindicates that 2019-nCoV is a novel beta\\n coronavirus \\nfrom the subgenus S\\narbecovirus.\\nAs the sequence similarity plot revealed changes in \\ngenetic distances among viruses across the 2019-nCoV  \\ngenome, we did additional phylogenetic analyses of \\nthe major encoding regions of representative members \\nof the subgenus Sarbecovirus. Consistent with the \\ngenome phylogeny, 2019-nCoV, bat-SL-CoVZC45, and \\nbat-SL-CoVZXC21 clustered together in trees of \\nthe 1a and spike genes (appendix p 3). By contrast, \\n2019-nCoV did not cluster with bat-SL-CoVZC45 and \\nbat-SL-CoVZXC21 in the 1b tree, but instead formed a \\ndistinct clade with SARS-CoV, bat-SL-CoVZC45, and \\nbat-SL-CoVZXC21 (appendix p 3), indicative of potential \\nrecombination events in 1b, although these probably \\noccurred in the bat coronaviruses rather than 2019-nCoV. \\nPhylogenetic analysis of the 2019-nCoV genome ex\\n-\\ncluding 1b revealed similar evolutionary relationships as \\nthe full-length viral genome (appendix p 3).\\nThe envelope spike (S) protein mediates receptor \\nbinding and membrane fusion\\n24 and is crucial for \\ndetermining host tropism and transmission capacity\\n\\nsource:39034968\\ncontent:. 1). The steady-state plasma PK of\\nlorlatinib in combination with avelumab was similar to\\nthat of previously reported exposure with lorlatinib\\nmonotherapy (Supplementary Table 5).10 Serum con-\\ncentrations of avelumab were similar across patients in\\neither treatment group. Exposures for crizotinib and its\\nmetabolite in combination with avelumab were similar\\nto those reported for crizotinib monotherapy\\n(Supplementary Tables 6 and 7).40,41 The variability of\\ncrizotinib CL/F was high because of the small sample\\nsize.\\nDiscussion\\nJAVELIN Lung 101 evaluated the safety, ef ﬁcacy,\\nand PK of avelumab in combination with lorlatinib or\\ncrizotinib for patients with ALK-negative or ALK-posi-\\ntive advanced NSCLC, respectively. Most patients had\\nreceived previous treatment for advanced or meta-\\nstatic disease. Avelumab plus lorlatinib had a\\nmanageable safety pro ﬁle, whereas avelumab plus\\ncrizotinib resulted in DLT inﬁve of 12 patients (42%).\\nThe MTD and RP2D for avelumab plus lorlatinib was\\ndetermined to be avelumab 10 mg/kg IV every 2\\nweeks plus lorlatinib 100 mg once daily. The MTD for\\navelumab plus crizotinib was exceeded at the doses\\nadministered (10 mg/kg IV every 2 wk and 250 mg\\ntwice daily, respectively), and this combination did\\nnot proceed to phase 2. Enrollment was stopped\\nearly based on the incidence of hepatic toxicity in the\\navelumab plus crizotinib group and the changing\\ntreatment landscape for patients with metastatic\\nALK-positive NSCLC. Co-administration of avelumab\\nwith lorlatinib or crizotinib had no clinically meaningful\\nTable 4.Summary of Responses per Investigator Assessment\\nResponse\\nAvelumab þ\\nLorlatinib\\n(n ¼ 31)\\nAvelumab þ\\nCrizotinib\\n(n ¼ 12)\\nConﬁrmed best overall\\nresponse, n (%)\\nCR 1 (3) 0\\nPR 15 (48) 3 (25)\\nSD 6 (19) 4 (33)\\nPD 7 (23) 5 (42)\\nNE 2 (7) 0\\nORR (95% CI), % 52 (33–70) 25 (6 –57)\\nMedian TTR\\n(range), mo\\n1.8 (1.3–3.7) 1.4 (1.4 –6.9)\\nMedian DOR\\n(95% CI), mo\\n14.7 (3.7–NE) 3.7 (3.7 –4.6)\\nDCR (95% CI), % 71 (52–86) 58 (28 –85)\\nCI, conﬁdence interval; CR, complete response; DCR, disease control rate;\\nDOR, duration of response; NE, not evaluable; ORR, objective response rate;\\nPD, progressive disease; PR, partial response; SD, stable disease; TTR, time\\nto response.\\nFigure 1.Best percentage change from baseline in sum of diameters for target lesions per investigator assessment in the (A)\\navelumab plus lorlatinib and (B) avelumab plus crizotinib groups. CR, complete response; PD, progressive disease; PD-L1,\\nprogrammed death-ligand 1; PR, partial response; SD, stable disease.\\nJuly 2024 Avelumab + Lorlatinib or Crizotinib in NSCLC 7effect on lorlatinib or crizotinib PK based on compari-\\nsons with historical monotherapy data.3,9\\nAntitumor activity was observed with avelumab plus\\nlorlatinib in patients with ALK-positive tumors and\\navelumab plus crizotinib in patients withALK-negative\\ntumors. The ORR for avelumab plus lorlatinib was\\nsimilar to the ORR reported in a prior study of lorlatinib\\nmonotherapy in patients with advanced, ALK-positive\\nNSCLC who had previously received two or more TKIs\\n(52% versus 46%, respectively).\\n10 Median PFS with\\navelumab plus lorlatinib was comparable with results\\nfrom a phase 2 study of lorlatinib monotherapy in sub-\\ngroups of patients with ALK-positive NSCLC who had\\nprior ALK TKI treatment (6.4 versus 6.9 –7.3 mo,\\nrespectively).\\n11 The ORR for avelumab plus crizotinib in\\nour study (25%) was similar to the ORR reported with\\navelumab monotherapy in a cohort of patients with un-\\ntreated advanced NSCLC without ALK or EGFR alter-\\nations who were unselected based on PD-L1 status\\n(20%) and higher than the ORR reported with avelumab\\nmonotherapy in a cohort of platinum-treated patients\\n(12%).\\n14,42 No meaningful difference was observed in\\nresponse to treatment based on PD-L1 status. This study\\nwas limited by the small sample size, andﬁrm conclu-\\nsions cannot be drawn.\\nIn all previous clinical studies of ICI plus ALK TKI\\ncombination therapy in patients with ALK-positive\\nNSCLC, increased toxicity has been reported compared\\nwith single-agent treatment. 27–30 The reason for the\\nhigher toxicity in the avelumab plus crizotinib group\\nversus the avelumab plus lorlatinib is unknown but\\ncould be due to the lower kinase selectivity of crizotinib\\nversus lorlatinib, which may result in additional effects\\nwhen crizotinib is administered in combination with an\\nICI. The high toxicity with avelumab plus crizotinib in\\nthis study is consistent with previous reports of ICIs\\ncombined with crizotinib. In the CheckMate 370 trial,\\n38% of patients treated with nivolumab plus crizotinib\\nhad grade 3 or higher hepatotoxicity, and the study was\\nterminated early.\\n27 Similarly, with pembrolizumab plus\\ncrizotinib, the incidence of grade 3 or higher ALT or AST\\nwas high, and the study was terminated early. 30\\nAlthough combination therapy with nivolumab plus\\nceritinib or atezolizumab plus alectinib was found to\\nhave preliminary antitumor activity, increased toxicity\\ncompared with single agents was also observed.\\n28,29\\nConclusion\\nAvelumab plus lorlatinib combination treatment was\\nfeasible in patients withALK-positive advanced NSCLC,\\nbut antitumor activity did not seem to be substantially\\nincreased compared with previous studies of lorlatinib\\nmonotherapy. Avelumab plus crizotinib combination\\ntreatment was associated with toxicity issues at the\\ndoses evaluated. Results from this study and changes in\\nthe treatment landscape in advanced NSCLC do not\\nsupport further studies of these combinations.\\nData Sharing Statement\\nAny requests for data by qualiﬁed scientiﬁc and med-\\nical researchers for legitimate research purposes will be\\nsubject to the healthcare business of Merck KGaA, Darm-\\nstadt, Germany ’s (CrossRef Funder ID: 10.13039/\\n100009945) Data Sharing Policy. All requests should be\\nsubmitted in writing to the healthcare business of Merck\\nKGaA, Darmstadt, Germany’s data sharing portal (https://\\nwww.emdgroup.com/e n/research/our-approach-to-re\\nsearch-and-development/healthcare/clinical-trials/\\ncommitment-responsible-data-sharing.html ). When\\nthe healthcare business of Merck KGaA, Darmstadt, Ger-\\nmany, has a co-research, co-development, or co-marketing\\nor co-promotion agreement, or when the product has been\\noutlicensed, the responsibility for disclosure might be\\ndependent on the agreement between parties. Under\\nthese circumstances, the healthcare business of Merck\\nKGaA, Darmstadt, Germany, will endeavor to gain agree-\\nment to share data in response to requests.\\nFigure 2.Kaplan-Meier analysis of OS and PFS in the (A) avelumab plus lorlatinib and (B) avelumab plus crizotinib groups. OS,\\noverall survival; NR, not reached; PFS, progression-free survival.\\n8 Solomon et al JTO Clinical and Research Reports Vol. 5 No. 7CRediT Authorship Contribution\\nStatement\\nBenjamin J. Solomon: Investigation, Writing— re-\\nview and editing.\\nIbiayi Dagogo-Jack: Investigation, Writing— review\\nand editing.\\nSe-Hoon Lee: Investigation, Writing — review and\\nediting.\\nMichael J. Boyer:Investigation, Writing— review and\\nediting.\\nSuresh S. Ramalingam: Investigation, Writing— re-\\nview and editing.\\nEnric Carcereny:Investigation, Writing— review and\\nediting.\\nEnriqueta Felip:Investigation, Writing— review and\\nediting.\\nJi-Youn Han: Investigation, Writing — review and\\nediting.\\nToyoaki Hida: Investigation, Writing— review and\\nediting.\\nBrett G.M. Hughes: Investigation, Writing— review\\nand editing.\\nSang-We Kim: Investigation, Writing— review and\\nediting.\\nMakoto Nishio: Investigation, Writing— review and\\nediting.\\nTakashi Seto: Investigation, Writing— review and\\nediting.\\nTatsuro Okamoto: Investigation, Writing — review\\nand editing.\\nXiaoxi Zhang: Data curation, Formal analysis, Visu-\\nalization, Writing— review and editing,.\\nJean-Francois Martini: Data curation, Formal anal-\\nysis, Visualization, Writing— review and editing,.\\nErjian Wang: Data curation, Formal analysis, Visu-\\nalization, Writing— review and editing.\\nSteven De Beukelaer: Conceptualization, Data\\ncuration, Supervision, Writing— review and editing.\\nTodd M. Bauers:Investigation, Writing— review and\\nediting.\\nDisclosure\\nDr. Solomon reports providing a consulting or advi-\\nsory role for Amgen, AstraZeneca, BeiGene, Bristol Myers\\nSquibb, Cancer Council of Victoria, D3 Bio, Janssen, Lilly,\\nthe healthcare business of Merck KGaA, Darmstadt,\\nGermany, Pﬁzer, Roche/Genentech, Takeda, and Thoracic\\nOncology Group of Australasia; has provided speaker\\nservices for Amgen, AstraZeneca, P ﬁzer, and Roche/\\nGenentech; and has received institutional research\\nfunding from BeiGene, Bristol Myers Squibb, Lilly,\\nNovartis, Nuvalent, Pﬁzer, Roche/Genentech, and Sanoﬁ.\\nProf. Dagogo-Jack reports providing a consulting or\\nadvisory role for AstraZeneca, Bayer, Boehringer Ingel-\\nheim, BostonGene, Catalyst Pharmaceuticals, Genentech,\\nJanssen, Novocure, Pﬁzer, Sanoﬁ, Syros Pharmaceuticals,\\nand Xcovery; has received travel, accommodations, or\\nexpenses from Array BioPharma, Creative Educational\\nConcepts, DAVA Oncology, Medscape, OncLive/MJH Life\\nSciences, P ﬁzer, The ASCO Post, and Total Health\\nConferencing; and has received institutional research\\nfunding from Array BioPharma, Calithera Biosciences,\\nGenentech, Pﬁzer, Novartis, and Vivace Therapeutics.\\nProf. Lee reports providing a consulting or advisory role\\nfor AstraZeneca/MedImmune, Bristol Myers Squibb, and\\nNovartis; has received honoraria from AstraZeneca/\\nMedImmune, the healthcare business of Merck KGaA,\\nDarmstadt, Germany, and Roche; and has received travel,\\naccommodations, or expenses from Novartis. Prof. Boyer\\nreports providing a consulting or advisory role for\\nAstraZeneca/MedImmune, Bristol Myers Squibb, and\\nMerck & Co., Kenilworth, NJ; has received institutional\\nresearch funding from Amgen, Ascentage Pharma,\\nAstraZeneca, BeiGene, Boehringer Ingelheim, Bristol\\nMyers Squibb, Lilly, Merck & Co., Kenilworth, NJ,\\nOncoMed, Peregrine Pharmaceuticals, Pﬁzer, and Roche/\\nGenentech; and has received travel, accommodations, or\\nexpenses from Boehringer Ingelheim, Bristol Myers\\nSquibb, Merck & Co., Kenilworth, NJ, and Roche/Gen-\\nentech. Prof. Ramalingam reports providing a consulting\\nor advisory role for AbbVie, Amgen, AstraZeneca, Boeh-\\nringer Ingelheim, Bristol Myers Squibb, Celgene, Lilly/\\nImClone, Roche/Genentech, Takeda, and the healthcare\\nbusiness of Merck KGaA, Darmstadt, Germany. Dr. Car-\\ncereny has no relationships to disclose. Prof. Felip re-\\nports providing a consulting or advisory role for AbbVie,\\nAmgen, AstraZeneca, Bayer, Blueprint Medicines, Boeh-\\nringer Ingelheim, Bristol Myers Squibb, Lilly, Glax-\\noSmithKline, Janssen, Merck & Co., Kenilworth, NJ,\\nNovartis, P ﬁzer, Puma Biotechnology, Roche, Sano ﬁ,\\nGenzyme, Takeda, and the healthcare business of Merck\\nKGaA, Darmstadt, Germany; has provided speaker ser-\\nvices for AstraZeneca, Boehringer Ingelheim, Bristol\\nMyers Squibb, Lilly, Medscape, Merck & Co., Kenilworth,\\nNJ, Novartis, PeerVoice, Pﬁzer, Prime Oncology, Roche,\\nSpringer, Takeda, touchIME, and CME Out ﬁtters; has\\nreceived research funding from grant for Oncology\\nInnovation, Fundación Merck Salud, Madrid, Spain, an\\nafﬁliate of Merck KGaA, Darmstadt, Germany; and is an\\nindependent member of the board of Grifol. Dr. Han re-\\nports providing a consulting or advisory role for Merck &\\nCo., Kenilworth, NJ; has received honoraria from\\nNovartis; and has received institutional research funding\\nfrom Roche. Dr\\n\\nsource:06463922\\ncontent:Thorax1984;39:475-476\\nRuptureofananeurysmofthesinusofValsalvaintothesuperiorvenacava\\nGCKAYE,SJEDMONSON,JLCAPLIN,DSTUNSTALL-PEDOE\\nFromtheDepartmentofCardiology,StBartholomew'sHospital,London\\nWepresentanunusualcaseinwhichruptureofasinusof\\nValsalvaduringasepticaemicillnessproducedafistula\\nbetweentheaortaandthesuperiorvenacava.\\nCasereport\\nA64yearoldman(caucasian),previouslyingoodhealth,\\npresentedinJuly1982witha24hourhistoryofrigors,\\nabdominalpain,andjointpains.Onexaminationhewas\\nfebrile(39.5°C),withabloodpressureof140/60mmHg.\\nAsoftapicalsystolicmurmurwasnoted.Thehaemoglobin\\nconcentrationwas13-1g/dlandthewhitecellcount8-9x\\n109/1(84%neutrophils).Amidstreamspecimenofurine\\ngrewEscherichiacoli,asdidoneofsixbloodculturebot-\\ntles.Adiagnosisofsepticaemiasecondarytoaurinary\\ntractinfectionwasmade.Hewastreatedfor10dayswith\\noralamoxycillin,500mgthricedaily,andmadearapid\\nrecovery.\\nInAugust1982hehadarecurrenceofrigors.Againhe\\nwasfebrile(395°C)andbilateralpleuralfrictionrubswere\\nnotedaswellasarightpleuraleffusion.Thehaemoglobin\\nconcentrationwas12-9g/dlandthewhitecellcount161x\\n109/l(92%neutrophils).Amidstreamspecimenofurine\\nwassterilebutbloodculturesgrewEcoli.Investigationof\\ntherenaltractshowedasmallfluidfilledcystintheright\\nkidneyandsomeprostaticenlargement.Itwasconsidered\\nthattherecurrentrenaltractinfectioncouldbeattributed\\ntothecystortotheprostatichypertrophy.Therenalcyst\\nwasconsideredtoosmalltoallowaspiration.Aftera\\ncourseofgentamicin,80mgthricedaily,atransurethral\\nprostatectomywasperformedandbenignprostatictissue\\nobtained.Thiswasfollowedbyfurtherrigorsandfever,\\nwhichrespondedtointramuscularcephradine,1gthrice\\ndaily.\\nThepatientwasreadmittedinDecember1982com-\\nplainingofincreasingdyspnoeaandpalpitations;hehad\\nclinicalsignsoffluidretention.Onexaminationhewas\\nafebrileandatrialfibrillationwasnotedforthefirsttime.\\nThepulsewasboundingandthebloodpressurewas140/\\n70mmHg.Thejugularvenouspressurewasraisedby5\\ncm.Aloud,longsystolicandanearlydiastolicmurmur\\nwereheardmaximallyatthethirdleftintercostalspace.\\nBilateralbasalcrepitationsandasmallrightpleural\\neffusionwerepresent.Theliverwasenlargedandnon-\\npulsatile.\\nThehaemoglobinconcentrationwas9-8g/dl.Achest\\nradiographshowedcardiomegaly;theelectrocardiogram\\nAddressforreprintrequests:DrGKaye,DepartmentofCardiol-\\nogy,StBartholomew'sHospital,LondonEClA7BE.\\nAccepted17January1984\\nFig1Angiogramintherightanteriorobliqueposition.Ao-aorta;f-fistula;SVC-superiorvenacava;Ra-right\\natrium.\\nwasunremarkableexceptforatrialfibrillation.An\\nechocardiogramshoweddiastolicflutteroftheanterior\\nmitralvalveleafletcompatiblewithaorticregurgitation\\nandleftventriculardilatation.Cardiaccatheterisation\\nshowedastepupintheoxygensaturationbetweenthe\\nhighsuperiorvenacavaandthelowsuperiorvenacava\\nfrom68%to89%,withapulmonaryarterysaturationof\\n78%.Anaortogramintherightanteriorobliqueposition\\nshowedaconnectionbetweentheaorticrootandthe\\nsuperiorvenacava(fig1).Trivialaorticregurgitationwas\\nseen.\\nAtoperation(fig2)therewasamoderatelylarge,clear\\npericardialeffusion.Withfullcardiopulmonarybypass,\\naftercrossclamping,theaortawasopened.Inrelationto\\nthenon-coronarysinustherewasadefect3cmindia-\\nmeter,theroofofwhichcommunicatedwiththemedial\\nwallofthesuperiorvenacavaviaasmoothwalledsinus.\\nTheaorticvalvewasnormal.Theholeinthesuperiorvena\\ncavawasclosedwithinterruptedsutures.Thedefectinthe\\naortawastoolargetobecloseddirectly,necessitating\\nreplacementabovethecoronaryostiawithaDacrontube\\ngraft.TheaorticwallgrewStreptococcusfaeciumandanti-\\n475\\nKaye,Edmonson,Caplin,Tunstall-Pedoe\\nROOTOFAORTAOPENED\\nFistulainSVC\\nSinus-\\nLeftcoronaryartery\\nRightcoronaryartery\\nFig2Diagramofdefectsfoundatoperation.\\nSVC-superiorvenacava.\\nbioticswerecontinuedintravenouslyforsixweeks.The\\npatientisnowfullyrecovered.\\nDiscussion\\nConventionallyaneurysmsofthesinusofValsalvahave\\nbeenthoughtofascongenitaldefectsintheaorticmedia.'\\nTheyaccountfor3-5%ofcasesofcongenitalheartdisease\\ntreatedsurgically2andareusuallyassociatedwithother\\ncongenitallesions,notablyventricularseptaldefects,34\\naorticcoarctation,andbicuspidaorticvalve.2Acquired\\naneurysmsofthesinusofValsalvaaremuchlesscommon,\\nthecausesincludingsyphilis,tuberculosis,andbacterial\\nendocarditis.'I\\nAneurysmsofthesinusofValsalvamorecommonly\\nariseintherightcoronarysinus.'3IThosearisinginthe\\nnon-coronaryandleftcoronarycuspsarelesscommonand\\nallmayruptureintotherightsideoftheheart.Without\\nsurgerysurvivalafterruptureisintheregionofoneyear,3\\nalthoughpatientssurviving15yearshavebeenreported.\\nCasesofruptureintotheleftatrium,leftventricle,inter-\\nventricularseptum,andpulmonaryinfundibulumhave\\nbeendescribed.6\\nRuptureintothesuperiorvenacavahasnotpreviously\\nbeendocumented.Thiscaseisofparticularinterestasthe\\naneurysmaroseinthenon-coronarycusp,whichtendsto\\nruptureintotherightheartchambers.'3Astheedgesof\\nthesinusweresowelldemarcated,acongenitaldefect\\nseemedlikely.Sfaeciumwasisolatedfromtheaorticwall,\\nhowever,suggestingthattheorganism,presumablyintro-\\nducedduringprostatectomy,mayhavedamagedthesinus\\nfurther,eitherproducingafistulaor,lesslikely,openingup\\nasmalldefectthatwasalreadypresent.Casesofbacterial\\nendocarditisproducingaorticocardiacfistulaswithout\\naneurysmformationhavebeennoted;2possiblytheinitial\\ndamagetotheaortawascausedbytheEcolisepticaemia,\\nSfaeciumcausingthesubsequentaneurysmtorupture.\\nReferences\\n'SakakibaraS,KonnoS.Congenitalaneurysmofthesinusof\\nValsalva:anatomyandclassification.AmHeartJ\\n1962;63:405-22.\\n2CondeCA,MellerJ,DonosoE,DackS.Bacterialendocarditis\\nwithrupturedsinusofValsalvaandaorticocardiacfistula.Am\\nJCardiol1975;35:912-6.\\nGerbodeF,OsborneJJ,JohnstonJB,KeithWJ.Ruptured\\naneurysmsoftheaorticsinus.AmJSurg1961;102:268-77.\\n4JonesAM,LangleyFA.Aorticsinusaneurysm.BrHeartJ\\n1949;11:325-41.\\n5OramS,EastT.Ruptureofaneurysmofaorticsinus(ofVal-\\nsalva)intorightsideoftheheart.BrHeartJ1955;17:541-51.\\n6DeSa-NetoA,PadnikMB,DesserKB,SteinhoffNG.RightsinusofValsalvaaneurysmcausingarightatrialfistulafollow-\\ningnon-penetratingchesttrauma.Circulation1979;60:205-9.\\n476\\n\\nsource:27578435\\ncontent:. Wong SK, Li WH, Moore MJ, Choe H, Farzan M. A 193-amino acid fragment of the SARS \\ncoronavirus S protein efficiently binds angiotensin-converting enzyme 2. J Biol Chem. 2004; \\n279:3197–201. [PubMed: 14670965] \\n47. Lin HX, Fen Y , Wong G, Wang LP, Li B, et al. Identification of residues in the receptor-binding \\ndomain (RBD) of the spike protein of human coronavirus NL63 that are critical for the RBD-\\nACE2 receptor interaction. J Gen Virol. 2008; 89:1015–24. [PubMed: 18343844] \\n48. Hofmann H, Simmons G, Rennekamp AJ, Chaipan C, Gramberg T, et al. Highly conserved regions \\nwithin the spike proteins of human coronaviruses 229E and NL63 determine recognition of their \\nrespective cellular receptors. J Virol. 2006; 80:8639–52. [PubMed: 16912312] \\n49. Godet M, Grosclaude J, Delmas B, Laude H. Major receptor-binding and neutralization \\ndeterminants are located within the same domain of the transmissible gastroenteritis virus \\n(coronavirus) spike protein. J Virol. 1994; 68:8008–16. [PubMed: 7525985] \\n50. Du L, Zhao G, Kou Z, Ma C, Sun S, et al. Identification of a receptor-binding domain in the S \\nprotein of the novel human coronavirus Middle East respiratory syndrome coronavirus as an \\nessential target for vaccine development. J Virol. 2013; 87:9939–42. [PubMed: 23824801] \\n51. Mou H, Raj VS, van Kuppeveld FJ, Rottier PJ, Haagmans BL, Bosch BJ. The receptor binding \\ndomain of the new MERS coronavirus maps to a 231-residue region in the spike protein that \\nefficiently elicits neutralizing antibodies. J Virol. 2013; doi: 10.1128/JVI.01277-13\\n52. Li F, Li WH, Farzan M, Harrison SC. Structure of SARS coronavirus spike receptor-binding \\ndomain complexed with receptor. Science. 2005; 309:1864–68. [PubMed: 16166518] \\n53. Li, F., Li, WH., Farzan, M., Harrison, SC. Interactions between SARS coronavirus and its receptor. \\nIn: Perlman, S., Holmes, KV ., editors. Nidoviruses: Toward Control of SARS and Other Nidovirus \\nDiseases. New York: Springer; 2006. p. 229-34.\\n54. Towler P, Staker B, Prasad SG, Menon S, Tang J, et al. ACE2 X-ray structures reveal a large hinge-\\nbending motion important for inhibitor binding and catalysis. J Biol Chem. 2004; 279:17996–\\n8007. [PubMed: 14754895] \\n55. Li WH, Zhang CS, Sui JH, Kuhn JH, Moore MJ, et al. Receptor and viral determinants of SARS-\\ncoronavirus adaptation to human ACE2. EMBO J. 2005; 24:1634–43. [PubMed: 15791205] \\n56. Guan Y , Zheng BJ, He YQ, Liu XL, Zhuang ZX, et al. Isolation and characterization of viruses \\nrelated to the SARS coronavirus from animals in southern China. Science. 2003; 302:276–78. \\n[PubMed: 12958366] \\n57. Li F. Structural analysis of major species barriers between humans and palm civets for severe acute \\nrespiratory syndrome coronavirus infections. J Virol. 2008; 82:6984–91. [PubMed: 18448527] \\n58. Wu KL, Peng GQ, Wilken M, Geraghty RJ, Li F. Mechanisms of host receptor adaptation by \\nsevere acute respiratory syndrome coronavirus. J Biol Chem. 2012; 287:8904–11. [PubMed: \\n22291007] \\n59. Wu K, Chen L, Peng G, Zhou W, Pennell CA, et al. A virus-binding hot spot on human \\nangiotensin-converting enzyme 2 is critical for binding of two different coronaviruses. J Virol. \\n2011; 85:5331–37. [PubMed: 21411533] \\n60. Song HD, Tu CC, Zhang GW, Wang SY , Zheng K, et al. Cross-host evolution of severe acute \\nrespiratory syndrome coronavirus in palm civet and human. PNAS. 2005; 102:2430–35. [PubMed: \\n15695582] \\n61. Qu XX, Hao P, Song XJ, Jiang SM, Liu YX, et al. Identification of two critical amino acid residues \\nof the severe acute respiratory syndrome coronavirus spike protein for its variation in zoonotic \\ntropism transition via a double substitution strategy. J Biol Chem. 2005; 280:29588–95. [PubMed: \\n15980414] \\nLi Page 15\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript62. Li WH, Greenough TC, Moore MJ, Vasilieva N, Somasundaran M, et al. Efficient replication of \\nsevere acute respiratory syndrome coronavirus in mouse cells is limited by murine angiotensin-\\nconverting enzyme 2. J Virol. 2004; 78:11429–33. [PubMed: 15452268] \\n63. Frieman M, Yount B, Agnihothram S, Page C, Donaldson E, et al. Molecular determinants of \\nsevere acute respiratory syndrome coronavirus pathogenesis and virulence in young and aged \\nmouse models of human disease. J Virol. 2012; 86:884–97. [PubMed: 22072787] \\n64. Ge XY , Li JL, Yang XL, Chmura AA, Zhu G, et al. Isolation and characterization of a bat SARS-\\nlike coronavirus that uses the ACE2 receptor. Nature. 2013; 503:535–38. [PubMed: 24172901] \\n65. Menachery VD, Yount BL Jr, Debbink K, Agnihothram S, Gralinski LE, et al. A SARS-like cluster \\nof circulating bat coronaviruses shows potential for human emergence. Nat Med. 2015; 21:1508–\\n13. [PubMed: 26552008] \\n66. Li WD, Shi ZL, Yu M, Ren WZ, Smith C, et al. Bats are natural reservoirs of SARS-like \\ncoronaviruses. Science. 2005; 310:676–79. [PubMed: 16195424] \\n67. Lau SKP, Woo PCY , Li KSM, Huang Y , Tsoi HW, et al. Severe acute respiratory syndrome \\ncoronavirus-like virus in Chinese horseshoe bats. PNAS. 2005; 102:14040–45. [PubMed: \\n16169905] \\n68. Hou YX, Peng C, Yu M, Li Y , Han ZG, et al. Angiotensin-converting enzyme 2 (ACE2) proteins of \\ndifferent bat species confer variable susceptibility to SARS-CoV entry. Arch Virol. 2010; \\n155:1563–69. [PubMed: 20567988] \\n69. Lu G, Hu Y , Wang Q, Qi J, Gao F, et al. Molecular basis of binding between novel human \\ncoronavirus MERS-CoV and its receptor CD26. Nature. 2013; 500:227–31. [PubMed: 23831647] \\n70. Wang N, Shi X, Jiang L, Zhang S, Wang D, et al. Structure of MERS-CoV spike receptor-binding \\ndomain complexed with human receptor DPP4. Cell Res. 2013; 23:986–93. [PubMed: 23835475] \\n71. Chen Y , Rajashankar KR, Yang Y , Agnihothram SS, Liu C, et al. Crystal structure of the \\nreceptorbinding domain from newly emerged Middle East respiratory syndrome coronavirus. J \\nVirol. 2013; 87:10777–83. [PubMed: 23903833] \\n72. Rasmussen HB, Branner S, Wiberg FC, Wagtmann N. Crystal structure of human dipeptidyl \\npeptidase IV/CD26 in complex with a substrate analog. Nat Struct Biol. 2003; 10:19–25. \\n[PubMed: 12483204] \\n73. Peck KM, Cockrell AS, Yount BL, Scobey T, Baric RS, Heise MT. Glycosylation of mouse DPP4 \\nplays a role in inhibiting Middle East respiratory syndrome coronavirus infection. J Virol. 2015; \\n89:4696–99. [PubMed: 25653445] \\n74. Cockrell AS, Peck KM, Yount BL, Agnihothram SS, Scobey T, et al. Mouse dipeptidyl peptidase 4 \\nis not a functional receptor for Middle East respiratory syndrome coronavirus infection. J Virol. \\n2014; 88:5195–99. [PubMed: 24574399] \\n75. Fukuma A, Tani H, Taniguchi S, Shimojima M, Saijo M, Fukushi S. Inability of rat DPP4 to allow \\nMERS-CoV infection revealed by using a VSV pseudotype bearing truncated MERS-CoV spike \\nprotein. Arch Virol. 2015; 160:2293–300. [PubMed: 26138557] \\n76. Barlan A, Zhao J, Sarkar MK, Li K, McCray PB Jr, et al. Receptor variation and susceptibility to \\nMiddle East respiratory syndrome coronavirus infection. J Virol. 2014; 88:4953–61. [PubMed: \\n24554656] \\n77. Haagmans BL, Al Dhahiry SH, Reusken CB, Raj VS, Galiano M, et al. Middle East respiratory \\nsyndrome coronavirus in dromedary camels: an outbreak investigation. Lancet Infect Dis. 2014; \\n14:140–45. [PubMed: 24355866] \\n78. Alagaili AN, Briese T, Mishra N, Kapoor V , Sameroff SC, et al. Middle East respiratory syndrome \\ncoronavirus infection in dromedary camels in Saudi Arabia. mBio. 2014; 5:e00884–14. [PubMed: \\n24570370] \\n79. Annan A, Baldwin HJ, Corman VM, Klose SM, Owusu M, et al. Human β-coronavirus 2c EMC/\\n2012-related viruses in bats, Ghana and Europe. Emerg Infect Dis. 2013; 19:456–59. [PubMed: \\n23622767] \\n80. Holmes KV , Dominguez SR. The new age of virus discovery: genomic analysis of a novel human \\nβ-coronavirus isolated from a fatal case of pneumonia. mBio. 2013; 4:e00548–12. [PubMed: \\n23300251] \\nLi Page 16\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript81. Lau SK, Li KS, Tsang AK, Lam CS, Ahmed S, et al. Genetic characterization of Betacoronavirus \\nlineage C viruses in bats reveals marked sequence divergence in the spike protein of Pipistrellus \\nbat coronavirus HKU5 in Japanese pipistrelle: implications for the origin of the novel Middle East \\nrespiratory syndrome coronavirus. J Virol. 2013; 87:8638–50. [PubMed: 23720729] \\n82. Wang Q, Qi J, Yuan Y , Xuan Y , Han P, et al. Bat origins of MERS-CoV supported by bat \\ncoronavirus HKU4 usage of human receptor CD26. Cell Host Microbe. 2014; 16:328–37. \\n[PubMed: 25211075] \\n83. Wu K, Li W, Peng G, Li F. Crystal structure of NL63 respiratory coronavirus receptor-binding \\ndomain complexed with its human receptor. PNAS. 2009; 106:19970–74. [PubMed: 19901337] \\n84. Reguera J, Santiago C, Mudgal G, Ordono D, Enjuanes L, Casasnovas JM. Structural bases of \\ncoronavirus attachment to host aminopeptidase N and its inhibition by neutralizing antibodies. \\nPLOS Pathog. 2012; 8:e1002859. [PubMed: 22876187] \\n85. Chen L, Lin YL, Peng G, Li F. Structural basis for multifunctional roles of mammalian aminopep-\\ntidase N. PNAS. 2012; 109:17966–71. [PubMed: 23071329] \\n86. Wong AH, Zhou D, Rini JM. The X-ray crystal structure of human aminopeptidase N reveals a \\nnovel dimer and the basis for peptide processing. J Biol Chem. 2012; 287:36804–13. [PubMed: \\n22932899] \\n87. Tusell SM, Schittone SA, Holmes KV . Mutational analysis of aminopeptidase N, a receptor for \\nseveral group 1 coronaviruses, identifies key determinants of viral host range. J Virol. 2007; \\n81:1261–73. [PubMed: 17093189] \\n88. Peng GQ, Sun DW, Rajashankar KR, Qian ZH, Holmes KV , Li F\\n\\nsource:27578435\\ncontent:.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptThe structures of S1-NTDs from alpha-, gamma-, and deltacoronaviruses are unavailable, \\nbut on the basis of the following observations, they probably all have a galectin fold. First, \\nthe related structural topology between alpha-and betacoronavirus S1-CTDs suggests that \\nthe S1 subunits across different genera have a common evolutionary origin (42). Second, the \\nS1-NTDs from the alphacoronaviruses TGEV and PEDV and the gammacoronavirus IBV all \\nrecognize sugar receptors, although TGEV S1-NTD recognizes N-glycolylneuraminic acid \\n(Neu5Gc) and N-acetylneuraminic acid (Neu5Ac), PEDV S1-NTD recognizes Neu5Ac, and \\nIBV S1-NTD recognizes Neu5Gc (23, 31, 34, 44). Therefore, the S1-NTDs from different \\ngenera are likely all evolutionarily and structurally related.\\nBased on the above studies, the following evolutionary scenario has been proposed for \\ncoronavirus S1-NTDs (19). Through gene capture, ancestral coronaviruses might have \\nacquired a host galectin, which would become the S1-NTD of their spikes. Consequently, \\ncoronaviruses would recognize sugar receptors for cell entry. To aid viral release from \\ninfected cells, some coronaviruses would also evolve a hemagglutinin-esterase protein (HE) \\nas a receptor-destroying enzyme. Later, coronavirus S1-NTDs would evolve a ceiling-like \\nstructure that could protect their sugar-binding site from host immune surveillance; this \\nceiling-like structure is absent in galectins because as host proteins, galectins are not \\nrecognized by the host immune system. The outer surface of the ceiling-like structure in \\nMHV would further evolve structural features that could function as RBMs and recognize \\nCEACAM1. Because protein receptors in general provide higher affinity and specificity for \\nviral attachment than sugar receptors do, the sugar-binding function of MHV S1-NTD \\nwould become dispensable and be lost. The S1-NTDs of some other contemporary \\ncoronaviruses, such as the alphacoronaviruses TGEV and PEDV , the betacoronaviruses \\nBCoV and OC43, and the gammacoronavirus IBV , have retained the lectin function, but \\ntheir sugar specificities have diverged. The galectin-like domain has also been found in a \\nnumber of other viral spikes (103) (Figure 5d,g), including influenza virus HA1, rotavirus \\nVP4, and the adenovirus galectin domain. These viral galectin-like domains display diverse \\nsugar-binding modes, but their sugar-binding sites are all located in cavities, possibly to \\nevade host immune surveillance. Overall, it appears to be a successful strategy for viruses to \\nacquire a host lectin and evolve it into viral RBDs with novel specificity for a protein \\nreceptor or altered specificity for a different sugar receptor.\\nSTRUCTURAL MECHANISM FOR MEMBRANE FUSION BY CORONAVIRUS \\nSPIKE PROTEINS\\nThe coronavirus spike is believed to be a member of the class I viral membrane fusion \\nproteins that also include those from influenza virus, human immunodeficiency virus (HIV), \\nand Ebola virus (Figure 6a). Among these proteins, the hemagglutinin glycoprotein (HA) of \\nthe influenza virus is arguably the best studied (104, 105). HA is expressed as a single-chain \\nprecursor. During molecular maturation, it trimerizes and is cleaved by host proteases into \\nreceptor-binding subunit HA1 and membrane-fusion subunit HA2, which still associate \\ntogether through noncovalent interactions (106, 107). This prefusion state of HA on the \\nnewly packaged virions is a proteolytically primed and metastable trimer. During cell entry, \\nHA1 binds to a sugar receptor on the host cell surface for viral attachment, and then HA1 \\nLi Page 7\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscriptdissociates and HA2 undergoes a dramatic conformational change to transition to the \\npostfusion state. During this transition, three pairs of heptad repeat regions HR-N and HR-C \\nin trimeric HA2 form a six-helix bundle structure. Three previously buried hydrophobic \\nfusion peptides in trimeric HA2 become exposed and insert into the target host membrane. \\nThe fusion peptides and transmembrane anchors are eventually positioned on the same end \\nof the six-helix bundle, bringing the viral and host membranes together to fuse. Because the \\nsix-helix bundle structure is energetically stable, a large amount of energy is released during \\nthe conformational transition of HA, driving membrane fusion forward. However, an initial \\nenergy barrier for the conformational transition of HA must be overcome through the \\naforementioned proteolytic priming and one or more subsequent triggers. These triggers can \\nbe receptor binding (e.g., HIV) (108), low pH (e.g., influenza virus) (109), or a combination \\nof the two (e.g., avian leucosis virus) (110). Consequently, membrane fusion occurs either \\non the host cell surface (e.g., HIV) or in the endosomes (e.g., influenza virus).\\nThe prefusion structures of betacoronavirus MHV and HKU1 spikes were recently \\ndetermined using cryo-electron microscopy (15, 16) (Figure 2a). The overall architecture of \\nthe prefusion coronavirus spikes is similar to, albeit significantly larger and more complex \\nthan, that of influenza virus HA. In each spike, three S1 heads sit on top of a trimeric S2 \\nstalk, preventing S2 from undergoing conformational transitions. Between the two major S1 \\ndomains, S1-CTD is located at the very top of the spike, whereas S1-NTD directly contacts \\nand structurally constrains S2. In the S2 stalk, HR-N forms several helices and arranges \\nitself along the central threefold symmetry axis of trimeric S2, whereas HR-C is poorly \\nordered. Unlike the fusion peptide in influenza virus HA, which is located at the N terminus \\nof HA2, the fusion peptide in coronavirus spikes is located downstream from the N terminus \\nof S2 and is hence an internal fusion peptide. The coronavirus fusion peptide forms a short \\nhelix and a loop, with most of the hydrophobic residues buried inside the prefusion structure. \\nTwo proteolysis sites are essential for the conformational transition of S2, one at the S1/S2 \\nboundary and the other at the N terminus of the internal fusion peptide (111, 112). Like \\nprefusion influenza virus HA, the prefusion coronavirus spike is in a metastable state and \\nprimed to undergo conformational transitions for membrane fusion.\\nAtomic models of full-length postfusion coronavirus spikes are not available, but a negative-\\nstain electron microscopy study on SARS-CoV spike provided a direct view of its \\nconformational transition that is likely associated with membrane fusion during virus entry \\n(18) (Figure 6b). In vitro triggers (e.g., trypsin cleavage and urea incubation) induce the \\nprefusion SARS-CoV spike to transition to its postfusion state, in which S1 dissociates and \\nS2 forms a dumbbell-shaped structure with a rod-like structure in the middle and a globular \\nstructure at each end. These trimeric S2 molecules further associate at one end to form \\nrosettes in solution. Comparison with influenza virus HA suggests that the rod-like structure \\nin the middle likely represents the six-helix bundle formed by HR-N and HR-C, whereas the \\nglobular structures at the two ends likely correspond to the regions N-terminal to HR-N and \\nbetween HR-N and HR-C, respectively (Figures 1c and 6c). Three hydrophobic fusion \\npeptides, located N-terminal to HR-N, were previously buried in the prefusion spike but \\nbecome exposed in the postfusion S2 and associate to form rosettes in solution (they would \\ninsert into host membranes, were the membranes present). Crystal structures of the six-helix \\nbundle have been determined for a number of coronaviruses such as MHV , SARS-CoV , \\nLi Page 8\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptMERS-CoV , and HCoV-NL63 (113–119) (Figures 1d and 6c). Compared with influenza \\nvirus HA, the six-helix bundle formed by coronavirus HR-N and HR-C is unusually long \\n(18, 117), indicating the abundant amount of energy that can be released during the \\nconformational transition of S2 and available for use in membrane fusion. Overall, these \\nstudies reveal that coronavirus spikes fuse membranes using the same structural mechanism \\nas other class I membrane fusion proteins, but not for lack of some unique features such as \\ntheir large size, internal fusion peptide, double cleavage sites, and long six-helix bundle.\\nTRIGGERS FOR MEMBRANE FUSION BY CORONAVIRUS SPIKE PROTEINS\\nThe triggers for coronavirus spikes to undergo conformational transitions demonstrate a \\nmore complex pattern than many other class I membrane fusion proteins, probably a \\nreflection of their unique structural features discussed above. For example, although \\ninfluenza virus HA is primed by proteolysis during virus packaging, many coronavirus \\nspikes are not. Instead, coronavirus spikes are often subjected to proteolysis later in the cell \\nentry process, sometimes after receptor binding. Thus, proteolysis of coronavirus spikes can \\nlead directly to membrane fusion and thereby serves as an essential trigger for membrane \\nfusion (111, 112). The host proteases that cleave coronavirus spikes mainly come from four \\ndifferent stages of the virus infection cycle: (a) proprotein convertases (e.g., furin) during \\nvirus packaging in virus-producing cells, (b) extracellular proteases (e.g., elastase) after \\nvirus release into extracellular space, (c) cell surface proteases [e.g., type II transmembrane \\nserine protease (TMPRSS2)] after virus attachment to virus-targeting cells, and (d) \\nlysosomal proteases (e.g., cathepsin L and cathepsin B) after virus endocytosis in virus-\\ntargeting cells (120) (Figure 7). In addition to proteolysis, traditional triggers such as \\nreceptor binding and low pH may also play a role in membrane fusion. Here, I discuss \\ndetailed triggers for membrane fusion by the spikes from three representative coronaviruses.\\nAs the prototypic coronavirus, MHV has been extensively examined for its cell entry \\nmechanism. The findings are complicated and in some cases contradictory. First, MHV \\nspike is cleaved by proprotein convertases during virus packaging in the virus-producing \\ncells (121). This proteolysis is critical for MHV entry into virus-targeting cells. Second, the \\nbinding of CEACAM1 triggers the conformational transition of MHV spike and hence \\nmembrane fusion. This is supported by the observations that incubation of MHV spike with \\nrecombinant soluble CEACAM1 led to enhanced hydrophobicity of MHV S2 and the \\nappearance of a protease-resistant S2 fragment (122, 123). These observations indicated the \\nexposure of fusion peptides and formation of the six-helix bundle, respectively, in postfusion \\nS2. Third, there have been contradictory reports about whether MHV enters target cells at \\nthe cell membrane or through endocytosis and whether MHV spike undergoes \\nconformational transitions at low, neutral, or even elevated pH (123–125). This may depend \\non the MHV strains examined or experimental approaches used in the studies. The roles of \\npH and endocytosis in MHV entry still need to be further clarified. Interestingly, the \\nneurotropic strain MHV-JHM can mediate virus entry into host cells that do not express \\nCEACAM1 (126–128). This receptor-independent entry by MHV-JHM is unique among \\nviruses. Biochemical characterization of MHV-JHM spike suggested that it is more labile \\nthan the spikes from other MHV strains, meaning that it undergoes conformational \\ntransitions spontaneously in the absence of the receptor (94, 129, 130). It is believed that \\nLi Page 9\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptMHV-JHM is able to infect neural cells where CEACAM1 expression level is very low, at \\nleast in part because its spike can mediate receptor-independent entry (131, 132). Taken \\ntogether, the membrane fusion mechanism of MHV spike depends on both proteolysis and \\nreceptor binding, and it may or may not depend on the low pH of endosomes; in addition, \\nreceptor-independent membrane fusion by MHV-JHM spike contributes to the neutral \\ntropism of MHV-JHM.\\nResearch on the cell entry mechanism of SARS-CoV has led to novel findings. First, SARS-\\nCoV spike is not cleaved by proprotein convertases during virus packaging and hence \\nremains intact on mature virions (133, 134). Instead, SARS-CoV enters host cells through \\nendocytosis, and its spike is processed by lysosomal proteases (e.g., cathepsin L and \\ncathepsin B) (135–137). This is supported by the observation that inhibitors against either \\nendosomal acidification or lysosomal cysteine proteases block SARS-CoV entry. However, \\nlow pH itself is not a trigger for SARS-CoV entry\\n\\nsource:27578435\\ncontent:. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptMHV-JHM is able to infect neural cells where CEACAM1 expression level is very low, at \\nleast in part because its spike can mediate receptor-independent entry (131, 132). Taken \\ntogether, the membrane fusion mechanism of MHV spike depends on both proteolysis and \\nreceptor binding, and it may or may not depend on the low pH of endosomes; in addition, \\nreceptor-independent membrane fusion by MHV-JHM spike contributes to the neutral \\ntropism of MHV-JHM.\\nResearch on the cell entry mechanism of SARS-CoV has led to novel findings. First, SARS-\\nCoV spike is not cleaved by proprotein convertases during virus packaging and hence \\nremains intact on mature virions (133, 134). Instead, SARS-CoV enters host cells through \\nendocytosis, and its spike is processed by lysosomal proteases (e.g., cathepsin L and \\ncathepsin B) (135–137). This is supported by the observation that inhibitors against either \\nendosomal acidification or lysosomal cysteine proteases block SARS-CoV entry. However, \\nlow pH itself is not a trigger for SARS-CoV entry. This is supported by the observation that \\nwhen expressed on the cell surface and cleaved by exogenous proteases, SARS-CoV spike \\ncan mediate cell-cell fusion with ACE2-expressing cells at neutral pH (136). Thus, the role \\nof low pH in SARS-CoV entry is to activate lysosomal proteases, which further activate \\nSARS-CoV spike for membrane fusion. This is different from influenza virus HA, which is \\nactivated through binding protons in the low-pH environment of endosomes. Second, in \\naddition to lysosomal proteases, both extracellular proteases (e.g., elastases in the respiratory \\ntract) and cell surface proteases (e.g., TMPRSS2 on the surface of lung cells) also activate \\nSARS-CoV spike for membrane fusion (138–142). Due to their cell and tissue specificities, \\nthese proteases likely contribute to the respiratory tract and lung tropism of SARS-CoV . \\nThird, in addition to the cleavage site at the S1/S2 boundary, a second site, S2′, has been \\nidentified at the N terminus of the internal fusion peptide within S2 (143, 144). Whereas the \\ncleavage at the S1/S2 boundary removes the structural constraint of S1 on S2, the cleavage \\nat the S2′ site releases the internal fusion peptide for insertion into target membranes \\n(Figures 1c and 2b). Fourth, it is not clear whether the binding of receptor ACE2 is a trigger \\nfor SARS-CoV spike to fuse membranes. Two electron microscopy studies observed no or \\nmoderate conformational changes of SARS-CoV spike associated with ACE2 binding (18, \\n145). However, other studies suggested that ACE2 binding triggers a conformational change \\nin SARS-CoV spike, which exposes previously cryptic protease sites for cleavage (135, \\n146). The role of ACE2 binding in triggering membrane fusion waits to be further \\ninvestigated. Nevertheless, SARS-CoV entry does not depend on low pH, but it requires at \\nleast two protease cleavages in the spike by lysosomal proteases, extracellular proteases, or \\ncell surface proteases.\\nThe overall cell entry mechanism of MERS-CoV is similar to that of SARS-CoV . Like \\nSARS-CoV spike, MERS-CoV spike must be cleaved at both the S1/S2 boundary and the \\nS2′ site for membrane fusion to occur (147). MERS-CoV also enters host cells through \\nendocytosis and is activated by lysosomal cysteine proteases for membrane fusion (27, 148). \\nMoreover, extracellular proteases and cell surface proteases help activate MERS-CoV entry \\n(148–150). Unlike SARS-CoV , MERS-CoV spike is cleaved by host proprotein convertases \\nduring virus packaging (27, 151). Interestingly, despite recognizing the same receptor DPP4, \\nMERS-CoV and HKU4 spikes differ in their activities to mediate virus entry: HKU4 spike \\nmediates virus entry into bat cells but not human cells, whereas MERS-CoV spike mediates \\nLi Page 10\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscriptvirus entry into both bat and human cells (152). Two residue differences have been identified \\nbetween MERS-CoV and HKU4 spikes that account for this functional difference (152); \\nthey allow MERS-CoV spike, but not HKU4 spike, to be activated by human proprotein \\nconvertases and lysosomal cysteine proteases. Thus, the corresponding two mutations played \\na critical role in the transmission of MERS-CoV from its likely natural reservoir, bats, to \\nhumans, either directly or through intermediate host camels. On the other hand, HKU4 spike \\ncan be activated by bat lysosomal proteases but not human lysosomal proteases, suggesting \\nthat human and bat lysosomal proteases process viral spikes differently. These studies on \\nMERS-CoV entry reveal that different activities of spike-processing proteases from different \\nhosts can pose a barrier for cross-species transmissions of viruses.\\nIn sum, proteolysis has been established as an essential trigger for coronavirus spikes to fuse \\nmembranes, as cleavages at the S1/S2 boundary and S2′ site can remove the structural \\nconstraint of S1 on S2 and release the internal fusion peptide, respectively. Among the host \\nproteases, lysosomal proteases provide the most reliable source for spike processing because \\nthey are ubiquitous and abundant in many cell types. The availability of some other \\nproteases (e.g., proprotein convertases, extracellular proteases, and cell surface proteases) \\ndepends on the types of cells and tissues, regulating tissue tropisms of coronaviruses. \\nMoreover, protease activities from different host species may vary, regulating host ranges of \\ncoronaviruses. Some other triggers for coronavirus entry (e.g., receptor binding and low pH) \\nmay depend on specific coronaviruses or different strains of the same coronavirus. The \\noverall goal of these triggers is to overcome the energy barrier for the conformational \\ntransition of coronavirus spikes.\\nEVOLUTION OF CORONAVIRUS SPIKE PROTEINS\\nCoronavirus spikes, like other class I viral membrane fusion proteins, are amazing \\nmolecules. They single-handedly lead coronaviruses to enter host cells by first binding to a \\nreceptor on the host cell surface and then fusing the viral and host membranes. They exist in \\ntwo distinct conformations: The prefusion trimeric spike contains three receptor-binding S1 \\nheads and a trimeric membranefusion S2 stalk, whereas the postfusion trimeric S2 is a six-\\nhelix bundle with exposed fusion peptides. The transition of the spikes from prefusion to \\npostfusion conformation is regulated by a variety of triggers. Both receptor recognition and \\nmembrane fusion are critical determinants of the host range and tissue tropism of \\ncoronaviruses. How have these complex structures and functions of coronavirus spikes \\nevolved?\\nStructure determinations of coronavirus S1 domains provide insight into the evolution of \\ncoronavirus S1. The finding that betacoronavirus S1-NTDs have a galectin fold indicates a \\nhost origin of coronavirus S1-NTDs. The origin of coronavirus S1-CTDs is less clear. \\nAlphacoronavirus S1-CTDs and host galectins also share some similarity in the structural \\ntopologies of their (β -sandwich folds (Figure 8a), although this similarity is less significant \\nthan that between S1-NTDs and host galectins (Figure 5e,f). β-Sandwich folds are common \\nand stable structures, and two β-sandwich folds may result from convergent evolution with \\nprotein stability as the evolutionary driving force. However, two β-sandwich folds with \\nrelated structural topologies may indicate a common evolutionary ancestor when a \\nLi Page 11\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscriptsignificant number of their constituent β -strands are connected in the same order. Thus, \\nthere is a possibility that S1-CTD and host galectins are evolutionarily related. One possible \\nscenario is that after S1-NTD was generated through gene capture, S1-CTD was generated \\nthrough gene duplication of S1-NTD (Figure 8b). S1-CTDs appear to evolve at a quickened \\npace, as evidenced by the different tertiary structures between alpha-and betacoronavirus S1-\\nCTDs. This may be associated with their location on the very top of the prefusion trimeric \\nspike (Figure 2), which is the most protruding and exposed area on virions. Hence, S1-CTDs \\nare under heavy selective pressure to escape host immune surveillance. The resulting fast-\\npaced evolution of S1-CTDs may have permanently erased their evolutionary traces, except \\nfor the limited information from their structural topology. Whether S1-CTDs originated \\nfrom host galectins or not, the two-domain structure of S1 gives coronaviruses two potential \\nreceptor-binding domains: The more structurally and functionally conserved S1-NTD uses \\nsugar as the fallback receptor, whereas the more aggressively evolving S1-CTD exploits \\nnovel protein receptors (Figure 8).\\nThe structural and functional similarities between coronavirus S2 and other class I viral \\nmembrane fusion proteins are profound. These proteins all exist in prefusion and postfusion \\nconformations. Their prefusion structures can be triggered in a number of similar ways, \\nundergo similar conformational rearrangement, and transition to highly similar postfusion \\nsix-helix bundle structures with exposed fusion peptides. Although it cannot be completely \\nruled out that the same membrane fusion mechanism evolved independently in these viruses, \\nthe complexity and intricacy of this mechanism indicate that class I viral membrane fusion \\nproteins likely share a common evolutionary ancestor.\\nWhich function evolved first for coronaviruses: receptor recognition by S1, membrane \\nfusion by S2, or both simultaneously? Because coronaviruses must enter cells for \\nreplication, membrane fusion is the central function of coronavirus spikes. Receptor \\nrecognition, though, can specifically attach coronaviruses to host cell surfaces and position \\nthe spikes within striking distance of target host membranes. The spike of neurotropic strain \\nMHV-JHM can mediate receptor-independent virus entry into cells that do not express its \\nreceptor, suggesting that receptor binding can be circumvented under some extreme \\nsituations. Therefore, the primordial form of coronavirus spikes might contain S2 only \\n(Figure 8b). Such a primordial spike might function inefficiently because the ancestral virus \\nwould have to diffuse nonspecifically to the close proximity of target cells so that membrane \\nfusion could occur. Later, the spike would evolve a galectin-like S1-NTD through gene \\ncapture, which would enhance its efficiency in mediating virus entry. Next, the spike would \\nevolve an S1-CTD through gene duplication of its S1-NTD or some other mechanism, \\nfurther strengthening its receptor recognition function. Understanding the structure and \\nfunction of coronavirus spikes and their evolution can enhance our understanding of the \\norigin of viruses and the evolutionary relationship between viruses and host cells.\\nAcknowledgments\\nThis work was supported by the National Institutes of Health (R01AI089728, R01AI110700).\\nLi Page 12\\nAnnu Rev Virol. Author manuscript; available in PMC 2017 June 05.\\nAuthor Manuscript Author Manuscript Author Manuscript Author ManuscriptLITERATURE CITED\\n1. Ksiazek TG, Erdman D, Goldsmith CS, Zaki SR, Peret T, et al. Anovel coronavirus associated with \\nsevere acute respiratory syndrome. N Engl J Med. 2003; 348:1953–66. [PubMed: 12690092] \\n2. Peiris JSM, Lai ST, Poon LLM, Guan Y , Yam LYC, et al. Coronavirus as a possible cause of severe \\nacute respiratory syndrome. Lancet. 2003; 361:1319–25. [PubMed: 12711465] \\n3. Marra MA, Jones SJM, Astell CR, Holt RA, Brooks-Wilson A, et al. The genome sequence of the \\nSARS-associated coronavirus. Science. 2003; 300:1399–404. [PubMed: 12730501] \\n4. Rota PA, Oberste MS, Monroe SS, Nix W A, Campagnoli R, et al. Characterization of a novel \\ncoronavirus associated with severe acute respiratory syndrome. Science. 2003; 300:1394–99. \\n[PubMed: 12730500] \\n5. Zaki AM, van Boheemen S, Bestebroer TM, Osterhaus A, Fouchier RAM. Isolation of a novel \\ncoronavirus from a man with pneumonia in Saudi Arabia. N Engl J Med. 2012; 367:1814–20. \\n[PubMed: 23075143] \\n6. de Groot RJ, Baker SC, Baric RS, Brown CS, Drosten C, et al. Middle East respiratory syndrome \\ncoronavirus (MERS-CoV): announcement of the Coronavirus Study Group. J Virol. 2013; 87:7790–\\n92. [PubMed: 23678167] \\n7. Mole B. Deadly pig virus slips through US borders. Nature. 2013; 499:388. [PubMed: 23887408] \\n8. Stevenson GW, Hoang H, Schwartz KJ, Burrough ER, Sun D, et al\""
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "context = \"\\n\\n\".join([f\"source:{r.metadata['source']}\\ncontent:{r.page_content}\" for r in results])\n",
    "context"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mcp-birthday",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
